// @heathmont/moon-components/lib/cardBalance/CardBalance.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _Stack = _interopRequireDefault(require("../stack/Stack"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * Styles
 */
var Unit = _styledComponents["default"].abbr.withConfig({
  displayName: "CardBalance__Unit",
  componentId: "sc-1e6lf9t-0"
})({
  cursor: 'help',
  textDecoration: 'none',
  fontSize: (0, _moonUtils.rem)(16),
  lineHeight: (0, _moonUtils.rem)(24)
});

var Title = _styledComponents["default"].p.withConfig({
  displayName: "CardBalance__Title",
  componentId: "sc-1e6lf9t-1"
})({
  margin: 0,
  fontSize: (0, _moonUtils.rem)(32),
  lineHeight: (0, _moonUtils.rem)(40)
});

var Text = _styledComponents["default"].p.withConfig({
  displayName: "CardBalance__Text",
  componentId: "sc-1e6lf9t-2"
})(function (_ref) {
  var color = _ref.theme.color;
  return {
    margin: 0,
    color: color.trunks[100],
    fontSize: (0, _moonUtils.rem)(16),
    lineHeight: (0, _moonUtils.rem)(24)
  };
});

var Balance = function Balance(_ref2) {
  var from = _ref2.from;
  return /*#__PURE__*/_react["default"].createElement(_Stack["default"], {
    as: "section",
    space: 0
  }, /*#__PURE__*/_react["default"].createElement(Text, null, from.currency), /*#__PURE__*/_react["default"].createElement(Title, null, "".concat(from.value, " "), /*#__PURE__*/_react["default"].createElement(Unit, {
    title: from.unitTitle || from.currency
  }, from.unit)));
};
/**
 * Component
 */


var CardBalance = function CardBalance(_ref3) {
  var from = _ref3.from,
      badge = _ref3.badge;
  return badge ? /*#__PURE__*/_react["default"].createElement(_Stack["default"], null, badge && /*#__PURE__*/_react["default"].createElement("span", null, badge), /*#__PURE__*/_react["default"].createElement(Balance, {
    from: from
  })) : /*#__PURE__*/_react["default"].createElement(Balance, {
    from: from
  });
};

var _default = CardBalance;
exports["default"] = _default;
