// @heathmont/moon-components/lib/tabLink/TabLink.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _rem = _interopRequireDefault(require("polished/lib/helpers/rem"));

var _private = require("../tabs/private");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var TabLink = _styledComponents["default"].a.withConfig({
  displayName: "TabLink",
  componentId: "sc-1p8l4dw-0"
})(function (_ref) {
  var theme = _ref.theme,
      fullWidth = _ref.fullWidth;
  return [fullWidth && {
    display: 'flex',
    flex: 'auto',
    justifyContent: 'center'
  }, {
    position: 'relative',
    padding: 0,
    paddingBottom: "".concat((0, _private.underlineOffset)(theme), "px"),
    margin: 0,
    fontSize: (0, _rem["default"])(14),
    lineHeight: 1.3,
    fontFamily: 'inherit',
    backgroundColor: 'transparent',
    color: theme.color.trunks[100],
    transition: "color ".concat(theme.transitionDuration["default"], "s ease"),
    textDecoration: 'none',
    border: 0,
    appearance: 'none',
    '&:after': {
      content: '""',
      position: 'absolute',
      height: "".concat((0, _private.underlineWidth)(theme), "px"),
      width: '100%',
      left: 0,
      right: 0,
      bottom: 0,
      margin: '0 auto',
      backgroundColor: theme.color.piccolo[100],
      transform: 'scaleX(0)',
      transition: "transform ".concat(theme.transitionDuration["default"], "s ease")
    },
    '&:hover, &:focus, &:active, &.active, &[aria-current=page]': {
      color: theme.color.bulma[100],
      cursor: 'pointer',
      '&:after': {
        transform: 'scaleX(1)'
      }
    }
  }];
});

var _default = TabLink;
exports["default"] = _default;
