// @heathmont/moon-components/lib/es/datepicker/RangeCalendar.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import React from 'react';
import addMonths from 'date-fns/addMonths';
import subMonths from 'date-fns/subMonths';
import endOfDay from 'date-fns/endOfDay';
import format from 'date-fns/format';
import isAfter from 'date-fns/isAfter';
import isSameDay from 'date-fns/isSameDay';
import startOfDay from 'date-fns/startOfDay';
import isValid from 'date-fns/isValid';
import { getMonthDays } from './private/helpers/getMonthDays';
import { Picker } from './private/Picker';
import { getWeekDayLabels } from './private/helpers/getWeekDayLabels';
import { getDatesFromRange } from './private/helpers/getDatesFromRange';
import { getPlaceholder } from './private/helpers/getPlaceholder';

var RangeCalendar = function RangeCalendar(_ref) {
  var initialStartDate = _ref.startDate,
      initialEndDate = _ref.endDate,
      onDateChange = _ref.onDateChange,
      config = _ref.config,
      range = _ref.range,
      translations = _ref.translations,
      hasClickedOutside = _ref.hasClickedOutside,
      setIsOpen = _ref.setIsOpen,
      setPlaceholder = _ref.setPlaceholder;

  var _React$useState = React.useState(_objectSpread({
    startDate: initialStartDate,
    endDate: initialEndDate,
    range: range,
    hoveredDate: undefined,
    cursorDate: initialStartDate ? new Date(initialStartDate) : new Date()
  }, getDatesFromRange({
    range: range,
    config: config,
    fallbackStartDate: initialStartDate,
    fallbackEndDate: initialEndDate
  }))),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      datesState = _React$useState2[0],
      setDatesState = _React$useState2[1];

  React.useEffect(function () {
    if (hasClickedOutside) {
      onDateChange({
        startDate: datesState.startDate,
        endDate: datesState.endDate,
        range: datesState.range
      });
      setPlaceholder(getPlaceholder({
        startDate: datesState.startDate,
        endDate: datesState.endDate,
        range: datesState.range,
        config: config,
        translations: translations
      }));
      setIsOpen(false);
    }
  });

  var apply = function apply() {
    onDateChange({
      startDate: datesState.startDate,
      endDate: datesState.endDate,
      range: datesState.range
    });
    setPlaceholder(getPlaceholder({
      startDate: datesState.startDate,
      endDate: datesState.endDate,
      range: datesState.range,
      config: config,
      translations: translations
    }));
    setIsOpen(false);
  };

  var selectDay = function selectDay(selectedDate) {
    if (datesState.startDate && !datesState.endDate && (isSameDay(datesState.startDate, selectedDate) || isAfter(selectedDate, datesState.startDate))) {
      var newEndDate = endOfDay(selectedDate);
      setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
        hoveredDate: undefined,
        endDate: newEndDate,
        range: ''
      }));
    } else if (datesState.startDate && datesState.endDate) {
      var newStartDate = startOfDay(selectedDate);
      setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
        hoveredDate: undefined,
        startDate: newStartDate,
        endDate: undefined,
        range: ''
      }));
    } else {
      var _newStartDate = startOfDay(selectedDate);

      setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
        hoveredDate: undefined,
        startDate: _newStartDate,
        range: ''
      }));
    }
  };

  var setStartDate = function setStartDate(startDate) {
    if (isValid(startDate) && startDate) {
      setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
        startDate: startDate,
        range: ''
      }));
    }
  };

  var setEndDate = function setEndDate(endDate) {
    if (isValid(endDate) && endDate) {
      setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
        endDate: endDate,
        range: ''
      }));
    }
  };

  var reset = function reset() {
    setDatesState({
      startDate: undefined,
      endDate: undefined,
      range: '',
      hoveredDate: undefined,
      cursorDate: new Date()
    });
  }; // TODO - onClickOutsideHandler


  var hoverDay = function hoverDay(hoveredDate) {
    if (datesState.startDate && !datesState.endDate && !isSameDay(hoveredDate, datesState.startDate) && isAfter(hoveredDate, datesState.startDate)) {
      setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
        hoveredDate: hoveredDate
      }));
    }
  };

  var selectRange = function selectRange(newRange) {
    var _getDatesFromRange = getDatesFromRange({
      range: newRange,
      fallbackStartDate: datesState.startDate,
      fallbackEndDate: datesState.endDate,
      config: config
    }),
        startDate = _getDatesFromRange.startDate,
        endDate = _getDatesFromRange.endDate;

    setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
      startDate: startDate,
      endDate: endDate,
      hoveredDate: undefined,
      range: newRange,
      cursorDate: startDate || datesState.cursorDate
    }));
  };

  var nextMonth = function nextMonth() {
    setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
      cursorDate: addMonths(datesState.cursorDate, 2)
    }));
  };

  var prevMonth = function prevMonth() {
    setDatesState(_objectSpread(_objectSpread({}, datesState), {}, {
      cursorDate: subMonths(datesState.cursorDate, 2)
    }));
  };

  var firstMonth = getMonthDays({
    date: datesState.cursorDate,
    weekStartsOn: config && config.weekStartsOn
  });
  var nextMonthCurrentDate = addMonths(datesState.cursorDate, 1);
  var secondMonth = getMonthDays({
    date: nextMonthCurrentDate,
    weekStartsOn: config && config.weekStartsOn
  });
  var labels = {
    firstMonthLabel: format(datesState.cursorDate, config && config.monthLabelFormat || 'MMMM', config),
    firstMonthYearLabel: format(datesState.cursorDate, config && config.yearLabelFormat || 'yyyy', config),
    secondMonthLabel: format(nextMonthCurrentDate, config && config.monthLabelFormat || 'MMMM', config),
    secondMonthYearLabel: format(nextMonthCurrentDate, config && config.yearLabelFormat || 'yyyy', config),
    weekDayLabels: config && getWeekDayLabels(config)
  };
  return /*#__PURE__*/React.createElement(Picker, {
    datesState: datesState,
    labels: labels,
    translations: translations,
    config: config,
    onDayClick: selectDay,
    firstMonth: firstMonth,
    secondMonth: secondMonth,
    onDayMouseEnter: hoverDay,
    selectRange: selectRange,
    nextMonth: nextMonth,
    prevMonth: prevMonth,
    setStartDate: setStartDate,
    setEndDate: setEndDate,
    onDateChange: onDateChange,
    apply: apply,
    reset: reset
  });
};

export default RangeCalendar;
