// @heathmont/moon-components/lib/tooltip/Tooltip.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _IconClose = _interopRequireDefault(require("../private/icons/IconClose"));

var _Toggle = _interopRequireDefault(require("../toggle/Toggle"));

var _animation = require("./animation");

var _excluded = ["onClose", "active", "id", "children"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/**
 * Settings
 */
var tooltipSpace = '--tooltip-space';
var tooltipIconSize = '0.8rem';
var tooltipIconOffset = "calc(var(".concat(tooltipSpace, ") + ").concat(tooltipIconSize, ")");
var tooltipAnimation = (0, _styledComponents.css)(["animation:", " ", "s ease-in-out;animation-fill-mode:forwards;"], _animation.slideUpFade, function (_ref) {
  var theme = _ref.theme;
  return theme.transitionDuration["default"];
});
/**
 * Styles
 */

var TooltipBubble = _styledComponents["default"].div.withConfig({
  displayName: "Tooltip__TooltipBubble",
  componentId: "al2zmf-0"
})(function (_ref2) {
  var _ref3;

  var active = _ref2.active,
      _ref2$align = _ref2.align,
      align = _ref2$align === void 0 ? 'left' : _ref2$align,
      error = _ref2.error,
      _ref2$theme = _ref2.theme,
      border = _ref2$theme.border,
      boxShadow = _ref2$theme.boxShadow,
      color = _ref2$theme.color,
      radius = _ref2$theme.radius,
      space = _ref2$theme.space,
      zIndex = _ref2$theme.zIndex;
  return [(_ref3 = {}, _defineProperty(_ref3, tooltipSpace, (0, _moonUtils.rem)(space.small)), _defineProperty(_ref3, align, 0), _defineProperty(_ref3, "border", border), _defineProperty(_ref3, "boxShadow", boxShadow), _defineProperty(_ref3, "zIndex", zIndex.toggle), _defineProperty(_ref3, "position", 'absolute'), _defineProperty(_ref3, "maxWidth", '100%'), _defineProperty(_ref3, "padding", "var(".concat(tooltipSpace, ")")), _defineProperty(_ref3, "marginTop", "var(".concat(tooltipSpace, ")")), _defineProperty(_ref3, "backgroundColor", !error ? color.goku[100] : color.chiChi[100]), _defineProperty(_ref3, "borderColor", color.beerus[100]), _defineProperty(_ref3, "borderRadius", (0, _moonUtils.rem)(radius.small)), _defineProperty(_ref3, "color", !error ? color.bulma[100] : color.goten[100]), _defineProperty(_ref3, "willChange", 'transform'), _defineProperty(_ref3, "".concat(_IconClose["default"]), {
    color: color.trunks[100]
  }), _ref3), tooltipAnimation, active && {
    animationDirection: 'reverse'
  }];
});

var TooltipCaption = _styledComponents["default"].div.withConfig({
  displayName: "Tooltip__TooltipCaption",
  componentId: "al2zmf-1"
})(function (_ref4) {
  var onClose = _ref4.onClose;
  return [{
    position: 'relative'
  }, typeof onClose === 'function' && {
    paddingRight: tooltipIconOffset
  }];
});

var tooltipToggle = {
  display: 'inline-flex',
  position: 'absolute',
  top: '50%',
  transform: 'translateY(-50%)',
  right: 0,
  fontSize: tooltipIconSize
};
/**
 * Component
 */

var Tooltip = function Tooltip(_ref5) {
  var onClose = _ref5.onClose,
      active = _ref5.active,
      id = _ref5.id,
      children = _ref5.children,
      props = _objectWithoutProperties(_ref5, _excluded);

  return /*#__PURE__*/_react["default"].createElement("div", {
    "aria-hidden": !active,
    role: "tooltip",
    id: id,
    style: {
      position: 'relative'
    }
  }, /*#__PURE__*/_react["default"].createElement(TooltipBubble, _extends({
    active: active
  }, props), /*#__PURE__*/_react["default"].createElement(TooltipCaption, {
    onClose: onClose
  }, children, typeof onClose === 'function' && /*#__PURE__*/_react["default"].createElement(_Toggle["default"], {
    topRight: false,
    style: tooltipToggle
  }, /*#__PURE__*/_react["default"].createElement(_IconClose["default"], {
    onClick: onClose,
    fontSize: "inherit"
  })))));
};

var _default = Tooltip;
exports["default"] = _default;
