// @heathmont/moon-themes/lib/es/commsLight/commsLight.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var boxShadow = '0px 2.4px 7.2px rgba(0, 0, 0, 0.18), 0px 12.8px 28.8px rgba(0, 0, 0, 0.22)';
var color = {
  piccolo: {
    120: '#141519',
    100: '#272D36',
    80: '#4D5259'
  },
  hit: {
    120: '#D6DDE0',
    100: '#F3F5F6',
    80: '#E9EDEF'
  },
  beerus: {
    100: '#E2E6E9'
  },
  goku: {
    120: '#D6DDE0',
    100: '#F3F5F6',
    80: '#E9EDEF',
    10: '#F9FAFA'
  },
  gohan: {
    100: '#FFFFFF'
  },
  goten: {
    100: '#FFFFFF'
  },
  bulma: {
    100: '#272D36'
  },
  trunks: {
    100: '#8697A2'
  }
};

var commsLight = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'Comms',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  boxShadow: boxShadow,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  radius: {
    small: 8,
    "default": 16,
    largest: 24
  },
  colorScheme: 'light'
});

export default commsLight;
