// @heathmont/moon-assets/duotones/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  CasinoBonus: true,
  ClaimedReward: true,
  DynamicRewards: true,
  EmailNotVerified: true,
  EmailVerification: true,
  EventEnded: true,
  Gift: true,
  HustleToTop: true,
  InviteFriends: true,
  MoneyBack: true,
  NoBets: true,
  NoCasinoGames: true,
  NoData: true,
  NoEvents: true,
  NoFavorite: true,
  NoFreeBet: true,
  NoInternet: true,
  NoMarkets: true,
  NoNotification: true,
  NoParticipations: true,
  NoReward: true,
  NoWithdrawals: true,
  PriceBoost: true,
  TiersAndRewards: true,
  TrackYourProgress: true,
  Trophy: true,
  YouPlayWePay: true
};
Object.defineProperty(exports, "CasinoBonus", {
  enumerable: true,
  get: function get() {
    return _CasinoBonus["default"];
  }
});
Object.defineProperty(exports, "ClaimedReward", {
  enumerable: true,
  get: function get() {
    return _ClaimedReward["default"];
  }
});
Object.defineProperty(exports, "DynamicRewards", {
  enumerable: true,
  get: function get() {
    return _DynamicRewards["default"];
  }
});
Object.defineProperty(exports, "EmailNotVerified", {
  enumerable: true,
  get: function get() {
    return _EmailNotVerified["default"];
  }
});
Object.defineProperty(exports, "EmailVerification", {
  enumerable: true,
  get: function get() {
    return _EmailVerification["default"];
  }
});
Object.defineProperty(exports, "EventEnded", {
  enumerable: true,
  get: function get() {
    return _EventEnded["default"];
  }
});
Object.defineProperty(exports, "Gift", {
  enumerable: true,
  get: function get() {
    return _Gift["default"];
  }
});
Object.defineProperty(exports, "HustleToTop", {
  enumerable: true,
  get: function get() {
    return _HustleToTop["default"];
  }
});
Object.defineProperty(exports, "InviteFriends", {
  enumerable: true,
  get: function get() {
    return _InviteFriends["default"];
  }
});
Object.defineProperty(exports, "MoneyBack", {
  enumerable: true,
  get: function get() {
    return _MoneyBack["default"];
  }
});
Object.defineProperty(exports, "NoBets", {
  enumerable: true,
  get: function get() {
    return _NoBets["default"];
  }
});
Object.defineProperty(exports, "NoCasinoGames", {
  enumerable: true,
  get: function get() {
    return _NoCasinoGames["default"];
  }
});
Object.defineProperty(exports, "NoData", {
  enumerable: true,
  get: function get() {
    return _NoData["default"];
  }
});
Object.defineProperty(exports, "NoEvents", {
  enumerable: true,
  get: function get() {
    return _NoEvents["default"];
  }
});
Object.defineProperty(exports, "NoFavorite", {
  enumerable: true,
  get: function get() {
    return _NoFavorite["default"];
  }
});
Object.defineProperty(exports, "NoFreeBet", {
  enumerable: true,
  get: function get() {
    return _NoFreeBet["default"];
  }
});
Object.defineProperty(exports, "NoInternet", {
  enumerable: true,
  get: function get() {
    return _NoInternet["default"];
  }
});
Object.defineProperty(exports, "NoMarkets", {
  enumerable: true,
  get: function get() {
    return _NoMarkets["default"];
  }
});
Object.defineProperty(exports, "NoNotification", {
  enumerable: true,
  get: function get() {
    return _NoNotification["default"];
  }
});
Object.defineProperty(exports, "NoParticipations", {
  enumerable: true,
  get: function get() {
    return _NoParticipations["default"];
  }
});
Object.defineProperty(exports, "NoReward", {
  enumerable: true,
  get: function get() {
    return _NoReward["default"];
  }
});
Object.defineProperty(exports, "NoWithdrawals", {
  enumerable: true,
  get: function get() {
    return _NoWithdrawals["default"];
  }
});
Object.defineProperty(exports, "PriceBoost", {
  enumerable: true,
  get: function get() {
    return _PriceBoost["default"];
  }
});
Object.defineProperty(exports, "TiersAndRewards", {
  enumerable: true,
  get: function get() {
    return _TiersAndRewards["default"];
  }
});
Object.defineProperty(exports, "TrackYourProgress", {
  enumerable: true,
  get: function get() {
    return _TrackYourProgress["default"];
  }
});
Object.defineProperty(exports, "Trophy", {
  enumerable: true,
  get: function get() {
    return _Trophy["default"];
  }
});
Object.defineProperty(exports, "YouPlayWePay", {
  enumerable: true,
  get: function get() {
    return _YouPlayWePay["default"];
  }
});

var _CasinoBonus = _interopRequireWildcard(require("./CasinoBonus"));

Object.keys(_CasinoBonus).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _CasinoBonus[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _CasinoBonus[key];
    }
  });
});

var _ClaimedReward = _interopRequireWildcard(require("./ClaimedReward"));

Object.keys(_ClaimedReward).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ClaimedReward[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _ClaimedReward[key];
    }
  });
});

var _DynamicRewards = _interopRequireWildcard(require("./DynamicRewards"));

Object.keys(_DynamicRewards).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _DynamicRewards[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _DynamicRewards[key];
    }
  });
});

var _EmailNotVerified = _interopRequireWildcard(require("./EmailNotVerified"));

Object.keys(_EmailNotVerified).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _EmailNotVerified[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _EmailNotVerified[key];
    }
  });
});

var _EmailVerification = _interopRequireWildcard(require("./EmailVerification"));

Object.keys(_EmailVerification).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _EmailVerification[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _EmailVerification[key];
    }
  });
});

var _EventEnded = _interopRequireWildcard(require("./EventEnded"));

Object.keys(_EventEnded).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _EventEnded[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _EventEnded[key];
    }
  });
});

var _Gift = _interopRequireWildcard(require("./Gift"));

Object.keys(_Gift).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Gift[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Gift[key];
    }
  });
});

var _HustleToTop = _interopRequireWildcard(require("./HustleToTop"));

Object.keys(_HustleToTop).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _HustleToTop[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _HustleToTop[key];
    }
  });
});

var _InviteFriends = _interopRequireWildcard(require("./InviteFriends"));

Object.keys(_InviteFriends).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _InviteFriends[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _InviteFriends[key];
    }
  });
});

var _MoneyBack = _interopRequireWildcard(require("./MoneyBack"));

Object.keys(_MoneyBack).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _MoneyBack[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _MoneyBack[key];
    }
  });
});

var _NoBets = _interopRequireWildcard(require("./NoBets"));

Object.keys(_NoBets).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoBets[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoBets[key];
    }
  });
});

var _NoCasinoGames = _interopRequireWildcard(require("./NoCasinoGames"));

Object.keys(_NoCasinoGames).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoCasinoGames[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoCasinoGames[key];
    }
  });
});

var _NoData = _interopRequireWildcard(require("./NoData"));

Object.keys(_NoData).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoData[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoData[key];
    }
  });
});

var _NoEvents = _interopRequireWildcard(require("./NoEvents"));

Object.keys(_NoEvents).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoEvents[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoEvents[key];
    }
  });
});

var _NoFavorite = _interopRequireWildcard(require("./NoFavorite"));

Object.keys(_NoFavorite).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoFavorite[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoFavorite[key];
    }
  });
});

var _NoFreeBet = _interopRequireWildcard(require("./NoFreeBet"));

Object.keys(_NoFreeBet).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoFreeBet[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoFreeBet[key];
    }
  });
});

var _NoInternet = _interopRequireWildcard(require("./NoInternet"));

Object.keys(_NoInternet).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoInternet[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoInternet[key];
    }
  });
});

var _NoMarkets = _interopRequireWildcard(require("./NoMarkets"));

Object.keys(_NoMarkets).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoMarkets[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoMarkets[key];
    }
  });
});

var _NoNotification = _interopRequireWildcard(require("./NoNotification"));

Object.keys(_NoNotification).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoNotification[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoNotification[key];
    }
  });
});

var _NoParticipations = _interopRequireWildcard(require("./NoParticipations"));

Object.keys(_NoParticipations).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoParticipations[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoParticipations[key];
    }
  });
});

var _NoReward = _interopRequireWildcard(require("./NoReward"));

Object.keys(_NoReward).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoReward[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoReward[key];
    }
  });
});

var _NoWithdrawals = _interopRequireWildcard(require("./NoWithdrawals"));

Object.keys(_NoWithdrawals).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _NoWithdrawals[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _NoWithdrawals[key];
    }
  });
});

var _PriceBoost = _interopRequireWildcard(require("./PriceBoost"));

Object.keys(_PriceBoost).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PriceBoost[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _PriceBoost[key];
    }
  });
});

var _TiersAndRewards = _interopRequireWildcard(require("./TiersAndRewards"));

Object.keys(_TiersAndRewards).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _TiersAndRewards[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _TiersAndRewards[key];
    }
  });
});

var _TrackYourProgress = _interopRequireWildcard(require("./TrackYourProgress"));

Object.keys(_TrackYourProgress).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _TrackYourProgress[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _TrackYourProgress[key];
    }
  });
});

var _Trophy = _interopRequireWildcard(require("./Trophy"));

Object.keys(_Trophy).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Trophy[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Trophy[key];
    }
  });
});

var _YouPlayWePay = _interopRequireWildcard(require("./YouPlayWePay"));

Object.keys(_YouPlayWePay).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _YouPlayWePay[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _YouPlayWePay[key];
    }
  });
});

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
