// @heathmont/moon-components/lib/numberRange/NumberRange.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _TextInput = _interopRequireDefault(require("../textInput/TextInput"));

var _Text = _interopRequireDefault(require("../text/Text"));

var _Stack = _interopRequireDefault(require("../stack/Stack"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Container = _styledComponents["default"].div.withConfig({
  displayName: "NumberRange__Container",
  componentId: "sc-1hlq9gz-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      space = _ref$theme.space,
      radius = _ref$theme.radius,
      boxShadow = _ref$theme.boxShadow;
  return {
    background: color.gohan[100],
    padding: (0, _moonUtils.rem)(space["default"]),
    borderRadius: (0, _moonUtils.rem)(radius["default"]),
    boxShadow: boxShadow,
    '& > :first-child': {
      marginBottom: (0, _moonUtils.rem)(12)
    }
  };
});

var NumberRange = function NumberRange(_ref2) {
  var _ref2$text = _ref2.text,
      text = _ref2$text === void 0 ? 'Enter a minimum, maximum or range limit' : _ref2$text,
      _ref2$lteLabel = _ref2.lteLabel,
      lteLabel = _ref2$lteLabel === void 0 ? 'Max' : _ref2$lteLabel,
      _ref2$gteLabel = _ref2.gteLabel,
      gteLabel = _ref2$gteLabel === void 0 ? 'Min' : _ref2$gteLabel,
      _ref2$lteError = _ref2.lteError,
      lteError = _ref2$lteError === void 0 ? false : _ref2$lteError,
      _ref2$gteError = _ref2.gteError,
      gteError = _ref2$gteError === void 0 ? false : _ref2$gteError,
      _ref2$lteDisabled = _ref2.lteDisabled,
      lteDisabled = _ref2$lteDisabled === void 0 ? false : _ref2$lteDisabled,
      _ref2$gteDisabled = _ref2.gteDisabled,
      gteDisabled = _ref2$gteDisabled === void 0 ? false : _ref2$gteDisabled,
      value = _ref2.value,
      _onChange = _ref2.onChange,
      _ref2$maxWidth = _ref2.maxWidth,
      maxWidth = _ref2$maxWidth === void 0 ? 248 : _ref2$maxWidth;
  return /*#__PURE__*/_react["default"].createElement(Container, {
    style: {
      maxWidth: maxWidth
    }
  }, /*#__PURE__*/_react["default"].createElement(_Text["default"], {
    color: "trunks.100",
    size: 12
  }, text), /*#__PURE__*/_react["default"].createElement(_Stack["default"], null, /*#__PURE__*/_react["default"].createElement(_TextInput["default"], {
    type: "number",
    label: gteLabel,
    value: String(value.gte),
    disabled: gteDisabled,
    onChange: function onChange(e) {
      return _onChange({
        gte: Number(e.target.value),
        lte: value.lte
      });
    },
    error: gteError
  }), /*#__PURE__*/_react["default"].createElement(_TextInput["default"], {
    type: "number",
    label: lteLabel,
    value: String(value.lte),
    disabled: lteDisabled,
    onChange: function onChange(e) {
      return _onChange({
        lte: Number(e.target.value),
        gte: value.gte
      });
    },
    error: lteError
  })));
};

var _default = NumberRange;
exports["default"] = _default;
