// @heathmont/moon-components/lib/es/checkboxMultiselect/CheckboxMultiselect.js
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

import { rem } from '@heathmont/moon-utils';
import React from 'react';
import styled from 'styled-components';
import Checkbox from '../checkbox/Checkbox';
import Inline from '../inline/Inline';
import Text from '../text/Text';
var List = styled.ul.withConfig({
  displayName: "CheckboxMultiselect__List",
  componentId: "e8lanu-0"
})(function () {
  return {
    overflow: 'auto',
    listStyle: 'none',
    padding: 0,
    margin: "0 0 0 -".concat(rem(12))
  };
});
var Option = styled.label.withConfig({
  displayName: "CheckboxMultiselect__Option",
  componentId: "e8lanu-1"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    display: 'flex',
    flex: 1,
    alignItems: 'center',
    fontSize: rem(14),
    lineHeight: rem(20),
    padding: "".concat(rem(10), " ").concat(rem(12)),
    margin: 0,
    textDecoration: 'none',
    outline: 'none',
    border: 'none',
    textAlign: 'left',
    cursor: 'pointer',
    color: theme.color.bulma[100],
    backgroundColor: 'transparent',
    'span::before': {
      display: 'none'
    },
    '&:hover, &:focus': {
      backgroundColor: theme.color.goku[100]
    }
  };
});

var CheckboxMultiselect = function CheckboxMultiselect(_ref2) {
  var options = _ref2.options,
      _ref2$value = _ref2.value,
      value = _ref2$value === void 0 ? [] : _ref2$value,
      onChange = _ref2.onChange,
      maxHeight = _ref2.maxHeight;

  var onCheckboxChange = function onCheckboxChange(optionValue) {
    if (value.includes(optionValue)) {
      onChange(value.filter(function (currentValue) {
        return currentValue !== optionValue;
      }));
    } else {
      onChange([].concat(_toConsumableArray(value), [optionValue]));
    }
  };

  return /*#__PURE__*/React.createElement(List, {
    style: {
      maxHeight: maxHeight
    }
  }, options.map(function (option) {
    return /*#__PURE__*/React.createElement(Inline, {
      key: option.value,
      space: "xsmall"
    }, /*#__PURE__*/React.createElement(Option, null, /*#__PURE__*/React.createElement(Checkbox, {
      id: option.id,
      checked: value.some(function (currentValue) {
        return currentValue === option.value;
      }),
      onChange: function onChange() {
        return onCheckboxChange(option.value);
      },
      disabled: option.count === undefined ? false : !option.count,
      label: option.label
    })), option.count !== undefined && /*#__PURE__*/React.createElement(Text, {
      size: 14,
      color: "trunks.100",
      "test-id": "select-count"
    }, option.count));
  }));
};

CheckboxMultiselect.defaultProps = {
  maxHeight: 250
};
export default CheckboxMultiselect;
