// @heathmont/moon-themes/lib/es/luckyslotsDark/luckyslotsDark.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var color = {
  piccolo: {
    120: '#E00038',
    100: '#FA003E',
    80: '#F32055'
  },
  hit: {
    120: '#51516C',
    100: '#696987',
    80: '#7F7F9E'
  },
  goten: {
    100: '#FFFFFF'
  },
  goku: {
    100: '#232432',
    80: '#303245',
    40: '#3B3D51',
    10: '#4A4C65'
  },
  gohan: {
    100: '#3A3A50',
    80: '#4A5167',
    40: '#555E75',
    10: '#646D86'
  },
  beerus: {
    100: '##525260, 100%'
  },
  bulma: {
    100: '#FFFFFF'
  },
  trunks: {
    100: '#DBE2F5'
  }
};

var luckyslotsDark = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'LuckySlots',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), {}, {
    frieza: {
      100: '#262FB7',
      10: '#5159C5'
    }
  }, color),
  colorScheme: 'dark'
});

export default luckyslotsDark;
