// @heathmont/moon-core/lib/badge/Badge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var StyledBadge = _styledComponents["default"].span.withConfig({
  displayName: "Badge__StyledBadge",
  componentId: "fdg9gi-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      fontWeight = _ref$theme.fontWeight,
      radius = _ref$theme.radius;
  return {
    display: 'flex',
    alignItems: 'center',
    fontWeight: fontWeight.semibold,
    textTransform: 'uppercase',
    borderRadius: (0, _moonUtils.rem)(radius.largest),
    letterSpacing: (0, _moonUtils.rem)(1) // TODO

  };
}, function (_ref2) {
  var size = _ref2.size;
  return {
    padding: size === 'small' ? "0 ".concat((0, _moonUtils.rem)(4)) : "0 ".concat((0, _moonUtils.rem)(8)),
    fontSize: size === 'small' ? (0, _moonUtils.rem)(8) : (0, _moonUtils.rem)(10),
    lineHeight: size === 'small' ? (0, _moonUtils.rem)(12) : (0, _moonUtils.rem)(16)
  };
},
/* If a color or backgroundColor is set, override the modifier styles. */
function (_ref3) {
  var color = _ref3.color,
      backgroundColor = _ref3.backgroundColor,
      theme = _ref3.theme;
  return {
    color: color ? (0, _moonUtils.themed)('color', color)(theme) : theme.color.goten[100],
    backgroundColor: (0, _moonUtils.themed)('color', backgroundColor)(theme)
  };
});

StyledBadge.defaultProps = {
  color: 'goten.100',
  backgroundColor: 'piccolo.100'
};

var IconRightWrapper = _styledComponents["default"].span.withConfig({
  displayName: "Badge__IconRightWrapper",
  componentId: "fdg9gi-1"
})(["margin-left:", ";"], (0, _moonUtils.rem)(8));

var IconLeftWrapper = _styledComponents["default"].span.withConfig({
  displayName: "Badge__IconLeftWrapper",
  componentId: "fdg9gi-2"
})(["margin-right:", ";"], (0, _moonUtils.rem)(8));

var Badge = function Badge(_ref4) {
  var children = _ref4.children,
      color = _ref4.color,
      backgroundColor = _ref4.backgroundColor,
      iconLeft = _ref4.iconLeft,
      iconRight = _ref4.iconRight,
      size = _ref4.size;
  return /*#__PURE__*/_react["default"].createElement(StyledBadge, {
    backgroundColor: backgroundColor,
    color: color,
    iconLeft: iconLeft,
    iconRight: iconRight,
    size: size
  }, iconLeft && /*#__PURE__*/_react["default"].createElement(IconLeftWrapper, {
    size: size
  }, iconLeft), children, iconRight && /*#__PURE__*/_react["default"].createElement(IconRightWrapper, {
    size: size
  }, iconRight));
};
/*
default
color - goten
backgroundColor - piccolo.100
*/


var _default = Badge;
exports["default"] = _default;
