// @heathmont/moon-themes/lib/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  aposta10Dark: true,
  aposta10Light: true,
  betaddaDark: true,
  betaddaLight: true,
  bitcasinoDark: true,
  bitcasinoLight: true,
  bombayClub: true,
  btcxe: true,
  commsDark: true,
  commsLight: true,
  empireDark: true,
  empireLight: true,
  hub88: true,
  livecasinoDark: true,
  livecasinoLight: true,
  luckyslots: true,
  luckyslotsDark: true,
  luckyslotsLight: true,
  missionsToolDark: true,
  missionsToolLight: true,
  moneyballDark: true,
  moneyballLight: true,
  moonDesignDark: true,
  moonDesignLight: true,
  shangrilaDark: true,
  shangrilaLight: true,
  avertaStd: true,
  sharedTokens: true,
  slotsDark: true,
  slotsLight: true,
  sportsbetDark: true,
  sportsbetLight: true,
  supportColors: true,
  tajCasinoDark: true,
  tajCasinoLight: true,
  ThemeContext: true,
  ThemeProvider: true,
  ThemeTransition: true,
  useTheme: true
};
Object.defineProperty(exports, "aposta10Dark", {
  enumerable: true,
  get: function get() {
    return _aposta10Dark["default"];
  }
});
Object.defineProperty(exports, "aposta10Light", {
  enumerable: true,
  get: function get() {
    return _aposta10Light["default"];
  }
});
Object.defineProperty(exports, "betaddaDark", {
  enumerable: true,
  get: function get() {
    return _betaddaDark["default"];
  }
});
Object.defineProperty(exports, "betaddaLight", {
  enumerable: true,
  get: function get() {
    return _betaddaLight["default"];
  }
});
Object.defineProperty(exports, "bitcasinoDark", {
  enumerable: true,
  get: function get() {
    return _bitcasinoDark["default"];
  }
});
Object.defineProperty(exports, "bitcasinoLight", {
  enumerable: true,
  get: function get() {
    return _bitcasinoLight["default"];
  }
});
Object.defineProperty(exports, "bombayClub", {
  enumerable: true,
  get: function get() {
    return _bombayClub["default"];
  }
});
Object.defineProperty(exports, "btcxe", {
  enumerable: true,
  get: function get() {
    return _btcxe["default"];
  }
});
Object.defineProperty(exports, "commsDark", {
  enumerable: true,
  get: function get() {
    return _commsDark["default"];
  }
});
Object.defineProperty(exports, "commsLight", {
  enumerable: true,
  get: function get() {
    return _commsLight["default"];
  }
});
Object.defineProperty(exports, "empireDark", {
  enumerable: true,
  get: function get() {
    return _empireDark["default"];
  }
});
Object.defineProperty(exports, "empireLight", {
  enumerable: true,
  get: function get() {
    return _empireLight["default"];
  }
});
Object.defineProperty(exports, "hub88", {
  enumerable: true,
  get: function get() {
    return _hub["default"];
  }
});
Object.defineProperty(exports, "livecasinoDark", {
  enumerable: true,
  get: function get() {
    return _livecasinoDark["default"];
  }
});
Object.defineProperty(exports, "livecasinoLight", {
  enumerable: true,
  get: function get() {
    return _livecasinoLight["default"];
  }
});
Object.defineProperty(exports, "luckyslots", {
  enumerable: true,
  get: function get() {
    return _luckyslots["default"];
  }
});
Object.defineProperty(exports, "luckyslotsDark", {
  enumerable: true,
  get: function get() {
    return _luckyslotsDark["default"];
  }
});
Object.defineProperty(exports, "luckyslotsLight", {
  enumerable: true,
  get: function get() {
    return _luckyslotsLight["default"];
  }
});
Object.defineProperty(exports, "missionsToolDark", {
  enumerable: true,
  get: function get() {
    return _missionsToolDark["default"];
  }
});
Object.defineProperty(exports, "missionsToolLight", {
  enumerable: true,
  get: function get() {
    return _missionsToolLight["default"];
  }
});
Object.defineProperty(exports, "moneyballDark", {
  enumerable: true,
  get: function get() {
    return _moneyballDark["default"];
  }
});
Object.defineProperty(exports, "moneyballLight", {
  enumerable: true,
  get: function get() {
    return _moneyballLight["default"];
  }
});
Object.defineProperty(exports, "moonDesignDark", {
  enumerable: true,
  get: function get() {
    return _moonDesignDark["default"];
  }
});
Object.defineProperty(exports, "moonDesignLight", {
  enumerable: true,
  get: function get() {
    return _moonDesignLight["default"];
  }
});
Object.defineProperty(exports, "shangrilaDark", {
  enumerable: true,
  get: function get() {
    return _shangrilaDark["default"];
  }
});
Object.defineProperty(exports, "shangrilaLight", {
  enumerable: true,
  get: function get() {
    return _shangrilaLight["default"];
  }
});
Object.defineProperty(exports, "avertaStd", {
  enumerable: true,
  get: function get() {
    return _avertaStd["default"];
  }
});
Object.defineProperty(exports, "sharedTokens", {
  enumerable: true,
  get: function get() {
    return _sharedTokens["default"];
  }
});
Object.defineProperty(exports, "slotsDark", {
  enumerable: true,
  get: function get() {
    return _slotsDark["default"];
  }
});
Object.defineProperty(exports, "slotsLight", {
  enumerable: true,
  get: function get() {
    return _slotsLight["default"];
  }
});
Object.defineProperty(exports, "sportsbetDark", {
  enumerable: true,
  get: function get() {
    return _sportsbetDark["default"];
  }
});
Object.defineProperty(exports, "sportsbetLight", {
  enumerable: true,
  get: function get() {
    return _sportsbetLight["default"];
  }
});
Object.defineProperty(exports, "supportColors", {
  enumerable: true,
  get: function get() {
    return _supportColors["default"];
  }
});
Object.defineProperty(exports, "tajCasinoDark", {
  enumerable: true,
  get: function get() {
    return _tajCasinoDark["default"];
  }
});
Object.defineProperty(exports, "tajCasinoLight", {
  enumerable: true,
  get: function get() {
    return _tajCasinoLight["default"];
  }
});
Object.defineProperty(exports, "ThemeContext", {
  enumerable: true,
  get: function get() {
    return _ThemeContext["default"];
  }
});
Object.defineProperty(exports, "ThemeProvider", {
  enumerable: true,
  get: function get() {
    return _ThemeProvider["default"];
  }
});
Object.defineProperty(exports, "ThemeTransition", {
  enumerable: true,
  get: function get() {
    return _ThemeTransition["default"];
  }
});
Object.defineProperty(exports, "useTheme", {
  enumerable: true,
  get: function get() {
    return _useTheme["default"];
  }
});

var _aposta10Dark = _interopRequireWildcard(require("./aposta10Dark/aposta10Dark"));

Object.keys(_aposta10Dark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _aposta10Dark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _aposta10Dark[key];
    }
  });
});

var _aposta10Light = _interopRequireWildcard(require("./aposta10Light/aposta10Light"));

Object.keys(_aposta10Light).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _aposta10Light[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _aposta10Light[key];
    }
  });
});

var _betaddaDark = _interopRequireWildcard(require("./betaddaDark/betaddaDark"));

Object.keys(_betaddaDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _betaddaDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _betaddaDark[key];
    }
  });
});

var _betaddaLight = _interopRequireWildcard(require("./betaddaLight/betaddaLight"));

Object.keys(_betaddaLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _betaddaLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _betaddaLight[key];
    }
  });
});

var _bitcasinoDark = _interopRequireWildcard(require("./bitcasinoDark/bitcasinoDark"));

Object.keys(_bitcasinoDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _bitcasinoDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _bitcasinoDark[key];
    }
  });
});

var _bitcasinoLight = _interopRequireWildcard(require("./bitcasinoLight/bitcasinoLight"));

Object.keys(_bitcasinoLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _bitcasinoLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _bitcasinoLight[key];
    }
  });
});

var _bombayClub = _interopRequireWildcard(require("./bombayClub/bombayClub"));

Object.keys(_bombayClub).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _bombayClub[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _bombayClub[key];
    }
  });
});

var _btcxe = _interopRequireWildcard(require("./btcxe/btcxe"));

Object.keys(_btcxe).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _btcxe[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _btcxe[key];
    }
  });
});

var _commsDark = _interopRequireWildcard(require("./commsDark/commsDark"));

Object.keys(_commsDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _commsDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _commsDark[key];
    }
  });
});

var _commsLight = _interopRequireWildcard(require("./commsLight/commsLight"));

Object.keys(_commsLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _commsLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _commsLight[key];
    }
  });
});

var _empireDark = _interopRequireWildcard(require("./empireDark/empireDark"));

Object.keys(_empireDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _empireDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _empireDark[key];
    }
  });
});

var _empireLight = _interopRequireWildcard(require("./empireLight/empireLight"));

Object.keys(_empireLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _empireLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _empireLight[key];
    }
  });
});

var _hub = _interopRequireWildcard(require("./hub88/hub88"));

Object.keys(_hub).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _hub[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _hub[key];
    }
  });
});

var _livecasinoDark = _interopRequireWildcard(require("./livecasinoDark/livecasinoDark"));

Object.keys(_livecasinoDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _livecasinoDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _livecasinoDark[key];
    }
  });
});

var _livecasinoLight = _interopRequireWildcard(require("./livecasinoLight/livecasinoLight"));

Object.keys(_livecasinoLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _livecasinoLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _livecasinoLight[key];
    }
  });
});

var _luckyslots = _interopRequireWildcard(require("./luckyslots/luckyslots"));

Object.keys(_luckyslots).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _luckyslots[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _luckyslots[key];
    }
  });
});

var _luckyslotsDark = _interopRequireWildcard(require("./luckyslotsDark/luckyslotsDark"));

Object.keys(_luckyslotsDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _luckyslotsDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _luckyslotsDark[key];
    }
  });
});

var _luckyslotsLight = _interopRequireWildcard(require("./luckyslotsLight/luckyslotsLight"));

Object.keys(_luckyslotsLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _luckyslotsLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _luckyslotsLight[key];
    }
  });
});

var _missionsToolDark = _interopRequireWildcard(require("./missionsToolDark/missionsToolDark"));

Object.keys(_missionsToolDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _missionsToolDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _missionsToolDark[key];
    }
  });
});

var _missionsToolLight = _interopRequireWildcard(require("./missionsToolLight/missionsToolLight"));

Object.keys(_missionsToolLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _missionsToolLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _missionsToolLight[key];
    }
  });
});

var _moneyballDark = _interopRequireWildcard(require("./moneyball/moneyballDark"));

Object.keys(_moneyballDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _moneyballDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _moneyballDark[key];
    }
  });
});

var _moneyballLight = _interopRequireWildcard(require("./moneyball/moneyballLight"));

Object.keys(_moneyballLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _moneyballLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _moneyballLight[key];
    }
  });
});

var _moonDesignDark = _interopRequireWildcard(require("./moonDesignDark/moonDesignDark"));

Object.keys(_moonDesignDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _moonDesignDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _moonDesignDark[key];
    }
  });
});

var _moonDesignLight = _interopRequireWildcard(require("./moonDesignLight/moonDesignLight"));

Object.keys(_moonDesignLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _moonDesignLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _moonDesignLight[key];
    }
  });
});

var _shangrilaDark = _interopRequireWildcard(require("./shangrilaDark/shangrilaDark"));

Object.keys(_shangrilaDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _shangrilaDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _shangrilaDark[key];
    }
  });
});

var _shangrilaLight = _interopRequireWildcard(require("./shangrilaLight/shangrilaLight"));

Object.keys(_shangrilaLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _shangrilaLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _shangrilaLight[key];
    }
  });
});

var _avertaStd = _interopRequireWildcard(require("./sharedTokens/avertaStd"));

Object.keys(_avertaStd).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _avertaStd[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _avertaStd[key];
    }
  });
});

var _sharedTokens = _interopRequireWildcard(require("./sharedTokens/sharedTokens"));

Object.keys(_sharedTokens).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _sharedTokens[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _sharedTokens[key];
    }
  });
});

var _slotsDark = _interopRequireWildcard(require("./slotsDark/slotsDark"));

Object.keys(_slotsDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _slotsDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _slotsDark[key];
    }
  });
});

var _slotsLight = _interopRequireWildcard(require("./slotsLight/slotsLight"));

Object.keys(_slotsLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _slotsLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _slotsLight[key];
    }
  });
});

var _sportsbetDark = _interopRequireWildcard(require("./sportsbetDark/sportsbetDark"));

Object.keys(_sportsbetDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _sportsbetDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _sportsbetDark[key];
    }
  });
});

var _sportsbetLight = _interopRequireWildcard(require("./sportsbetLight/sportsbetLight"));

Object.keys(_sportsbetLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _sportsbetLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _sportsbetLight[key];
    }
  });
});

var _supportColors = _interopRequireWildcard(require("./supportColors/supportColors"));

Object.keys(_supportColors).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _supportColors[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _supportColors[key];
    }
  });
});

var _tajCasinoDark = _interopRequireWildcard(require("./tajCasinoDark/tajCasinoDark"));

Object.keys(_tajCasinoDark).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _tajCasinoDark[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _tajCasinoDark[key];
    }
  });
});

var _tajCasinoLight = _interopRequireWildcard(require("./tajCasinoLight/tajCasinoLight"));

Object.keys(_tajCasinoLight).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _tajCasinoLight[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _tajCasinoLight[key];
    }
  });
});

var _ThemeContext = _interopRequireWildcard(require("./themeContext/ThemeContext"));

Object.keys(_ThemeContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ThemeContext[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _ThemeContext[key];
    }
  });
});

var _ThemeProvider = _interopRequireWildcard(require("./themeProvider/ThemeProvider"));

Object.keys(_ThemeProvider).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ThemeProvider[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _ThemeProvider[key];
    }
  });
});

var _ThemeTransition = _interopRequireWildcard(require("./themeTransition/ThemeTransition"));

Object.keys(_ThemeTransition).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ThemeTransition[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _ThemeTransition[key];
    }
  });
});

var _useTheme = _interopRequireWildcard(require("./useTheme/useTheme"));

Object.keys(_useTheme).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _useTheme[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _useTheme[key];
    }
  });
});

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
