// @heathmont/moon-components/lib/es/cardBalance/CardBalance.js
import React from 'react';
import styled from 'styled-components';
import { rem } from '@heathmont/moon-utils';
import Stack from '../stack/Stack';

/**
 * Styles
 */
var Unit = styled.abbr.withConfig({
  displayName: "CardBalance__Unit",
  componentId: "sc-1e6lf9t-0"
})({
  cursor: 'help',
  textDecoration: 'none',
  fontSize: rem(16),
  lineHeight: rem(24)
});
var Title = styled.p.withConfig({
  displayName: "CardBalance__Title",
  componentId: "sc-1e6lf9t-1"
})({
  margin: 0,
  fontSize: rem(32),
  lineHeight: rem(40)
});
var Text = styled.p.withConfig({
  displayName: "CardBalance__Text",
  componentId: "sc-1e6lf9t-2"
})(function (_ref) {
  var color = _ref.theme.color;
  return {
    margin: 0,
    color: color.trunks[100],
    fontSize: rem(16),
    lineHeight: rem(24)
  };
});

var Balance = function Balance(_ref2) {
  var from = _ref2.from;
  return /*#__PURE__*/React.createElement(Stack, {
    as: "section",
    space: 0
  }, /*#__PURE__*/React.createElement(Text, null, from.currency), /*#__PURE__*/React.createElement(Title, null, "".concat(from.value, " "), /*#__PURE__*/React.createElement(Unit, {
    title: from.unitTitle || from.currency
  }, from.unit)));
};
/**
 * Component
 */


var CardBalance = function CardBalance(_ref3) {
  var from = _ref3.from,
      badge = _ref3.badge;
  return badge ? /*#__PURE__*/React.createElement(Stack, null, badge && /*#__PURE__*/React.createElement("span", null, badge), /*#__PURE__*/React.createElement(Balance, {
    from: from
  })) : /*#__PURE__*/React.createElement(Balance, {
    from: from
  });
};

export default CardBalance;
