// @heathmont/moon-themes/lib/es/tajCasinoDark/tajCasinoDark.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var color = {
  piccolo: {
    120: '#d78b2d',
    100: '#d49649',
    80: '#d3a164'
  },
  hit: {
    120: '#050c1f',
    100: '#0a1533',
    80: '#111f45'
  },
  beerus: {
    100: '#26304a'
  },
  goku: {
    100: '#0a1533',
    80: '#111f45',
    40: '#233567',
    10: '#3b4e81'
  },
  gohan: {
    100: '#131e3f',
    80: '#1b2850',
    40: '#2f3f6f',
    10: '#47588a'
  },
  bulma: {
    100: '#ffffff'
  },
  goten: {
    100: '#ffffff'
  },
  trunks: {
    100: '#939aae'
  }
};

var tajCasinoDark = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'Tajcasino',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  radius: {
    small: 4,
    "default": 8,
    largest: 8
  },
  colorScheme: 'dark'
});

export default tajCasinoDark;
