// @heathmont/moon-core/lib/es/loader/Loader.js
import React from 'react';
import styled, { css, keyframes } from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';

/**
 * Animation
 */
var ring = keyframes(["0%{transform:rotate(0deg);}100%{transform:rotate(360deg);}"]);
var animation = css(["animation:", " 1.2s cubic-bezier(0.5,0,0.5,1) infinite;"], ring);
/**
 * Sub-components
 */

var LoaderContainer = styled.strong.withConfig({
  displayName: "Loader__LoaderContainer",
  componentId: "sc-13o5dnc-0"
})(function (_ref) {
  var space = _ref.theme.space;
  return {
    display: 'inline-block',
    position: 'relative',
    width: rem(space["default"]),
    height: rem(space["default"])
  };
});
var LoaderRing = styled.span.withConfig({
  displayName: "Loader__LoaderRing",
  componentId: "sc-13o5dnc-1"
})(function (_ref2) {
  var _ref2$theme = _ref2.theme,
      color = _ref2$theme.color,
      borderStyle = _ref2$theme.borderStyle,
      borderWidth = _ref2$theme.borderWidth;
  return {
    '--loader-color': color.bulma[100],
    display: 'block',
    position: 'absolute',
    width: '100%',
    height: '100%',
    borderRadius: '50%',
    boxSizing: 'border-box',
    borderWidth: borderWidth * 2,
    borderStyle: borderStyle,
    borderColor: 'transparent',
    borderTopColor: 'var(--loader-color)'
  };
}, animation, function (_ref3) {
  var animationDelay = _ref3.animationDelay;
  return {
    animationDelay: animationDelay
  };
}, function (_ref4) {
  var color = _ref4.color,
      theme = _ref4.theme;
  return {
    '--loader-color': themed('color', color)(theme)
  };
});
/**
 * Component
 */

var Loader = function Loader(_ref5) {
  var color = _ref5.color;
  return /*#__PURE__*/React.createElement(LoaderContainer, {
    role: "progressbar"
  }, ['-0.45s', '-0.3s', '-0.15s', '0s'].map(function (delay) {
    return /*#__PURE__*/React.createElement(LoaderRing, {
      key: "loader-".concat(delay),
      color: color,
      animationDelay: delay
    });
  }));
};

export default Loader;
