// @heathmont/moon-themes/lib/es/shangrilaLight/shangrilaLight.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var color = {
  piccolo: {
    120: '#977135',
    100: '#a57f40',
    80: '#b38b4d'
  },
  hit: {
    120: '#bbb9b9',
    100: '#c2c2c2',
    80: '#c9c9c9'
  },
  beerus: {
    100: '#83837c'
  },
  goku: {
    100: '#f2eee8',
    80: '#ede7de',
    40: '#e5d9c7',
    10: '#e0ccae'
  },
  gohan: {
    100: '#f6f6f4',
    80: '#efefeb',
    40: '#e2e2da',
    10: '#d5d5c8'
  },
  goten: {
    100: '#ffffff'
  },
  bulma: {
    100: '#33332e'
  },
  trunks: {
    100: '#83837c'
  }
};

var shangrilaLight = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'ShangriLa.io',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  colorScheme: 'light'
});

export default shangrilaLight;
