// @heathmont/moon-core/lib/screenLoader/ScreenLoader.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _moonThemes = require("@heathmont/moon-themes");

var _moonUtils = require("@heathmont/moon-utils");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * Animations
 */
var rotate = (0, _styledComponents.keyframes)(["0%{transform:rotate(0deg) scale(0.8);}50%{transform:rotate(360deg) scale(1.2);}100%{transform:rotate(720deg) scale(0.8);}"]);
var rotateAnimation = (0, _styledComponents.css)(["animation:", " 1s infinite;"], rotate);
var ball1 = (0, _styledComponents.keyframes)(["0%{box-shadow:", " 0 0 var(--third-color);}50%{box-shadow:0 0 0 var(--third-color);margin-bottom:0;transform:translate(", ",", ");}100%{box-shadow:", " 0 0 var(--third-color);margin-bottom:", ";}"], (0, _moonUtils.rem)(30), (0, _moonUtils.rem)(15), (0, _moonUtils.rem)(15), (0, _moonUtils.rem)(30), (0, _moonUtils.rem)(10));
var ball1Animation = (0, _styledComponents.css)(["animation:", " 1s infinite;"], ball1);
var ball2 = (0, _styledComponents.keyframes)(["0%{box-shadow:", " 0 0 var(--fourth-color);}50%{box-shadow:0 0 0 var(--fourth-color);margin-top:-", ";transform:translate(", ",", ");}100%{box-shadow:", " 0 0 var(--fourth-color);margin-top:0;}"], (0, _moonUtils.rem)(30), (0, _moonUtils.rem)(20), (0, _moonUtils.rem)(15), (0, _moonUtils.rem)(15), (0, _moonUtils.rem)(30));
var ball2Animation = (0, _styledComponents.css)(["animation:", " 1s infinite;"], ball2);
/**
 * Sub-component
 */

var LoaderContainer = _styledComponents["default"].div.withConfig({
  displayName: "ScreenLoader__LoaderContainer",
  componentId: "sc-1kqs93o-0"
})(["--first-color:", ";--second-color:", ";--third-color:", ";--fourth-color:", ";width:", ";height:", ";&:before{border-radius:50%;content:'';display:block;height:", ";width:", ";background-color:var(--first-color);box-shadow:", " 0 0 var(--first-color);margin-bottom:", ";", "}&:after{border-radius:50%;content:'';display:block;height:", ";width:", ";background-color:var(--second-color);box-shadow:", " 0 0 var(--second-color);", "}", ""], function (_ref) {
  var theme = _ref.theme,
      firstColor = _ref.firstColor;
  return firstColor || theme.color.chiChi[100];
}, function (_ref2) {
  var theme = _ref2.theme,
      secondColor = _ref2.secondColor;
  return secondColor || theme.color.piccolo[100];
}, function (_ref3) {
  var theme = _ref3.theme,
      thirdColor = _ref3.thirdColor;
  return thirdColor || theme.color.cell[100];
}, function (_ref4) {
  var theme = _ref4.theme,
      fourthColor = _ref4.fourthColor;
  return fourthColor || theme.color.frieza[100];
}, (0, _moonUtils.rem)(50), (0, _moonUtils.rem)(50), (0, _moonUtils.rem)(20), (0, _moonUtils.rem)(20), (0, _moonUtils.rem)(30), (0, _moonUtils.rem)(10), ball1Animation, (0, _moonUtils.rem)(20), (0, _moonUtils.rem)(20), (0, _moonUtils.rem)(30), ball2Animation, rotateAnimation);

var ScreenLoader = function ScreenLoader(_ref5) {
  var firstColor = _ref5.firstColor,
      secondColor = _ref5.secondColor,
      thirdColor = _ref5.thirdColor,
      fourthColor = _ref5.fourthColor;
  var theme = (0, _moonThemes.useTheme)();
  return /*#__PURE__*/_react["default"].createElement(LoaderContainer, {
    firstColor: (0, _moonUtils.themed)('color', firstColor)(theme),
    secondColor: (0, _moonUtils.themed)('color', secondColor)(theme),
    thirdColor: (0, _moonUtils.themed)('color', thirdColor)(theme),
    fourthColor: (0, _moonUtils.themed)('color', fourthColor)(theme),
    role: "progressbar"
  });
};

var _default = ScreenLoader;
exports["default"] = _default;
