// @heathmont/moon-themes/lib/es/tajCasinoLight/tajCasinoLight.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var color = {
  piccolo: {
    120: '#d78b2d',
    100: '#d49649',
    80: '#d3a164'
  },
  hit: {
    120: '#f7f7f7',
    100: '#ffffff',
    80: '#ffffff'
  },
  beerus: {
    100: '#f0f0f0'
  },
  goku: {
    100: '#ffffff',
    80: '#f7f7f7',
    40: '#e8e8e8',
    10: '#d9d9d9'
  },
  gohan: {
    100: '#f3f3f2',
    80: '#e7e7e4',
    40: '#d0d0c8',
    10: '#b9b9ac'
  },
  goten: {
    100: '#ffffff'
  },
  bulma: {
    100: '#000000'
  },
  trunks: {
    100: '#939aae'
  }
};

var tajCasinoLight = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'Tajcasino',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  radius: {
    small: 4,
    "default": 8,
    largest: 8
  },
  colorScheme: 'light'
});

export default tajCasinoLight;
