// @heathmont/moon-components/lib/datepicker/private/Day.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Day = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _format = _interopRequireDefault(require("date-fns/format"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var DayStyled = _styledComponents["default"].div.withConfig({
  displayName: "Day__DayStyled",
  componentId: "sc-173weuh-0"
})(function (_ref) {
  var theme = _ref.theme,
      isInRange = _ref.isInRange,
      isInRangePreview = _ref.isInRangePreview,
      isStartEdge = _ref.isStartEdge,
      isEndEdge = _ref.isEndEdge,
      isToday = _ref.isToday;
  return [{
    height: (0, _moonUtils.rem)(32),
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 500,
    fontSize: (0, _moonUtils.rem)(10),
    color: theme.color.bulma[100],
    userSelect: 'none',
    cursor: 'pointer',
    boxSizing: 'border-box',
    border: '1px solid transparent',
    // to prevent jumping on hover
    '&:hover': {
      border: "1px solid ".concat(theme.color.trunks[100]),
      borderRadius: (0, _moonUtils.rem)(theme.radius.largest)
    }
  }, (isInRange || isInRangePreview) && {
    backgroundColor: theme.color.goku[80]
  }, (isStartEdge || isEndEdge) && {
    background: theme.color.piccolo[100],
    borderRadius: (0, _moonUtils.rem)(theme.radius.largest),
    color: theme.color.goten[100]
  }, isToday && {
    fontWeight: 'bold'
  }];
});

var DayWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Day__DayWrapper",
  componentId: "sc-173weuh-1"
})(function (_ref2) {
  var theme = _ref2.theme,
      isStartEdge = _ref2.isStartEdge,
      isEndEdge = _ref2.isEndEdge,
      isInRangePreview = _ref2.isInRangePreview;
  return [isStartEdge && {
    backgroundColor: theme.color.goku[80],
    borderTopLeftRadius: (0, _moonUtils.rem)(theme.radius.largest),
    borderBottomLeftRadius: (0, _moonUtils.rem)(theme.radius.largest)
  }, isEndEdge && {
    backgroundColor: theme.color.goku[80],
    borderTopRightRadius: (0, _moonUtils.rem)(theme.radius.largest),
    borderBottomRightRadius: (0, _moonUtils.rem)(theme.radius.largest)
  }, isInRangePreview && {
    backgroundColor: theme.color.goku[80],
    '&:hover': {
      borderTopRightRadius: (0, _moonUtils.rem)(theme.radius.largest),
      borderBottomRightRadius: (0, _moonUtils.rem)(theme.radius.largest)
    }
  }];
});

var EmptyCell = _styledComponents["default"].div.withConfig({
  displayName: "Day__EmptyCell",
  componentId: "sc-173weuh-2"
})({
  height: (0, _moonUtils.rem)(32)
});

var formatLabel = function formatLabel(date) {
  return (0, _format["default"])(date, 'd');
};

var Day = function Day(_ref3) {
  var date = _ref3.date,
      isSameMonth = _ref3.isSameMonth,
      isStartEdge = _ref3.isStartEdge,
      isEndEdge = _ref3.isEndEdge,
      isInRange = _ref3.isInRange,
      isInRangePreview = _ref3.isInRangePreview,
      onDayClick = _ref3.onDayClick,
      _onMouseEnter = _ref3.onMouseEnter,
      isToday = _ref3.isToday;

  if (!isSameMonth) {
    return /*#__PURE__*/_react["default"].createElement(EmptyCell, null);
  }

  return /*#__PURE__*/_react["default"].createElement(DayWrapper, {
    isStartEdge: isStartEdge,
    isEndEdge: isEndEdge,
    isInRangePreview: isInRangePreview,
    isInRange: isInRange
  }, /*#__PURE__*/_react["default"].createElement(DayStyled, {
    isStartEdge: isStartEdge,
    isEndEdge: isEndEdge,
    isInRange: isInRange,
    isInRangePreview: isInRangePreview,
    isToday: isToday,
    onClick: function onClick() {
      return onDayClick(date);
    },
    onMouseEnter: function onMouseEnter() {
      return _onMouseEnter(date);
    }
  }, formatLabel(date)));
};

exports.Day = Day;
