// @heathmont/moon-components/lib/datepicker/private/Picker.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Picker = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _addMonths = _interopRequireDefault(require("date-fns/addMonths"));

var _moonUtils = require("@heathmont/moon-utils");

var _IconChevronLeft = _interopRequireDefault(require("../../private/icons/IconChevronLeft"));

var _IconChevronRight = _interopRequireDefault(require("../../private/icons/IconChevronRight"));

var _Month = require("./Month");

var _Sidebar = require("./Sidebar");

var _InputsPanel = require("./InputsPanel");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var PickerContainer = _styledComponents["default"].div.withConfig({
  displayName: "Picker__PickerContainer",
  componentId: "sc-1yiyb5l-0"
})(function (_ref) {
  var _ref2;

  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      space = _ref$theme.space,
      radius = _ref$theme.radius,
      breakpoint = _ref$theme.breakpoint,
      zIndex = _ref$theme.zIndex;
  return _ref2 = {
    width: 'fit-content',
    position: 'absolute',
    zIndex: zIndex.dialog,
    display: 'grid',
    gridTemplateAreas: "\n      \"sidebar\"\n      \"firstMonth\"\n      \"secondMonth\"\n      \"inputs\"\n    ",
    alignItems: 'center'
  }, _defineProperty(_ref2, (0, _moonUtils.mq)(breakpoint.medium), {
    gridTemplateAreas: "\n        \"sidebar firstMonth secondMonth\"\n        \"sidebar  inputs     inputs\"\n      ",
    gridTemplateColumns: '2fr 3fr 3fr',
    alignItems: 'baseline'
  }), _defineProperty(_ref2, "paddingBottom", (0, _moonUtils.rem)(space["default"])), _defineProperty(_ref2, "gap", (0, _moonUtils.rem)(space["default"])), _defineProperty(_ref2, "borderRadius", (0, _moonUtils.rem)(radius["default"])), _defineProperty(_ref2, "padding", space.small), _defineProperty(_ref2, "backgroundColor", color.goku[100]), _defineProperty(_ref2, "boxShadow", "4px 4px 12px ".concat(color.hit[80], ",\n      0px 10px 18px ").concat(color.hit[80])), _defineProperty(_ref2, "overflow", 'hidden'), _ref2;
});

var FirstMonth = _styledComponents["default"].div.withConfig({
  displayName: "Picker__FirstMonth",
  componentId: "sc-1yiyb5l-1"
})({
  gridArea: 'firstMonth',
  maxWidth: (0, _moonUtils.rem)(236),
  position: 'relative'
});

var SecondMonth = _styledComponents["default"].div.withConfig({
  displayName: "Picker__SecondMonth",
  componentId: "sc-1yiyb5l-2"
})({
  gridArea: 'secondMonth',
  maxWidth: (0, _moonUtils.rem)(236),
  position: 'relative'
});

var LeftArrow = (0, _styledComponents["default"])(_IconChevronLeft["default"]).withConfig({
  displayName: "Picker__LeftArrow",
  componentId: "sc-1yiyb5l-3"
})(function (_ref3) {
  var theme = _ref3.theme;
  return {
    cursor: 'pointer',
    color: theme.color.bulma[100],
    position: 'absolute',
    left: (0, _moonUtils.rem)(16),
    fontSize: (0, _moonUtils.rem)(16),
    top: (0, _moonUtils.rem)(2)
  };
});
var RightArrow = (0, _styledComponents["default"])(_IconChevronRight["default"]).withConfig({
  displayName: "Picker__RightArrow",
  componentId: "sc-1yiyb5l-4"
})(function (_ref4) {
  var theme = _ref4.theme;
  return {
    cursor: 'pointer',
    color: theme.color.bulma[100],
    position: 'absolute',
    right: (0, _moonUtils.rem)(16),
    fontSize: (0, _moonUtils.rem)(16),
    top: (0, _moonUtils.rem)(2)
  };
});

var Picker = function Picker(_ref5) {
  var datesState = _ref5.datesState,
      labels = _ref5.labels,
      translations = _ref5.translations,
      config = _ref5.config,
      onDayClick = _ref5.onDayClick,
      onDayMouseEnter = _ref5.onDayMouseEnter,
      firstMonth = _ref5.firstMonth,
      secondMonth = _ref5.secondMonth,
      selectRange = _ref5.selectRange,
      nextMonth = _ref5.nextMonth,
      prevMonth = _ref5.prevMonth,
      setStartDate = _ref5.setStartDate,
      setEndDate = _ref5.setEndDate,
      onDateChange = _ref5.onDateChange,
      apply = _ref5.apply,
      reset = _ref5.reset;
  return /*#__PURE__*/_react["default"].createElement(PickerContainer, null, /*#__PURE__*/_react["default"].createElement(_Sidebar.Sidebar, {
    range: datesState.range,
    selectRange: selectRange,
    translations: translations,
    config: config
  }), /*#__PURE__*/_react["default"].createElement(FirstMonth, null, /*#__PURE__*/_react["default"].createElement(LeftArrow, {
    onClick: prevMonth
  }), /*#__PURE__*/_react["default"].createElement(_Month.Month, {
    key: "first",
    monthDays: firstMonth,
    monthLabel: labels.firstMonthLabel,
    year: labels.firstMonthYearLabel,
    weekDayLabels: labels.weekDayLabels,
    onDayClick: onDayClick,
    onMouseEnter: onDayMouseEnter,
    cursorDate: datesState.cursorDate,
    startDate: datesState.startDate,
    endDate: datesState.endDate,
    hoveredDate: datesState.hoveredDate
  })), /*#__PURE__*/_react["default"].createElement(SecondMonth, null, /*#__PURE__*/_react["default"].createElement(RightArrow, {
    onClick: nextMonth
  }), /*#__PURE__*/_react["default"].createElement(_Month.Month, {
    key: "second",
    monthDays: secondMonth,
    monthLabel: labels.secondMonthLabel,
    year: labels.secondMonthYearLabel,
    weekDayLabels: labels.weekDayLabels,
    onDayClick: onDayClick,
    onMouseEnter: onDayMouseEnter,
    cursorDate: (0, _addMonths["default"])(datesState.cursorDate, 1),
    startDate: datesState.startDate,
    endDate: datesState.endDate,
    hoveredDate: datesState.hoveredDate
  })), /*#__PURE__*/_react["default"].createElement(_InputsPanel.InputsPanel, {
    reset: reset,
    startDate: datesState.startDate,
    endDate: datesState.endDate,
    setStartDate: setStartDate,
    setEndDate: setEndDate,
    onDateChange: onDateChange,
    translations: translations,
    apply: apply,
    config: config
  }));
};

exports.Picker = Picker;
