// @heathmont/moon-components/lib/datepicker/private/Sidebar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sidebar = void 0;

var _react = _interopRequireDefault(require("react"));

var _rem = _interopRequireDefault(require("polished/lib/helpers/rem"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _getRanges = require("./helpers/getRanges");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SidebarList = _styledComponents["default"].ul.withConfig({
  displayName: "Sidebar__SidebarList",
  componentId: "omhqqr-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      breakpoint = _ref$theme.breakpoint;
  return _defineProperty({
    gridArea: 'sidebar',
    background: color.goku[100],
    padding: 0,
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap'
  }, (0, _moonUtils.mq)(breakpoint.medium), {
    flexDirection: 'column',
    maxWidth: '100%' // maxWidth: rem(120),

  });
});

var RangeItem = _styledComponents["default"].li.withConfig({
  displayName: "Sidebar__RangeItem",
  componentId: "omhqqr-1"
})(function (_ref3) {
  var _ref4;

  var theme = _ref3.theme,
      isSelected = _ref3.isSelected;
  return _ref4 = {
    position: 'relative',
    listStyle: 'none',
    padding: (0, _rem["default"])(8),
    boxSizing: 'border-box',
    cursor: 'pointer',
    fontSize: (0, _rem["default"])(14),
    lineHeight: (0, _rem["default"])(20),
    background: isSelected ? theme.color.goku[80] : theme.color.goku[100],
    color: theme.color.bulma[100]
  }, _defineProperty(_ref4, (0, _moonUtils.mq)(theme.breakpoint.small), {
    paddingLeft: (0, _rem["default"])(16)
  }), _defineProperty(_ref4, '&:hover', {
    background: theme.color.goku[80]
  }), _defineProperty(_ref4, '&:hover:before', {
    content: "''",
    display: 'block',
    position: 'absolute',
    zIndex: 1,
    left: 0,
    top: 0,
    bottom: 0,
    width: (0, _rem["default"])(4),
    height: '100%',
    backgroundColor: theme.color.bulma[100],
    borderRadius: (0, _rem["default"])(4)
  }), _defineProperty(_ref4, '&:before', isSelected ? {
    content: "''",
    display: 'block',
    position: 'absolute',
    zIndex: 1,
    left: 0,
    top: 0,
    bottom: 0,
    width: (0, _rem["default"])(4),
    height: '100%',
    backgroundColor: theme.color.bulma[100],
    borderRadius: (0, _rem["default"])(4)
  } : ''), _ref4;
});

var Sidebar = function Sidebar(_ref5) {
  var currentRange = _ref5.range,
      selectRange = _ref5.selectRange,
      translations = _ref5.translations,
      config = _ref5.config;
  var ranges = (0, _getRanges.getRanges)(config);
  return /*#__PURE__*/_react["default"].createElement(SidebarList, null, ranges.map(function (range) {
    return /*#__PURE__*/_react["default"].createElement(RangeItem, {
      key: range,
      isSelected: range === currentRange,
      onClick: function onClick() {
        return selectRange(range);
      }
    }, translations[range]);
  }));
};

exports.Sidebar = Sidebar;
