// @heathmont/moon-components/lib/avatar/Avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _react = _interopRequireDefault(require("react"));

var _moonAssets = require("@heathmont/moon-assets");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var AvatarWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Avatar__AvatarWrapper",
  componentId: "sc-1silcs5-0"
})(function (_ref) {
  var size = _ref.size,
      imageUrl = _ref.imageUrl,
      color = _ref.color,
      backgroundColor = _ref.backgroundColor,
      theme = _ref.theme;
  return [{
    color: (0, _moonUtils.themed)('color', color)(theme),
    backgroundColor: (0, _moonUtils.themed)('color', backgroundColor)(theme),
    borderRadius: '50%',
    overflow: 'hidden',
    textTransform: 'uppercase',
    fontWeight: theme.fontWeight.semibold,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundImage: "url(".concat(imageUrl, ")"),
    backgroundSize: 'cover'
  }, size === 'xsmall' && {
    height: (0, _moonUtils.rem)(24),
    width: (0, _moonUtils.rem)(24),
    fontSize: (0, _moonUtils.rem)(14)
  }, size === 'small' && {
    height: (0, _moonUtils.rem)(32),
    width: (0, _moonUtils.rem)(32),
    fontSize: (0, _moonUtils.rem)(18)
  }, size === 'medium' && {
    height: (0, _moonUtils.rem)(40),
    width: (0, _moonUtils.rem)(40),
    fontSize: (0, _moonUtils.rem)(20)
  }, size === 'large' && {
    height: (0, _moonUtils.rem)(48),
    width: (0, _moonUtils.rem)(48),
    fontSize: (0, _moonUtils.rem)(24)
  }];
});

var Avatar = function Avatar(props) {
  var name = props.name,
      imageUrl = props.imageUrl,
      color = props.color;
  if (imageUrl) return /*#__PURE__*/_react["default"].createElement(AvatarWrapper, props);
  return /*#__PURE__*/_react["default"].createElement(AvatarWrapper, props, name || /*#__PURE__*/_react["default"].createElement(_moonAssets.IconUser, {
    color: color
  }));
};

Avatar.defaultProps = {
  imageUrl: '',
  size: 'medium'
};
var _default = Avatar;
exports["default"] = _default;
