// @heathmont/moon-themes/lib/es/hub88/hub88.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var color = {
  piccolo: {
    120: '#054cff',
    100: '#1a5bff',
    80: '#386ef5'
  },
  hit: {
    120: '#a4afb6',
    100: '#abb7bf',
    80: '#b5bec5'
  },
  goten: {
    100: '#ffffff'
  },
  goku: {
    100: '#f3f5f6',
    80: '#eaeef0',
    40: '#d8e0e3',
    10: '#c4d2d9'
  },
  gohan: {
    100: '#ffffff',
    80: '#f6f7f8',
    40: '#e4e8ec',
    10: '#d1d9e0'
  },
  beerus: {
    100: '#e2e6e9'
  },
  bulma: {
    100: '#010913'
  },
  trunks: {
    100: '#8697a2'
  }
};

var hub88 = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'Hub88',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  colorScheme: 'dark',
  space: _objectSpread(_objectSpread({}, sharedTokens.space), {}, {
    medium: sharedTokens.base.space * 1.25,
    large: sharedTokens.base.space * 1.5
  })
});

export default hub88;
