// @heathmont/moon-themes/lib/es/empireDark/empireDark.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var color = {
  piccolo: {
    120: '#e6b400',
    100: '#ffc800',
    80: '#f9ca1f'
  },
  hit: {
    120: '#151414',
    100: '#1c1c1c',
    80: '#242424'
  },
  beerus: {
    100: '#141414'
  },
  goku: {
    100: '#141414',
    80: '#1c1c1c',
    40: '#2b2b2b',
    10: '#3b3b3b'
  },
  gohan: {
    100: '#000000',
    80: '#080808',
    40: '#171717',
    10: '#262626'
  },
  bulma: {
    100: '#ffffff'
  },
  goten: {
    100: '#000000'
  },
  trunks: {
    100: '#828282'
  }
};

var empireDark = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'Empire.io',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  radius: {
    small: 4,
    "default": 8,
    largest: 8
  },
  colorScheme: 'dark'
});

export default empireDark;
