// @heathmont/moon-components/lib/banner/Banner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _moonUtils = require("@heathmont/moon-utils");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _IconBannerInfo = _interopRequireDefault(require("../private/icons/IconBannerInfo"));

var _IconCloseSmall = _interopRequireDefault(require("../private/icons/IconCloseSmall"));

var _IconError = _interopRequireDefault(require("../private/icons/IconError"));

var _IconWarning = _interopRequireDefault(require("../private/icons/IconWarning"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var BannerWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Banner__BannerWrapper",
  componentId: "sc-1uryy7q-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      space = _ref$theme.space;
  return {
    minWidth: (0, _moonUtils.rem)(280),
    width: '100%',
    padding: "".concat(space["default"], "px ").concat((0, _moonUtils.rem)(12)),
    backgroundColor: color.hit[100],
    borderRadius: (0, _moonUtils.rem)(12),
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'left',
    '& > * + *': {
      marginLeft: (0, _moonUtils.rem)(12)
    },
    alignItems: 'center',
    'P,  SPAN': {
      fontSize: (0, _moonUtils.rem)(14),
      lineHeight: (0, _moonUtils.rem)(20),
      margin: 0
    }
  };
});

var IconWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Banner__IconWrapper",
  componentId: "sc-1uryy7q-1"
})({
  marginRight: (0, _moonUtils.rem)(8)
});

var IconCloseWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Banner__IconCloseWrapper",
  componentId: "sc-1uryy7q-2"
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    backgroundColor: theme.color.goku[100],
    borderRadius: (0, _moonUtils.rem)(theme.radius.largest),
    marginLeft: 'auto',
    cursor: 'pointer',
    width: (0, _moonUtils.rem)(24),
    height: (0, _moonUtils.rem)(24),
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  };
});

var Banner = function Banner(_ref3) {
  var message = _ref3.message,
      onClose = _ref3.onClose,
      status = _ref3.status,
      _ref3$isCloseable = _ref3.isCloseable,
      isCloseable = _ref3$isCloseable === void 0 ? false : _ref3$isCloseable;

  var _React$useState = _react["default"].useState(true),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      visible = _React$useState2[0],
      setVisible = _React$useState2[1];

  var closeBanner = function closeBanner() {
    return setVisible(false);
  };

  if (!visible) {
    return null;
  }

  return /*#__PURE__*/_react["default"].createElement(BannerWrapper, null, status === 'error' && /*#__PURE__*/_react["default"].createElement(IconWrapper, null, /*#__PURE__*/_react["default"].createElement(_IconError["default"], {
    fontSize: "1.5rem"
  })), status === 'warning' && /*#__PURE__*/_react["default"].createElement(IconWrapper, null, /*#__PURE__*/_react["default"].createElement(_IconWarning["default"], {
    fontSize: "1.5rem",
    color: "krillin.100"
  })), status === 'info' && /*#__PURE__*/_react["default"].createElement(IconWrapper, null, /*#__PURE__*/_react["default"].createElement(_IconBannerInfo["default"], {
    fontSize: "1.5rem",
    color: "krillin.100"
  })), message && message, isCloseable && /*#__PURE__*/_react["default"].createElement(IconCloseWrapper, {
    onClick: function onClick() {
      if (onClose && typeof onClose === 'function') {
        onClose();
      }

      closeBanner();
    }
  }, /*#__PURE__*/_react["default"].createElement(_IconCloseSmall["default"], {
    fontSize: "1rem",
    color: "trunks.100"
  })));
};

var _default = Banner;
exports["default"] = _default;
