// @heathmont/moon-core/lib/es/screenLoader/ScreenLoader.js
import React from 'react';
import styled, { css, keyframes } from 'styled-components';
import { useTheme } from '@heathmont/moon-themes';
import { rem, themed } from '@heathmont/moon-utils';
/**
 * Animations
 */

var rotate = keyframes(["0%{transform:rotate(0deg) scale(0.8);}50%{transform:rotate(360deg) scale(1.2);}100%{transform:rotate(720deg) scale(0.8);}"]);
var rotateAnimation = css(["animation:", " 1s infinite;"], rotate);
var ball1 = keyframes(["0%{box-shadow:", " 0 0 var(--third-color);}50%{box-shadow:0 0 0 var(--third-color);margin-bottom:0;transform:translate(", ",", ");}100%{box-shadow:", " 0 0 var(--third-color);margin-bottom:", ";}"], rem(30), rem(15), rem(15), rem(30), rem(10));
var ball1Animation = css(["animation:", " 1s infinite;"], ball1);
var ball2 = keyframes(["0%{box-shadow:", " 0 0 var(--fourth-color);}50%{box-shadow:0 0 0 var(--fourth-color);margin-top:-", ";transform:translate(", ",", ");}100%{box-shadow:", " 0 0 var(--fourth-color);margin-top:0;}"], rem(30), rem(20), rem(15), rem(15), rem(30));
var ball2Animation = css(["animation:", " 1s infinite;"], ball2);
/**
 * Sub-component
 */

var LoaderContainer = styled.div.withConfig({
  displayName: "ScreenLoader__LoaderContainer",
  componentId: "sc-1kqs93o-0"
})(["--first-color:", ";--second-color:", ";--third-color:", ";--fourth-color:", ";width:", ";height:", ";&:before{border-radius:50%;content:'';display:block;height:", ";width:", ";background-color:var(--first-color);box-shadow:", " 0 0 var(--first-color);margin-bottom:", ";", "}&:after{border-radius:50%;content:'';display:block;height:", ";width:", ";background-color:var(--second-color);box-shadow:", " 0 0 var(--second-color);", "}", ""], function (_ref) {
  var theme = _ref.theme,
      firstColor = _ref.firstColor;
  return firstColor || theme.color.chiChi[100];
}, function (_ref2) {
  var theme = _ref2.theme,
      secondColor = _ref2.secondColor;
  return secondColor || theme.color.piccolo[100];
}, function (_ref3) {
  var theme = _ref3.theme,
      thirdColor = _ref3.thirdColor;
  return thirdColor || theme.color.cell[100];
}, function (_ref4) {
  var theme = _ref4.theme,
      fourthColor = _ref4.fourthColor;
  return fourthColor || theme.color.frieza[100];
}, rem(50), rem(50), rem(20), rem(20), rem(30), rem(10), ball1Animation, rem(20), rem(20), rem(30), ball2Animation, rotateAnimation);

var ScreenLoader = function ScreenLoader(_ref5) {
  var firstColor = _ref5.firstColor,
      secondColor = _ref5.secondColor,
      thirdColor = _ref5.thirdColor,
      fourthColor = _ref5.fourthColor;
  var theme = useTheme();
  return /*#__PURE__*/React.createElement(LoaderContainer, {
    firstColor: themed('color', firstColor)(theme),
    secondColor: themed('color', secondColor)(theme),
    thirdColor: themed('color', thirdColor)(theme),
    fourthColor: themed('color', fourthColor)(theme),
    role: "progressbar"
  });
};

export default ScreenLoader;
