// @heathmont/moon-assets/currencies/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  IconAustralianDollar: true,
  IconBitcoin: true,
  IconBrazilianReal: true,
  IconCanadianDollar: true,
  IconCardano: true,
  IconChilieanPeso: true,
  IconChineseYuan: true,
  IconDanishKrone: true,
  IconDogecoin: true,
  IconEthereum: true,
  IconEuro: true,
  IconGbPoundSterling: true,
  IconHongKongDollar: true,
  IconIcelandicKrona: true,
  IconIndianRupee: true,
  IconJapaneseYen: true,
  IconLitecoin: true,
  IconNewTaiwanDollar: true,
  IconNewZealandDollar: true,
  IconPolandZloty: true,
  IconRipple: true,
  IconRussianRuble: true,
  IconSingaporeDollar: true,
  IconSoc: true,
  IconSouthKoreanWon: true,
  IconSwedishKrona: true,
  IconSwissFranc: true,
  IconThaiBaht: true,
  IconTron: true,
  IconTurkishLira: true,
  IconUsaDollar: true
};
Object.defineProperty(exports, "IconAustralianDollar", {
  enumerable: true,
  get: function get() {
    return _IconAustralianDollar["default"];
  }
});
Object.defineProperty(exports, "IconBitcoin", {
  enumerable: true,
  get: function get() {
    return _IconBitcoin["default"];
  }
});
Object.defineProperty(exports, "IconBrazilianReal", {
  enumerable: true,
  get: function get() {
    return _IconBrazilianReal["default"];
  }
});
Object.defineProperty(exports, "IconCanadianDollar", {
  enumerable: true,
  get: function get() {
    return _IconCanadianDollar["default"];
  }
});
Object.defineProperty(exports, "IconCardano", {
  enumerable: true,
  get: function get() {
    return _IconCardano["default"];
  }
});
Object.defineProperty(exports, "IconChilieanPeso", {
  enumerable: true,
  get: function get() {
    return _IconChilieanPeso["default"];
  }
});
Object.defineProperty(exports, "IconChineseYuan", {
  enumerable: true,
  get: function get() {
    return _IconChineseYuan["default"];
  }
});
Object.defineProperty(exports, "IconDanishKrone", {
  enumerable: true,
  get: function get() {
    return _IconDanishKrone["default"];
  }
});
Object.defineProperty(exports, "IconDogecoin", {
  enumerable: true,
  get: function get() {
    return _IconDogecoin["default"];
  }
});
Object.defineProperty(exports, "IconEthereum", {
  enumerable: true,
  get: function get() {
    return _IconEthereum["default"];
  }
});
Object.defineProperty(exports, "IconEuro", {
  enumerable: true,
  get: function get() {
    return _IconEuro["default"];
  }
});
Object.defineProperty(exports, "IconGbPoundSterling", {
  enumerable: true,
  get: function get() {
    return _IconGbPoundSterling["default"];
  }
});
Object.defineProperty(exports, "IconHongKongDollar", {
  enumerable: true,
  get: function get() {
    return _IconHongKongDollar["default"];
  }
});
Object.defineProperty(exports, "IconIcelandicKrona", {
  enumerable: true,
  get: function get() {
    return _IconIcelandicKrona["default"];
  }
});
Object.defineProperty(exports, "IconIndianRupee", {
  enumerable: true,
  get: function get() {
    return _IconIndianRupee["default"];
  }
});
Object.defineProperty(exports, "IconJapaneseYen", {
  enumerable: true,
  get: function get() {
    return _IconJapaneseYen["default"];
  }
});
Object.defineProperty(exports, "IconLitecoin", {
  enumerable: true,
  get: function get() {
    return _IconLitecoin["default"];
  }
});
Object.defineProperty(exports, "IconNewTaiwanDollar", {
  enumerable: true,
  get: function get() {
    return _IconNewTaiwanDollar["default"];
  }
});
Object.defineProperty(exports, "IconNewZealandDollar", {
  enumerable: true,
  get: function get() {
    return _IconNewZealandDollar["default"];
  }
});
Object.defineProperty(exports, "IconPolandZloty", {
  enumerable: true,
  get: function get() {
    return _IconPolandZloty["default"];
  }
});
Object.defineProperty(exports, "IconRipple", {
  enumerable: true,
  get: function get() {
    return _IconRipple["default"];
  }
});
Object.defineProperty(exports, "IconRussianRuble", {
  enumerable: true,
  get: function get() {
    return _IconRussianRuble["default"];
  }
});
Object.defineProperty(exports, "IconSingaporeDollar", {
  enumerable: true,
  get: function get() {
    return _IconSingaporeDollar["default"];
  }
});
Object.defineProperty(exports, "IconSoc", {
  enumerable: true,
  get: function get() {
    return _IconSoc["default"];
  }
});
Object.defineProperty(exports, "IconSouthKoreanWon", {
  enumerable: true,
  get: function get() {
    return _IconSouthKoreanWon["default"];
  }
});
Object.defineProperty(exports, "IconSwedishKrona", {
  enumerable: true,
  get: function get() {
    return _IconSwedishKrona["default"];
  }
});
Object.defineProperty(exports, "IconSwissFranc", {
  enumerable: true,
  get: function get() {
    return _IconSwissFranc["default"];
  }
});
Object.defineProperty(exports, "IconThaiBaht", {
  enumerable: true,
  get: function get() {
    return _IconThaiBaht["default"];
  }
});
Object.defineProperty(exports, "IconTron", {
  enumerable: true,
  get: function get() {
    return _IconTron["default"];
  }
});
Object.defineProperty(exports, "IconTurkishLira", {
  enumerable: true,
  get: function get() {
    return _IconTurkishLira["default"];
  }
});
Object.defineProperty(exports, "IconUsaDollar", {
  enumerable: true,
  get: function get() {
    return _IconUsaDollar["default"];
  }
});

var _IconAustralianDollar = _interopRequireWildcard(require("./IconAustralianDollar"));

Object.keys(_IconAustralianDollar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconAustralianDollar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconAustralianDollar[key];
    }
  });
});

var _IconBitcoin = _interopRequireWildcard(require("./IconBitcoin"));

Object.keys(_IconBitcoin).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconBitcoin[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconBitcoin[key];
    }
  });
});

var _IconBrazilianReal = _interopRequireWildcard(require("./IconBrazilianReal"));

Object.keys(_IconBrazilianReal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconBrazilianReal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconBrazilianReal[key];
    }
  });
});

var _IconCanadianDollar = _interopRequireWildcard(require("./IconCanadianDollar"));

Object.keys(_IconCanadianDollar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconCanadianDollar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconCanadianDollar[key];
    }
  });
});

var _IconCardano = _interopRequireWildcard(require("./IconCardano"));

Object.keys(_IconCardano).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconCardano[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconCardano[key];
    }
  });
});

var _IconChilieanPeso = _interopRequireWildcard(require("./IconChilieanPeso"));

Object.keys(_IconChilieanPeso).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconChilieanPeso[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconChilieanPeso[key];
    }
  });
});

var _IconChineseYuan = _interopRequireWildcard(require("./IconChineseYuan"));

Object.keys(_IconChineseYuan).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconChineseYuan[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconChineseYuan[key];
    }
  });
});

var _IconDanishKrone = _interopRequireWildcard(require("./IconDanishKrone"));

Object.keys(_IconDanishKrone).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconDanishKrone[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconDanishKrone[key];
    }
  });
});

var _IconDogecoin = _interopRequireWildcard(require("./IconDogecoin"));

Object.keys(_IconDogecoin).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconDogecoin[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconDogecoin[key];
    }
  });
});

var _IconEthereum = _interopRequireWildcard(require("./IconEthereum"));

Object.keys(_IconEthereum).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconEthereum[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconEthereum[key];
    }
  });
});

var _IconEuro = _interopRequireWildcard(require("./IconEuro"));

Object.keys(_IconEuro).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconEuro[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconEuro[key];
    }
  });
});

var _IconGbPoundSterling = _interopRequireWildcard(require("./IconGbPoundSterling"));

Object.keys(_IconGbPoundSterling).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconGbPoundSterling[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconGbPoundSterling[key];
    }
  });
});

var _IconHongKongDollar = _interopRequireWildcard(require("./IconHongKongDollar"));

Object.keys(_IconHongKongDollar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconHongKongDollar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconHongKongDollar[key];
    }
  });
});

var _IconIcelandicKrona = _interopRequireWildcard(require("./IconIcelandicKrona"));

Object.keys(_IconIcelandicKrona).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconIcelandicKrona[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconIcelandicKrona[key];
    }
  });
});

var _IconIndianRupee = _interopRequireWildcard(require("./IconIndianRupee"));

Object.keys(_IconIndianRupee).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconIndianRupee[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconIndianRupee[key];
    }
  });
});

var _IconJapaneseYen = _interopRequireWildcard(require("./IconJapaneseYen"));

Object.keys(_IconJapaneseYen).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconJapaneseYen[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconJapaneseYen[key];
    }
  });
});

var _IconLitecoin = _interopRequireWildcard(require("./IconLitecoin"));

Object.keys(_IconLitecoin).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconLitecoin[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconLitecoin[key];
    }
  });
});

var _IconNewTaiwanDollar = _interopRequireWildcard(require("./IconNewTaiwanDollar"));

Object.keys(_IconNewTaiwanDollar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconNewTaiwanDollar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconNewTaiwanDollar[key];
    }
  });
});

var _IconNewZealandDollar = _interopRequireWildcard(require("./IconNewZealandDollar"));

Object.keys(_IconNewZealandDollar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconNewZealandDollar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconNewZealandDollar[key];
    }
  });
});

var _IconPolandZloty = _interopRequireWildcard(require("./IconPolandZloty"));

Object.keys(_IconPolandZloty).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconPolandZloty[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconPolandZloty[key];
    }
  });
});

var _IconRipple = _interopRequireWildcard(require("./IconRipple"));

Object.keys(_IconRipple).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconRipple[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconRipple[key];
    }
  });
});

var _IconRussianRuble = _interopRequireWildcard(require("./IconRussianRuble"));

Object.keys(_IconRussianRuble).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconRussianRuble[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconRussianRuble[key];
    }
  });
});

var _IconSingaporeDollar = _interopRequireWildcard(require("./IconSingaporeDollar"));

Object.keys(_IconSingaporeDollar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconSingaporeDollar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconSingaporeDollar[key];
    }
  });
});

var _IconSoc = _interopRequireWildcard(require("./IconSoc"));

Object.keys(_IconSoc).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconSoc[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconSoc[key];
    }
  });
});

var _IconSouthKoreanWon = _interopRequireWildcard(require("./IconSouthKoreanWon"));

Object.keys(_IconSouthKoreanWon).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconSouthKoreanWon[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconSouthKoreanWon[key];
    }
  });
});

var _IconSwedishKrona = _interopRequireWildcard(require("./IconSwedishKrona"));

Object.keys(_IconSwedishKrona).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconSwedishKrona[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconSwedishKrona[key];
    }
  });
});

var _IconSwissFranc = _interopRequireWildcard(require("./IconSwissFranc"));

Object.keys(_IconSwissFranc).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconSwissFranc[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconSwissFranc[key];
    }
  });
});

var _IconThaiBaht = _interopRequireWildcard(require("./IconThaiBaht"));

Object.keys(_IconThaiBaht).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconThaiBaht[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconThaiBaht[key];
    }
  });
});

var _IconTron = _interopRequireWildcard(require("./IconTron"));

Object.keys(_IconTron).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconTron[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconTron[key];
    }
  });
});

var _IconTurkishLira = _interopRequireWildcard(require("./IconTurkishLira"));

Object.keys(_IconTurkishLira).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconTurkishLira[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconTurkishLira[key];
    }
  });
});

var _IconUsaDollar = _interopRequireWildcard(require("./IconUsaDollar"));

Object.keys(_IconUsaDollar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _IconUsaDollar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _IconUsaDollar[key];
    }
  });
});

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
