// @heathmont/moon-core/lib/heading/Heading.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _getFontSize = _interopRequireDefault(require("../private/text/getFontSize"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Heading = _styledComponents["default"].h3.withConfig({
  displayName: "Heading",
  componentId: "sc-1jzu9kp-0"
})(function (_ref) {
  var color = _ref.color,
      theme = _ref.theme;
  return {
    color: (0, _moonUtils.themed)('color', color)(theme)
  };
}, function (_ref2) {
  var _ref2$size = _ref2.size,
      size = _ref2$size === void 0 ? 16 : _ref2$size;
  return (0, _getFontSize["default"])(size);
}, function (_ref3) {
  var textAlign = _ref3.textAlign;
  return textAlign && {
    textAlign: textAlign
  };
}, function (_ref4) {
  var isRegular = _ref4.isRegular,
      fontWeight = _ref4.theme.fontWeight;
  return isRegular ? {
    fontWeight: fontWeight.normal
  } : {
    fontWeight: fontWeight.semibold
  };
});

Heading.defaultProps = {
  color: 'bulma.100',
  size: 16
};
var _default = Heading;
exports["default"] = _default;
