// @heathmont/moon-themes/lib/es/luckyslots/luckyslots.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
/*
  This theme is a legacy and would be removed soon
  Use luckySlotsDark and luckslotsLight
*/

var color = {
  piccolo: {
    120: '#E00038',
    100: '#FA003E',
    80: '#F32055'
  },
  hit: {
    120: '#4345B5',
    100: '#5456E2',
    80: '#7678E8'
  },
  goten: {
    100: '#FFFFFF'
  },
  goku: {
    100: '#3133C4',
    80: '#2528B6',
    40: '#111397',
    10: '#020473'
  },
  gohan: {
    100: '#2A0D68',
    80: '#351579',
    40: '#4D2B97',
    10: '#6745B0'
  },
  beerus: {
    100: '#353B50'
  },
  bulma: {
    100: '#FFFFFF'
  },
  trunks: {
    100: '#DBE2F5'
  }
};

var luckyslots = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'LuckySlots',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  colorScheme: 'dark'
});

export default luckyslots;
