// @heathmont/moon-components/lib/bubble/Bubble.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _moonUtils = require("@heathmont/moon-utils");

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Bubble = _styledComponents["default"].div.withConfig({
  displayName: "Bubble",
  componentId: "vks08m-0"
})(function (_ref) {
  var theme = _ref.theme;
  return [{
    display: 'inline-block',
    textAlign: 'right',
    color: theme.color.goten[100],
    background: theme.color.hit[100],
    fontSize: (0, _moonUtils.rem)(12),
    lineHeight: (0, _moonUtils.rem)(16),
    padding: (0, _moonUtils.rem)(12),
    borderRadius: (0, _moonUtils.rem)(4),
    position: 'relative',
    boxShadow: theme.boxShadow,
    '&::after': {
      content: '""',
      borderColor: 'transparent',
      borderTopColor: theme.color.hit[100],
      borderWidth: (0, _moonUtils.rem)(8),
      borderBottomWidth: 0,
      borderStyle: 'solid',
      width: 0,
      height: 0,
      position: 'absolute',
      top: '100%',
      right: (0, _moonUtils.rem)(12),
      marginTop: "-".concat((0, _moonUtils.rem)(1))
    }
  }, function (_ref2) {
    var position = _ref2.position;
    return position === 'bottom' && {
      '&::after': {
        top: 'initial',
        marginTop: 0,
        borderBottomWidth: (0, _moonUtils.rem)(8),
        bottom: '100%',
        marginBottom: "-".concat((0, _moonUtils.rem)(1)),
        borderTopWidth: 0,
        borderBottomColor: theme.color.hit[100]
      }
    };
  }];
});

var _default = Bubble;
exports["default"] = _default;
