// @heathmont/moon-components/lib/es/modal/ModalContent.js
import React from 'react';
import styled, { keyframes, css } from 'styled-components';
import IconClose from '../private/icons/IconClose';
import { rem, themed } from '@heathmont/moon-utils';
var fadeOpen = keyframes(["0%{opacity:0;transform:scale(0.8);}100%{opacity:1;transform:scale(1);}"]);
var fadeClose = keyframes(["0%{opacity:1;transform:scale(1);}100%{opacity:0;transform:scale(0.9);}"]);
var Content = styled.div.withConfig({
  displayName: "ModalContent__Content",
  componentId: "wr3ipb-0"
})(["animation:", ";background:", ";border-radius:", ";box-shadow:", ";max-width:", ";padding:", ";backdrop-filter:", ";position:absolute;transition:all 2s;width:100%;z-index:99;"], function (_ref) {
  var showUnmountAnimation = _ref.showUnmountAnimation;
  return showUnmountAnimation ? css(["", " 0.2s ease-in-out"], fadeClose) : css(["", " 0.2s ease-in-out"], fadeOpen);
}, function (_ref2) {
  var backgroundColor = _ref2.backgroundColor,
      glassBackground = _ref2.glassBackground,
      theme = _ref2.theme;
  return glassBackground ? 'rgba(255,255,255,0.4)' : backgroundColor ? themed('color', backgroundColor)(theme) : themed('color', 'gohan.100')(theme);
}, function (_ref3) {
  var borderRadius = _ref3.borderRadius,
      theme = _ref3.theme;
  return borderRadius ? rem(borderRadius) : rem(theme.radius["default"]);
}, function (_ref4) {
  var glassBackground = _ref4.glassBackground;
  return glassBackground ? '0px -2px 8px rgba(102, 145, 247, 0.04), 0px 16px 24px rgba(102, 145, 246, 0.10)' : '0px 8px 24px -6px rgba(0, 0, 0, 0.16), 0px 0px 0px rgba(0, 0, 0, 0.4)';
}, function (_ref5) {
  var size = _ref5.size;
  return size == 'large' ? rem(1136) : size == 'medium' ? rem(752) : rem(580);
}, function (_ref6) {
  var padding = _ref6.padding;
  return padding ? rem(padding) : rem(16);
}, function (_ref7) {
  var glassBackground = _ref7.glassBackground;
  return glassBackground ? 'blur(10px)' : 'none';
});
var ButtonClose = styled(IconClose).withConfig({
  displayName: "ModalContent__ButtonClose",
  componentId: "wr3ipb-1"
})(["position:absolute;top:", ";right:", ";cursor:pointer;"], rem(16), rem(16));

var ModalContent = function ModalContent(_ref8) {
  var backgroundColor = _ref8.backgroundColor,
      borderRadius = _ref8.borderRadius,
      buttonClose = _ref8.buttonClose,
      className = _ref8.className,
      content = _ref8.content,
      glassBackground = _ref8.glassBackground,
      handleClose = _ref8.handleClose,
      padding = _ref8.padding,
      size = _ref8.size,
      showUnmountAnimation = _ref8.showUnmountAnimation;
  return /*#__PURE__*/React.createElement(Content, {
    backgroundColor: backgroundColor,
    borderRadius: borderRadius,
    className: className,
    glassBackground: glassBackground,
    padding: padding,
    size: size,
    showUnmountAnimation: showUnmountAnimation
  }, buttonClose && /*#__PURE__*/React.createElement(ButtonClose, {
    fontSize: "0.750rem",
    color: "bulma.100",
    onClick: handleClose
  }), content);
};

export default ModalContent;
