// @heathmont/moon-components/lib/es/modal/Modal.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import React, { useEffect, useState } from 'react';
import { createPortal } from 'react-dom';
import styled from 'styled-components';
import ModalOverlay from './ModalOverlay';
import ModalContent from './ModalContent';
var ModalWrapper = styled.div.withConfig({
  displayName: "Modal__ModalWrapper",
  componentId: "sc-1wqhe17-0"
})(["display:flex;justify-content:center;align-items:center;position:fixed;inset:0;"]);

var Modal = function Modal(_ref) {
  var backgroundColor = _ref.backgroundColor,
      borderRadius = _ref.borderRadius,
      buttonClose = _ref.buttonClose,
      content = _ref.content,
      className = _ref.className,
      clickOutsideClose = _ref.clickOutsideClose,
      disableOverlay = _ref.disableOverlay,
      handleClose = _ref.handleClose,
      glassBackground = _ref.glassBackground,
      isOpen = _ref.isOpen,
      padding = _ref.padding,
      size = _ref.size;

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      showUnmountAnimation = _useState2[0],
      setShowUnmountAnimation = _useState2[1];

  var closeOnEscapeKeyDown = function closeOnEscapeKeyDown(e) {
    if ((e.charCode || e.keyCode) === 27) {
      closeAnimation();
    }
  };

  useEffect(function () {
    document.body.addEventListener('keydown', closeOnEscapeKeyDown);
    return function cleanup() {
      document.body.removeEventListener('keydown', closeOnEscapeKeyDown);
    };
  }, []);

  var closeAnimation = function closeAnimation() {
    setShowUnmountAnimation(true);
    setTimeout(function () {
      setShowUnmountAnimation(false);
      handleClose && handleClose();
    }, 200);
  };

  if (!isOpen) {
    return null;
  }

  return /*#__PURE__*/React.createElement(React.Fragment, null, createPortal( /*#__PURE__*/React.createElement(ModalWrapper, {
    className: className
  }, /*#__PURE__*/React.createElement(ModalOverlay, {
    clickOutsideClose: clickOutsideClose,
    disableOverlay: disableOverlay,
    glassBackground: glassBackground,
    handleClose: closeAnimation,
    showUnmountAnimation: showUnmountAnimation
  }), /*#__PURE__*/React.createElement(ModalContent, {
    backgroundColor: backgroundColor,
    borderRadius: borderRadius,
    buttonClose: buttonClose,
    content: content,
    handleClose: closeAnimation,
    glassBackground: glassBackground,
    padding: padding,
    size: size,
    showUnmountAnimation: showUnmountAnimation
  })), document.body));
};

export default Modal;
