// @heathmont/moon-components/lib/es/modal/ModalOverlay.js
import React from 'react';
import styled, { keyframes, css } from 'styled-components';
var fadeOpen = keyframes(["0%{opacity:0;}100%{opacity:1;}"]);
var fadeClose = keyframes(["0%{opacity:1;}100%{opacity:0;}"]);
var StyledOverlay = styled.div.withConfig({
  displayName: "ModalOverlay__StyledOverlay",
  componentId: "whatqc-0"
})(["position:fixed;inset:0;background:", ";opacity:1;animation:", ";display:", ";"], function (_ref) {
  var glassBackground = _ref.glassBackground;
  return glassBackground ? 'rgba(255,255,255,0)' : 'rgba(0, 0, 0, 0.4)';
}, function (_ref2) {
  var showUnmountAnimation = _ref2.showUnmountAnimation;
  return showUnmountAnimation ? css(["", " 0.2s ease-in-out"], fadeClose) : css(["", " 0.2s ease-in-out"], fadeOpen);
}, function (_ref3) {
  var disableOverlay = _ref3.disableOverlay;
  return disableOverlay ? 'none' : 'block';
});

var ModalOverlay = function ModalOverlay(_ref4) {
  var children = _ref4.children,
      clickOutsideClose = _ref4.clickOutsideClose,
      disableOverlay = _ref4.disableOverlay,
      glassBackground = _ref4.glassBackground,
      handleClose = _ref4.handleClose,
      showUnmountAnimation = _ref4.showUnmountAnimation;
  return /*#__PURE__*/React.createElement(StyledOverlay, {
    disableOverlay: disableOverlay,
    glassBackground: glassBackground,
    clickOutsideClose: clickOutsideClose,
    onClick: clickOutsideClose ? handleClose : undefined,
    showUnmountAnimation: showUnmountAnimation
  }, children);
};

export default ModalOverlay;
