// @heathmont/moon-components/lib/es/search/private/components/Form.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import styled from 'styled-components';
import { rem } from '@heathmont/moon-utils';
import { SearchInputActive } from './Input';
import { Results, resultsActive } from './Results';
import { zIndex } from './settings';
/**
 * Search Form
 * The functional container for the search component.
 * We aren't using Design System form because we use focus-within logic here
 */

export var SearchForm = styled.form.withConfig({
  displayName: "Form__SearchForm",
  componentId: "sc-7uq563-0"
})(function (_ref) {
  var theme = _ref.theme;
  return _defineProperty({
    fontSize: rem(theme.space["default"]),
    position: 'relative',
    overflow: 'visible',
    width: '100%'
  }, "&:focus-within, ".concat(SearchInputActive), _defineProperty({
    zIndex: zIndex.searchInputActive
  }, "~ ".concat(Results, ":not(:empty)"), resultsActive(theme)));
});
