// @heathmont/moon-core/lib/es/ageLimit/AgeLimit.js
import * as React from 'react';
import styled from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';
import { useTheme } from '@heathmont/moon-themes';
var Wrapper = styled.div.withConfig({
  displayName: "AgeLimit__Wrapper",
  componentId: "sc-1v7pumr-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    boxSizing: 'border-box',
    border: "2px ".concat(theme.color.chiChi[100], " solid"),
    borderRadius: rem(theme.radius.largest),
    padding: rem(theme.space.small),
    width: rem(40),
    height: rem(40),
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    overflow: 'hidden'
  };
});
var Caption = styled.span.withConfig({
  displayName: "AgeLimit__Caption",
  componentId: "sc-1v7pumr-1"
})(function (_ref2) {
  var theme = _ref2.theme,
      color = _ref2.color;
  return {
    color: color,
    fontSize: rem(14),
    lineHeight: 1,
    fontWeight: theme.fontWeight.semibold
  };
});

var AgeLimit = function AgeLimit(_ref3) {
  var caption = _ref3.caption,
      _ref3$color = _ref3.color,
      color = _ref3$color === void 0 ? 'bulma.100' : _ref3$color;
  var theme = useTheme();
  var captionColor = themed('color', color)(theme);
  return /*#__PURE__*/React.createElement(Wrapper, null, /*#__PURE__*/React.createElement(Caption, {
    color: captionColor
  }, caption));
};

export default AgeLimit;
