// @heathmont/moon-components/lib/es/badge/Badge.js
import styled from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';
var Badge = styled.span.withConfig({
  displayName: "Badge",
  componentId: "gd8hvq-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      fontWeight = _ref$theme.fontWeight,
      radius = _ref$theme.radius;
  return {
    display: 'flex',
    fontWeight: fontWeight.semibold,
    textTransform: 'uppercase',
    borderRadius: rem(radius.largest),
    letterSpacing: rem(1)
  };
}, function (_ref2) {
  var size = _ref2.size;
  return {
    padding: size === 'small' ? "0 ".concat(rem(8)) : "0 ".concat(rem(4)),
    fontSize: size === 'small' ? rem(10) : rem(8),
    lineHeight: size === 'small' ? rem(16) : rem(12)
  };
},
/* If a color or backgroundColor is set, override styles. */
function (_ref3) {
  var color = _ref3.color,
      backgroundColor = _ref3.backgroundColor,
      theme = _ref3.theme;
  return {
    color: themed('color', color)(theme),
    backgroundColor: themed('color', backgroundColor)(theme)
  };
});
Badge.defaultProps = {
  size: 'xsmall'
};
export default Badge;
