// @heathmont/moon-components/lib/dialog/private/layout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DialogMain = exports.DialogContainer = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _rem = _interopRequireDefault(require("polished/lib/helpers/rem"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/* Layout
  =========================================== */
var DialogContainer = _styledComponents["default"].div.withConfig({
  displayName: "layout__DialogContainer",
  componentId: "dksgof-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      radius = _ref$theme.radius,
      space = _ref$theme.space;
  return {
    position: 'relative',
    outline: 'none',
    borderRadius: (0, _rem["default"])(radius.small),
    color: color.bulma[100],
    background: color.goku[100],
    boxShadow: "0 ".concat((0, _rem["default"])(space["default"]), " ").concat((0, _rem["default"])(space.large), " rgba(0, 0, 0, 0.5)")
  };
});
/**
 * 1. As bottom margins/paddings are ignored in `overflow: scroll;`, create the
 *    padding with a pseudo element.
 */


exports.DialogContainer = DialogContainer;

var DialogMain = _styledComponents["default"].main.withConfig({
  displayName: "layout__DialogMain",
  componentId: "dksgof-1"
})(function (_ref2) {
  var _ref2$theme = _ref2.theme,
      space = _ref2$theme.space,
      breakpoint = _ref2$theme.breakpoint;
  return _defineProperty({
    padding: (0, _rem["default"])(20)
  }, (0, _moonUtils.mq)(breakpoint.medium), {
    paddingTop: (0, _rem["default"])(space.xlarge),
    paddingLeft: (0, _rem["default"])(space.large),
    paddingRight: (0, _rem["default"])(space.large),
    paddingBottom: (0, _rem["default"])(space.large)
  });
});

exports.DialogMain = DialogMain;
