// @heathmont/moon-components/lib/filterDropdown/FilterDropdown.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _moonUtils = require("@heathmont/moon-utils");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Search = _interopRequireDefault(require("./private/Search"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var FilterDropdownWrapper = _styledComponents["default"].div.withConfig({
  displayName: "FilterDropdown__FilterDropdownWrapper",
  componentId: "sc-1ddlktp-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    display: 'inline-block',
    backgroundColor: theme.color.gohan[100],
    borderRadius: (0, _moonUtils.rem)(theme.radius["default"]),
    boxShadow: theme.boxShadow,
    overflow: 'hidden',
    fontSize: (0, _moonUtils.rem)(14),
    padding: (0, _moonUtils.rem)(12)
  };
});

var Header = _styledComponents["default"].div.withConfig({
  displayName: "FilterDropdown__Header",
  componentId: "sc-1ddlktp-1"
})({
  marginBottom: (0, _moonUtils.rem)(24)
});

var FilterDropdown = function FilterDropdown(_ref2) {
  var searchPlaceholder = _ref2.searchPlaceholder,
      onSearchChange = _ref2.onSearchChange,
      children = _ref2.children,
      _ref2$minWidth = _ref2.minWidth,
      minWidth = _ref2$minWidth === void 0 ? (0, _moonUtils.rem)(240) : _ref2$minWidth,
      _ref2$width = _ref2.width,
      width = _ref2$width === void 0 ? 'auto' : _ref2$width;
  return /*#__PURE__*/_react["default"].createElement(FilterDropdownWrapper, {
    style: {
      width: width,
      minWidth: minWidth
    }
  }, !!onSearchChange && /*#__PURE__*/_react["default"].createElement(Header, null, /*#__PURE__*/_react["default"].createElement(_Search["default"], {
    placeholder: searchPlaceholder,
    onChange: onSearchChange
  })), children);
};

var _default = FilterDropdown;
exports["default"] = _default;
