// @heathmont/moon-components/lib/es/filterDropdown/FilterDropdown.js
import { rem } from '@heathmont/moon-utils';
import React from 'react';
import styled from 'styled-components';
import Search from './private/Search';
var FilterDropdownWrapper = styled.div.withConfig({
  displayName: "FilterDropdown__FilterDropdownWrapper",
  componentId: "sc-1ddlktp-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    display: 'inline-block',
    backgroundColor: theme.color.gohan[100],
    borderRadius: rem(theme.radius["default"]),
    boxShadow: theme.boxShadow,
    overflow: 'hidden',
    fontSize: rem(14),
    padding: rem(12)
  };
});
var Header = styled.div.withConfig({
  displayName: "FilterDropdown__Header",
  componentId: "sc-1ddlktp-1"
})({
  marginBottom: rem(24)
});

var FilterDropdown = function FilterDropdown(_ref2) {
  var searchPlaceholder = _ref2.searchPlaceholder,
      onSearchChange = _ref2.onSearchChange,
      children = _ref2.children,
      _ref2$minWidth = _ref2.minWidth,
      minWidth = _ref2$minWidth === void 0 ? rem(240) : _ref2$minWidth,
      _ref2$width = _ref2.width,
      width = _ref2$width === void 0 ? 'auto' : _ref2$width;
  return /*#__PURE__*/React.createElement(FilterDropdownWrapper, {
    style: {
      width: width,
      minWidth: minWidth
    }
  }, !!onSearchChange && /*#__PURE__*/React.createElement(Header, null, /*#__PURE__*/React.createElement(Search, {
    placeholder: searchPlaceholder,
    onChange: onSearchChange
  })), children);
};

export default FilterDropdown;
