// @heathmont/moon-components/lib/fileInput/FileInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _hideVisually = _interopRequireDefault(require("polished/lib/mixins/hideVisually"));

var _moonUtils = require("@heathmont/moon-utils");

var _input = require("../private/input/input");

var _settings = require("../private/input/settings");

var _IconUpload = _interopRequireDefault(require("../private/icons/IconUpload"));

var _excluded = ["placeholder", "label", "error", "id"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var labelID = function labelID(id) {
  return "".concat(id, "-label");
};
/**
 * Styles
 *
 * Styling <input type='file'/> is tricky. The `label` acts as the main
 * interaction, with the input visually hidden from the user.
 */


var FileInputLabel = _styledComponents["default"].span.withConfig({
  displayName: "FileInput__FileInputLabel",
  componentId: "sc-1urkz9z-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    display: 'block',
    fontSize: (0, _moonUtils.rem)(16),
    color: (0, _settings.inputColors)('label')(theme),
    marginBottom: (0, _moonUtils.rem)(theme.space.small)
  };
});

var FileInputElement = (0, _styledComponents["default"])(_input.Input.withComponent('label')).withConfig({
  displayName: "FileInput__FileInputElement",
  componentId: "sc-1urkz9z-1"
})(function (_ref2) {
  var error = _ref2.error,
      theme = _ref2.theme;
  return [{
    width: 'auto',
    position: 'relative',
    display: 'block',
    color: (0, _settings.inputColors)('label')(theme),
    cursor: 'pointer',
    backgroundImage: (0, _moonUtils.inlineSvg)( /*#__PURE__*/_react["default"].createElement(_IconUpload["default"], {
      color: (0, _settings.inputColors)('label')(theme)
    })),
    '&:focus-within': {
      borderColor: (0, _settings.inputColors)('label')(theme)
    }
  }, error && {
    borderColor: theme.color.chiChi[100]
  }];
});
/**
 * Component
 */

var FileInput = function FileInput(_ref3) {
  var _ref3$placeholder = _ref3.placeholder,
      placeholder = _ref3$placeholder === void 0 ? 'Choose file…' : _ref3$placeholder,
      label = _ref3.label,
      error = _ref3.error,
      id = _ref3.id,
      props = _objectWithoutProperties(_ref3, _excluded);

  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(FileInputLabel, {
    id: labelID(id)
  }, label), /*#__PURE__*/_react["default"].createElement(FileInputElement, {
    withIcon: true,
    error: error
  }, /*#__PURE__*/_react["default"].createElement("span", null, placeholder), /*#__PURE__*/_react["default"].createElement("input", _extends({
    id: id,
    type: "file",
    "aria-describedby": labelID(id),
    style: (0, _hideVisually["default"])()
  }, props))));
};

var _default = FileInput;
exports["default"] = _default;
