// @heathmont/moon-utils/lib/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  breakout: true,
  container: true,
  disabled: true,
  em: true,
  focus: true,
  hideVisually: true,
  inlineSvg: true,
  mq: true,
  rem: true,
  spacing: true,
  themed: true,
  uniqueId: true
};
Object.defineProperty(exports, "breakout", {
  enumerable: true,
  get: function get() {
    return _breakout["default"];
  }
});
Object.defineProperty(exports, "container", {
  enumerable: true,
  get: function get() {
    return _container["default"];
  }
});
Object.defineProperty(exports, "disabled", {
  enumerable: true,
  get: function get() {
    return _disabled["default"];
  }
});
Object.defineProperty(exports, "em", {
  enumerable: true,
  get: function get() {
    return _em["default"];
  }
});
Object.defineProperty(exports, "focus", {
  enumerable: true,
  get: function get() {
    return _focus["default"];
  }
});
Object.defineProperty(exports, "hideVisually", {
  enumerable: true,
  get: function get() {
    return _hideVisually["default"];
  }
});
Object.defineProperty(exports, "inlineSvg", {
  enumerable: true,
  get: function get() {
    return _inlineSvg["default"];
  }
});
Object.defineProperty(exports, "mq", {
  enumerable: true,
  get: function get() {
    return _mq["default"];
  }
});
Object.defineProperty(exports, "rem", {
  enumerable: true,
  get: function get() {
    return _rem["default"];
  }
});
Object.defineProperty(exports, "spacing", {
  enumerable: true,
  get: function get() {
    return _spacing["default"];
  }
});
Object.defineProperty(exports, "themed", {
  enumerable: true,
  get: function get() {
    return _themed["default"];
  }
});
Object.defineProperty(exports, "uniqueId", {
  enumerable: true,
  get: function get() {
    return _uniqueId["default"];
  }
});

var _breakout = _interopRequireWildcard(require("./breakout/breakout"));

Object.keys(_breakout).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _breakout[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _breakout[key];
    }
  });
});

var _container = _interopRequireWildcard(require("./container/container"));

Object.keys(_container).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _container[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _container[key];
    }
  });
});

var _disabled = _interopRequireWildcard(require("./disabled/disabled"));

Object.keys(_disabled).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _disabled[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _disabled[key];
    }
  });
});

var _em = _interopRequireWildcard(require("./em/em"));

Object.keys(_em).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _em[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _em[key];
    }
  });
});

var _focus = _interopRequireWildcard(require("./focus/focus"));

Object.keys(_focus).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _focus[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _focus[key];
    }
  });
});

var _hideVisually = _interopRequireWildcard(require("./hideVisually/hideVisually"));

Object.keys(_hideVisually).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _hideVisually[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _hideVisually[key];
    }
  });
});

var _inlineSvg = _interopRequireWildcard(require("./inlineSvg/inlineSvg"));

Object.keys(_inlineSvg).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _inlineSvg[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _inlineSvg[key];
    }
  });
});

var _mq = _interopRequireWildcard(require("./mq/mq"));

Object.keys(_mq).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _mq[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _mq[key];
    }
  });
});

var _rem = _interopRequireWildcard(require("./rem/rem"));

Object.keys(_rem).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _rem[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _rem[key];
    }
  });
});

var _spacing = _interopRequireWildcard(require("./spacing/spacing"));

Object.keys(_spacing).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _spacing[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _spacing[key];
    }
  });
});

var _themed = _interopRequireWildcard(require("./themed/themed"));

Object.keys(_themed).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _themed[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _themed[key];
    }
  });
});

var _uniqueId = _interopRequireWildcard(require("./uniqueId/uniqueId"));

Object.keys(_uniqueId).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _uniqueId[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _uniqueId[key];
    }
  });
});

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
