// @heathmont/moon-components/lib/es/emptyState/EmptyState.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import styled from 'styled-components';
import { rem, mq } from '@heathmont/moon-utils';
var Wrapper = styled.div.withConfig({
  displayName: "EmptyState__Wrapper",
  componentId: "jz2esn-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      breakpoint = _ref$theme.breakpoint,
      color = _ref$theme.color;
  return _defineProperty({
    borderRadius: rem(12),
    backgroundColor: color.gohan[100],
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundRepeat: 'no-repeat',
    backgroundPositionY: '-200px',
    backgroundPositionX: '-300px',
    padding: rem(64)
  }, mq(breakpoint.large), {
    flexDirection: 'row',
    padding: rem(40),
    backgroundPositionX: '160px',
    backgroundPositionY: 'center'
  });
});
var IconWrapper = styled.div.withConfig({
  displayName: "EmptyState__IconWrapper",
  componentId: "jz2esn-1"
})(function (_ref3) {
  var breakpoint = _ref3.theme.breakpoint;
  return _defineProperty({
    marginBottom: rem(40)
  }, mq(breakpoint.large), {
    marginBottom: 0
  });
});
var TextWrapper = styled.div.withConfig({
  displayName: "EmptyState__TextWrapper",
  componentId: "jz2esn-2"
})(function (_ref5) {
  var _ref6;

  var breakpoint = _ref5.theme.breakpoint;
  return _ref6 = {
    textAlign: 'center'
  }, _defineProperty(_ref6, mq(breakpoint.large), {
    marginLeft: rem(24),
    textAlign: 'left',
    marginTop: 0
  }), _defineProperty(_ref6, "maxWidth", rem(400)), _defineProperty(_ref6, mq(breakpoint.xlarge), {
    maxWidth: rem(900)
  }), _ref6;
});
var TitleWrapper = styled.p.withConfig({
  displayName: "EmptyState__TitleWrapper",
  componentId: "jz2esn-3"
})(function (_ref7) {
  var fontWeight = _ref7.theme.fontWeight;
  return {
    margin: 0,
    fontWeight: fontWeight.semibold,
    fontSize: rem(20),
    lineHeight: rem(28)
  };
});
var MessageWrapper = styled.p.withConfig({
  displayName: "EmptyState__MessageWrapper",
  componentId: "jz2esn-4"
})(function (_ref8) {
  var _ref8$theme = _ref8.theme,
      color = _ref8$theme.color,
      breakpoint = _ref8$theme.breakpoint;
  return _defineProperty({
    margin: 0,
    fontSize: rem(16),
    lineHeight: rem(24),
    color: color.trunks[100],
    marginTop: rem(20)
  }, mq(breakpoint.large), {
    marginTop: rem(4)
  });
});
var ActionWrapper = styled.div.withConfig({
  displayName: "EmptyState__ActionWrapper",
  componentId: "jz2esn-5"
})(function (_ref10) {
  var breakpoint = _ref10.theme.breakpoint;
  return _defineProperty({
    marginTop: rem(32)
  }, mq(breakpoint.large), {
    marginTop: 0,
    marginLeft: 'auto'
  });
});

var EmptyState = function EmptyState(_ref12) {
  var title = _ref12.title,
      message = _ref12.message,
      icon = _ref12.icon,
      action = _ref12.action;
  return /*#__PURE__*/React.createElement(Wrapper, null, icon && /*#__PURE__*/React.createElement(IconWrapper, null, icon), (title || message) && /*#__PURE__*/React.createElement(TextWrapper, null, title && /*#__PURE__*/React.createElement(TitleWrapper, null, title), message && /*#__PURE__*/React.createElement(MessageWrapper, null, message)), action && /*#__PURE__*/React.createElement(ActionWrapper, null, action));
};

export default EmptyState;
