// @heathmont/moon-components/lib/modal/ModalContent.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _IconClose = _interopRequireDefault(require("../private/icons/IconClose"));

var _moonUtils = require("@heathmont/moon-utils");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var fadeOpen = (0, _styledComponents.keyframes)(["0%{opacity:0;transform:scale(0.8);}100%{opacity:1;transform:scale(1);}"]);
var fadeClose = (0, _styledComponents.keyframes)(["0%{opacity:1;transform:scale(1);}100%{opacity:0;transform:scale(0.9);}"]);

var Content = _styledComponents["default"].div.withConfig({
  displayName: "ModalContent__Content",
  componentId: "wr3ipb-0"
})(["animation:", ";background:", ";border-radius:", ";box-shadow:", ";max-width:", ";padding:", ";backdrop-filter:", ";position:absolute;transition:all 2s;width:100%;z-index:99;"], function (_ref) {
  var showUnmountAnimation = _ref.showUnmountAnimation;
  return showUnmountAnimation ? (0, _styledComponents.css)(["", " 0.2s ease-in-out"], fadeClose) : (0, _styledComponents.css)(["", " 0.2s ease-in-out"], fadeOpen);
}, function (_ref2) {
  var backgroundColor = _ref2.backgroundColor,
      glassBackground = _ref2.glassBackground,
      theme = _ref2.theme;
  return glassBackground ? 'rgba(255,255,255,0.4)' : backgroundColor ? (0, _moonUtils.themed)('color', backgroundColor)(theme) : (0, _moonUtils.themed)('color', 'gohan.100')(theme);
}, function (_ref3) {
  var borderRadius = _ref3.borderRadius,
      theme = _ref3.theme;
  return borderRadius ? (0, _moonUtils.rem)(borderRadius) : (0, _moonUtils.rem)(theme.radius["default"]);
}, function (_ref4) {
  var glassBackground = _ref4.glassBackground;
  return glassBackground ? '0px -2px 8px rgba(102, 145, 247, 0.04), 0px 16px 24px rgba(102, 145, 246, 0.10)' : '0px 8px 24px -6px rgba(0, 0, 0, 0.16), 0px 0px 0px rgba(0, 0, 0, 0.4)';
}, function (_ref5) {
  var size = _ref5.size;
  return size == 'large' ? (0, _moonUtils.rem)(1136) : size == 'medium' ? (0, _moonUtils.rem)(752) : (0, _moonUtils.rem)(580);
}, function (_ref6) {
  var padding = _ref6.padding;
  return padding ? (0, _moonUtils.rem)(padding) : (0, _moonUtils.rem)(16);
}, function (_ref7) {
  var glassBackground = _ref7.glassBackground;
  return glassBackground ? 'blur(10px)' : 'none';
});

var ButtonClose = (0, _styledComponents["default"])(_IconClose["default"]).withConfig({
  displayName: "ModalContent__ButtonClose",
  componentId: "wr3ipb-1"
})(["position:absolute;top:", ";right:", ";cursor:pointer;"], (0, _moonUtils.rem)(16), (0, _moonUtils.rem)(16));

var ModalContent = function ModalContent(_ref8) {
  var backgroundColor = _ref8.backgroundColor,
      borderRadius = _ref8.borderRadius,
      buttonClose = _ref8.buttonClose,
      className = _ref8.className,
      content = _ref8.content,
      glassBackground = _ref8.glassBackground,
      handleClose = _ref8.handleClose,
      padding = _ref8.padding,
      size = _ref8.size,
      showUnmountAnimation = _ref8.showUnmountAnimation;
  return /*#__PURE__*/_react["default"].createElement(Content, {
    backgroundColor: backgroundColor,
    borderRadius: borderRadius,
    className: className,
    glassBackground: glassBackground,
    padding: padding,
    size: size,
    showUnmountAnimation: showUnmountAnimation
  }, buttonClose && /*#__PURE__*/_react["default"].createElement(ButtonClose, {
    fontSize: "0.750rem",
    color: "bulma.100",
    onClick: handleClose
  }), content);
};

var _default = ModalContent;
exports["default"] = _default;
