// @heathmont/moon-components/lib/es/avatar/Avatar.js
import styled from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';
import React from 'react';
import { IconUser } from '@heathmont/moon-assets';
var AvatarWrapper = styled.div.withConfig({
  displayName: "Avatar__AvatarWrapper",
  componentId: "sc-1silcs5-0"
})(function (_ref) {
  var size = _ref.size,
      imageUrl = _ref.imageUrl,
      color = _ref.color,
      backgroundColor = _ref.backgroundColor,
      theme = _ref.theme;
  return [{
    color: themed('color', color)(theme),
    backgroundColor: themed('color', backgroundColor)(theme),
    borderRadius: '50%',
    overflow: 'hidden',
    textTransform: 'uppercase',
    fontWeight: theme.fontWeight.semibold,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundImage: "url(".concat(imageUrl, ")"),
    backgroundSize: 'cover'
  }, size === 'xsmall' && {
    height: rem(24),
    width: rem(24),
    fontSize: rem(14)
  }, size === 'small' && {
    height: rem(32),
    width: rem(32),
    fontSize: rem(18)
  }, size === 'medium' && {
    height: rem(40),
    width: rem(40),
    fontSize: rem(20)
  }, size === 'large' && {
    height: rem(48),
    width: rem(48),
    fontSize: rem(24)
  }];
});

var Avatar = function Avatar(props) {
  var name = props.name,
      imageUrl = props.imageUrl,
      color = props.color;
  if (imageUrl) return /*#__PURE__*/React.createElement(AvatarWrapper, props);
  return /*#__PURE__*/React.createElement(AvatarWrapper, props, name || /*#__PURE__*/React.createElement(IconUser, {
    color: color
  }));
};

Avatar.defaultProps = {
  imageUrl: '',
  size: 'medium'
};
export default Avatar;
