// @heathmont/moon-components/lib/es/banner/Banner.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import React from 'react';
import { rem } from '@heathmont/moon-utils';
import styled from 'styled-components';
import IconBannerInfo from '../private/icons/IconBannerInfo';
import IconCloseSmall from '../private/icons/IconCloseSmall';
import IconError from '../private/icons/IconError';
import IconWarning from '../private/icons/IconWarning';
var BannerWrapper = styled.div.withConfig({
  displayName: "Banner__BannerWrapper",
  componentId: "sc-1uryy7q-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      space = _ref$theme.space;
  return {
    minWidth: rem(280),
    width: '100%',
    padding: "".concat(space["default"], "px ").concat(rem(12)),
    backgroundColor: color.hit[100],
    borderRadius: rem(12),
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'left',
    '& > * + *': {
      marginLeft: rem(12)
    },
    alignItems: 'center',
    'P,  SPAN': {
      fontSize: rem(14),
      lineHeight: rem(20),
      margin: 0
    }
  };
});
var IconWrapper = styled.div.withConfig({
  displayName: "Banner__IconWrapper",
  componentId: "sc-1uryy7q-1"
})({
  marginRight: rem(8)
});
var IconCloseWrapper = styled.div.withConfig({
  displayName: "Banner__IconCloseWrapper",
  componentId: "sc-1uryy7q-2"
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    backgroundColor: theme.color.goku[100],
    borderRadius: rem(theme.radius.largest),
    marginLeft: 'auto',
    cursor: 'pointer',
    width: rem(24),
    height: rem(24),
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  };
});

var Banner = function Banner(_ref3) {
  var message = _ref3.message,
      onClose = _ref3.onClose,
      status = _ref3.status,
      _ref3$isCloseable = _ref3.isCloseable,
      isCloseable = _ref3$isCloseable === void 0 ? false : _ref3$isCloseable;

  var _React$useState = React.useState(true),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      visible = _React$useState2[0],
      setVisible = _React$useState2[1];

  var closeBanner = function closeBanner() {
    return setVisible(false);
  };

  if (!visible) {
    return null;
  }

  return /*#__PURE__*/React.createElement(BannerWrapper, null, status === 'error' && /*#__PURE__*/React.createElement(IconWrapper, null, /*#__PURE__*/React.createElement(IconError, {
    fontSize: "1.5rem"
  })), status === 'warning' && /*#__PURE__*/React.createElement(IconWrapper, null, /*#__PURE__*/React.createElement(IconWarning, {
    fontSize: "1.5rem",
    color: "krillin.100"
  })), status === 'info' && /*#__PURE__*/React.createElement(IconWrapper, null, /*#__PURE__*/React.createElement(IconBannerInfo, {
    fontSize: "1.5rem",
    color: "krillin.100"
  })), message && message, isCloseable && /*#__PURE__*/React.createElement(IconCloseWrapper, {
    onClick: function onClick() {
      if (onClose && typeof onClose === 'function') {
        onClose();
      }

      closeBanner();
    }
  }, /*#__PURE__*/React.createElement(IconCloseSmall, {
    fontSize: "1rem",
    color: "trunks.100"
  })));
};

export default Banner;
