// @heathmont/moon-components/lib/es/filterDropdown/private/Search.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import { IconSearch } from '@heathmont/moon-assets';
import { rem } from '@heathmont/moon-utils';
import React, { useState } from 'react';
import styled from 'styled-components';
var SearchWrapper = styled.div.withConfig({
  displayName: "Search__SearchWrapper",
  componentId: "whj1cg-0"
})({
  position: 'relative'
});
var SearchInput = styled.input.withConfig({
  displayName: "Search__SearchInput",
  componentId: "whj1cg-1"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    width: '100%',
    border: 'none',
    fontSize: rem(14),
    backgroundColor: 'transparent',
    color: theme.color.bulma[100],
    padding: "".concat(rem(2), " ").concat(rem(20), " ").concat(rem(2), " 0"),
    caretColor: theme.color.piccolo[100],
    outline: 'none',
    '&::placeholder': {
      color: theme.color.trunks[100]
    }
  };
});
var Icon = styled(IconSearch).withConfig({
  displayName: "Search__Icon",
  componentId: "whj1cg-2"
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    position: 'absolute',
    color: theme.color.trunks[100],
    width: rem(16),
    height: rem(16),
    right: 0,
    top: '50%',
    transform: 'translateY(-50%)'
  };
});

var Search = function Search(_ref3) {
  var placeholder = _ref3.placeholder,
      onChange = _ref3.onChange;

  var _useState = useState(''),
      _useState2 = _slicedToArray(_useState, 2),
      value = _useState2[0],
      setValue = _useState2[1];

  var onInputChange = function onInputChange(event) {
    var newValue = event.target.value;
    setValue(newValue);
    onChange(newValue);
  };

  return /*#__PURE__*/React.createElement(SearchWrapper, null, /*#__PURE__*/React.createElement(SearchInput, {
    type: "text",
    placeholder: placeholder,
    value: value,
    onChange: onInputChange
  }), /*#__PURE__*/React.createElement(Icon, null));
};

export default Search;
