// @heathmont/moon-components/lib/es/private/label/label.js
var _excluded = ["text", "disabled", "flex", "children", "inline", "inputGrow", "dir"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import styled from 'styled-components';
import { mq, rem } from '@heathmont/moon-utils';
import { Input } from '../input/input';
import { inputColors } from '../input/settings';
import { SelectElement } from '../../select/private/element';
/**
 * Types
 */

/**
 * Styles
 */
var LabelContent = styled.span.withConfig({
  displayName: "label__LabelContent",
  componentId: "sc-4w9rwl-0"
})(function (_ref) {
  var disabled = _ref.disabled,
      flex = _ref.flex,
      dir = _ref.dir,
      theme = _ref.theme;
  return [{
    display: 'block',
    marginBottom: rem(theme.space.small),
    color: inputColors('label')(theme),
    fontSize: rem(theme.base.fontSize)
  }, flex && _defineProperty({}, mq(theme.breakpoint.small), {
    flex: 1,
    paddingRight: rem(theme.space["default"]),
    marginBottom: 0
  }), disabled && {
    opacity: 0.5
  }, dir === 'rtl' && {
    textAlign: 'right'
  }];
});
var LabelFlex = styled.label.withConfig({
  displayName: "label__LabelFlex",
  componentId: "sc-4w9rwl-1"
})(function (_ref3) {
  var flex = _ref3.flex,
      inputGrow = _ref3.inputGrow,
      breakpoint = _ref3.theme.breakpoint;
  return flex && _defineProperty({}, mq(breakpoint.small), _objectSpread({
    display: 'flex',
    alignItems: 'center'
  }, inputGrow && _defineProperty({}, "".concat(Input, ", ").concat(SelectElement), {
    flex: inputGrow
  })));
});
/**
 * Component
 */

var Label = function Label(_ref6) {
  var text = _ref6.text,
      disabled = _ref6.disabled,
      flex = _ref6.flex,
      children = _ref6.children,
      inline = _ref6.inline,
      _ref6$inputGrow = _ref6.inputGrow,
      inputGrow = _ref6$inputGrow === void 0 ? 1 : _ref6$inputGrow,
      dir = _ref6.dir,
      props = _objectWithoutProperties(_ref6, _excluded);

  return inline ? /*#__PURE__*/React.createElement("label", _extends({
    dir: dir
  }, props), text) : /*#__PURE__*/React.createElement(LabelFlex, _extends({
    flex: flex,
    inputGrow: inputGrow
  }, props), /*#__PURE__*/React.createElement(LabelContent, {
    dir: dir,
    disabled: disabled,
    flex: flex
  }, text), React.Children.only(children));
};

export { Label };
