// @heathmont/moon-components/lib/es/tagInput/TagInput.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { rem } from '@heathmont/moon-utils';
import styled from 'styled-components';
import Bubble from '../bubble/Bubble';
export var RemoveBubble = styled(Bubble).withConfig({
  displayName: "TagInput__RemoveBubble",
  componentId: "sc-1qkcwn0-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    position: 'absolute',
    bottom: '100%',
    right: 0,
    pointerEvents: 'none',
    transition: "opacity ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    opacity: 0,
    width: 'auto'
  };
});
export var Tag = styled.div.withConfig({
  displayName: "TagInput__Tag",
  componentId: "sc-1qkcwn0-1"
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    display: 'inline-flex',
    alignItems: 'center',
    backgroundColor: theme.color.goku[100],
    borderRadius: rem(8),
    padding: rem(8),
    color: theme.color.bulma[100],
    fontSize: rem(12),
    minHeight: rem(30),
    fontWeight: theme.fontWeight.normal,
    position: 'relative',
    gap: rem(6)
  };
});
export var TagInputField = styled.input.withConfig({
  displayName: "TagInput__TagInputField",
  componentId: "sc-1qkcwn0-2"
})(function (_ref3) {
  var theme = _ref3.theme;
  return {
    fontSize: rem(16),
    border: 'none',
    background: 'none',
    outline: 'none',
    caretColor: theme.color.piccolo[100],
    color: theme.color.bulma[100],
    minWidth: rem(52),
    margin: "0 0 ".concat(rem(4), " 0"),
    flex: 1,
    '&::placeholder': {
      color: theme.color.trunks[100],
      opacity: 1
    }
  };
});
export var TagInputWrapper = styled.div.withConfig({
  displayName: "TagInput__TagInputWrapper",
  componentId: "sc-1qkcwn0-3"
})({
  position: 'relative'
});
export var TagInputMain = styled.div.withConfig({
  displayName: "TagInput__TagInputMain",
  componentId: "sc-1qkcwn0-4"
})(function (_ref4) {
  var theme = _ref4.theme;
  return [_defineProperty({
    display: 'flex',
    width: '100%',
    minHeight: rem(48),
    alignItems: 'center',
    justifyContent: 'flex-start',
    flexWrap: 'wrap',
    position: 'relative',
    padding: "".concat(rem(8), " ").concat(rem(40), " ").concat(rem(4), " ").concat(rem(12)),
    backgroundColor: theme.color.gohan[100],
    border: "".concat(theme.borderWidth, "px solid ").concat(theme.color.beerus[100]),
    borderRadius: rem(theme.radius["default"]),
    transition: "border-color ".concat(theme.transitionDuration["default"], "s ease"),
    '&:hover': {
      borderColor: theme.color.goku[40]
    },
    '&:focus-within': {
      borderColor: theme.color.piccolo[100]
    }
  }, Tag, {
    margin: "0 ".concat(rem(4), " ").concat(rem(4), " 0")
  }), function (_ref6) {
    var focused = _ref6.focused;
    return focused && {
      borderColor: theme.color.piccolo[100],
      '&:hover': {
        borderColor: theme.color.piccolo[100]
      }
    };
  }, function (_ref7) {
    var error = _ref7.error;
    return error && {
      borderColor: theme.color.chiChi[100],
      '&:focus-within, &:hover': {
        borderColor: theme.color.chiChi[100]
      }
    };
  }, function (_ref8) {
    var disabled = _ref8.disabled;
    return disabled && _defineProperty({
      opacity: 0.5,
      cursor: 'not-allowed'
    }, TagInputField, {
      cursor: 'not-allowed'
    });
  }];
});
export var TagRemove = styled.button.withConfig({
  displayName: "TagInput__TagRemove",
  componentId: "sc-1qkcwn0-5"
})(function (_ref10) {
  var _ref11;

  var theme = _ref10.theme;
  return _ref11 = {
    appearance: 'none',
    background: 'none',
    border: 'none',
    padding: 0,
    fontSize: rem(16),
    display: 'inline-flex',
    alignItems: 'center',
    color: theme.color.trunks[100],
    cursor: 'pointer',
    transition: "color ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    position: 'relative',
    outline: 'none'
  }, _defineProperty(_ref11, RemoveBubble, {
    marginRight: rem(-12),
    marginBottom: rem(18)
  }), _defineProperty(_ref11, '&:hover, &:focus', _defineProperty({
    color: theme.color.chiChi[100]
  }, RemoveBubble, {
    opacity: 1
  })), _ref11;
});
export var ArrowContainer = styled.div.withConfig({
  displayName: "TagInput__ArrowContainer",
  componentId: "sc-1qkcwn0-6"
})(function (_ref12) {
  var theme = _ref12.theme;
  return {
    position: 'absolute',
    top: '50%',
    right: rem(18),
    fontSize: rem(16),
    transform: 'translateY(-50%)',
    color: theme.color.trunks[100]
  };
});
export var RemoveAllButton = styled.button.withConfig({
  displayName: "TagInput__RemoveAllButton",
  componentId: "sc-1qkcwn0-7"
})(function (_ref13) {
  var _ref14;

  var theme = _ref13.theme;
  return _ref14 = {
    background: 'none',
    border: 'none',
    padding: 0,
    fontSize: rem(16),
    color: theme.color.trunks[100],
    cursor: 'pointer',
    display: 'flex',
    alignItems: 'center',
    position: 'absolute',
    top: rem(16),
    right: rem(16),
    transition: "color ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    outline: 'none',
    whiteSpace: 'nowrap'
  }, _defineProperty(_ref14, RemoveBubble, {
    marginRight: rem(-12),
    marginBottom: rem(24)
  }), _defineProperty(_ref14, '&:hover, &:focus', _defineProperty({
    color: theme.color.piccolo[100]
  }, RemoveBubble, {
    opacity: 1
  })), _ref14;
});
export var DropdownWrapper = styled.div.withConfig({
  displayName: "TagInput__DropdownWrapper",
  componentId: "sc-1qkcwn0-8"
})({
  top: '100%',
  left: 0,
  position: 'absolute',
  marginTop: rem(10),
  maxWidth: '100%'
});
export default {
  RemoveBubble: RemoveBubble,
  Tag: Tag,
  TagInputWrapper: TagInputWrapper,
  TagInputMain: TagInputMain,
  TagInputField: TagInputField,
  TagRemove: TagRemove,
  ArrowContainer: ArrowContainer,
  RemoveAllButton: RemoveAllButton,
  DropdownWrapper: DropdownWrapper
};
