// @heathmont/moon-components/lib/tagList/TagList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.CopyButton = exports.CopyWrapper = exports.Dropdown = exports.ShowMore = exports.TagListInternal = exports.TagListWrapper = exports.CopyBubble = void 0;

var _moonUtils = require("@heathmont/moon-utils");

var _moonAssets = require("@heathmont/moon-assets");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Bubble = _interopRequireDefault(require("../bubble/Bubble"));

var _TagInput = require("../tagInput/TagInput");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var CopyBubble = (0, _styledComponents["default"])(_Bubble["default"]).withConfig({
  displayName: "TagList__CopyBubble",
  componentId: "sc-1ebbyur-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    position: 'absolute',
    top: '100%',
    right: 0,
    pointerEvents: 'none',
    transition: "opacity ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    opacity: 0,
    marginTop: (0, _moonUtils.rem)(16),
    marginRight: (0, _moonUtils.rem)(-8)
  };
});
exports.CopyBubble = CopyBubble;

var TagListWrapper = _styledComponents["default"].div.withConfig({
  displayName: "TagList__TagListWrapper",
  componentId: "sc-1ebbyur-1"
})(function (_ref2) {
  var theme = _ref2.theme;
  return _defineProperty({
    display: 'inline-block',
    position: 'relative',
    marginBottom: (0, _moonUtils.rem)(-4)
  }, _TagInput.Tag, {
    color: theme.color.bulma[100]
  });
});

exports.TagListWrapper = TagListWrapper;

var TagListInternal = _styledComponents["default"].div.withConfig({
  displayName: "TagList__TagListInternal",
  componentId: "sc-1ebbyur-2"
})(_defineProperty({
  display: 'flex',
  flexWrap: 'wrap'
}, _TagInput.Tag, {
  margin: "0 ".concat((0, _moonUtils.rem)(4), " ").concat((0, _moonUtils.rem)(4), " 0"),
  '&:last-child': {
    marginRight: 0
  }
}));

exports.TagListInternal = TagListInternal;

var ShowMore = _styledComponents["default"].button.withConfig({
  displayName: "TagList__ShowMore",
  componentId: "sc-1ebbyur-3"
})(function (_ref4) {
  var theme = _ref4.theme;
  return {
    display: 'inline-flex',
    alignItems: 'center',
    backgroundColor: theme.color.goku[100],
    borderRadius: (0, _moonUtils.rem)(8),
    padding: (0, _moonUtils.rem)(8),
    color: theme.color.bulma[100],
    fontSize: (0, _moonUtils.rem)(12),
    height: (0, _moonUtils.rem)(30),
    fontWeight: theme.fontWeight.normal,
    position: 'relative',
    cursor: 'pointer',
    border: 'none',
    transition: "color ".concat(theme.transitionDuration["default"], "s ease, background-color ").concat(theme.transitionDuration["default"], "s ease"),
    marginBottom: (0, _moonUtils.rem)(4),
    '&:hover, &:focus': {
      color: theme.color.goten[100],
      backgroundColor: theme.color.piccolo[100]
    }
  };
});

exports.ShowMore = ShowMore;

var Dropdown = _styledComponents["default"].div.withConfig({
  displayName: "TagList__Dropdown",
  componentId: "sc-1ebbyur-4"
})(function (_ref5) {
  var theme = _ref5.theme;
  return _defineProperty({
    display: 'flex',
    flexWrap: 'wrap',
    background: theme.color.goten[100],
    borderRadius: (0, _moonUtils.rem)(theme.radius["default"]),
    boxShadow: theme.boxShadow,
    position: 'absolute',
    padding: "".concat((0, _moonUtils.rem)(8), " ").concat((0, _moonUtils.rem)(44), " ").concat((0, _moonUtils.rem)(4), " ").concat((0, _moonUtils.rem)(12)),
    width: '100%',
    maxHeight: (0, _moonUtils.rem)(160),
    overflow: 'auto',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    zIndex: theme.zIndex.toggle
  }, _TagInput.Tag, {
    margin: "0 ".concat((0, _moonUtils.rem)(4), " ").concat((0, _moonUtils.rem)(4), " 0")
  });
});

exports.Dropdown = Dropdown;

var CopyWrapper = _styledComponents["default"].div.withConfig({
  displayName: "TagList__CopyWrapper",
  componentId: "sc-1ebbyur-5"
})({
  position: 'absolute',
  top: (0, _moonUtils.rem)(8),
  right: (0, _moonUtils.rem)(12)
});

exports.CopyWrapper = CopyWrapper;

var CopyButton = _styledComponents["default"].button.withConfig({
  displayName: "TagList__CopyButton",
  componentId: "sc-1ebbyur-6"
})(function (_ref7) {
  var _hoverFocus, _ref8;

  var theme = _ref7.theme;
  return _ref8 = {
    height: (0, _moonUtils.rem)(24),
    width: (0, _moonUtils.rem)(24),
    border: 'none',
    background: 'none',
    outline: 'none',
    padding: 0,
    cursor: 'pointer',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative'
  }, _defineProperty(_ref8, _moonAssets.IconCopy, {
    height: (0, _moonUtils.rem)(16)
  }), _defineProperty(_ref8, '&::before', {
    content: '""',
    position: 'absolute',
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    transitionDuration: "".concat(theme.transitionDuration["default"], "s"),
    transitionTimingFunction: 'ease',
    backgroundColor: theme.color.piccolo[100],
    borderRadius: '50%',
    opacity: 0,
    transform: 'none',
    transitionProperty: 'transform, opacity'
  }), _defineProperty(_ref8, '&:hover, &:focus', (_hoverFocus = {}, _defineProperty(_hoverFocus, CopyBubble, {
    opacity: 1
  }), _defineProperty(_hoverFocus, _moonAssets.IconCopy, {
    color: theme.color.piccolo[100]
  }), _defineProperty(_hoverFocus, '&::before', {
    transform: 'scale(1.4)',
    opacity: 0.1
  }), _hoverFocus)), _ref8;
});

exports.CopyButton = CopyButton;
var _default = {
  CopyBubble: CopyBubble,
  TagListWrapper: TagListWrapper,
  TagListInternal: TagListInternal,
  ShowMore: ShowMore,
  Dropdown: Dropdown,
  CopyWrapper: CopyWrapper,
  CopyButton: CopyButton
};
exports["default"] = _default;
