// @heathmont/moon-components/lib/textInput/Input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Input = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Input = _styledComponents["default"].input.withConfig({
  displayName: "Input",
  componentId: "bxe8t3-0"
})(function (_ref) {
  var theme = _ref.theme;
  return [{
    width: '100%',
    maxWidth: '100%',
    padding: "".concat((0, _moonUtils.rem)(theme.space.small), " ").concat((0, _moonUtils.rem)(theme.space["default"])),
    appearance: 'none',
    font: 'inherit',
    fontSize: (0, _moonUtils.rem)(16),
    lineHeight: 1.5,
    color: theme.color.bulma[100],
    backgroundColor: theme.color.gohan[100],
    border: "".concat(theme.borderWidth, "px solid ").concat(theme.color.beerus[100]),
    borderRadius: (0, _moonUtils.rem)(theme.radius.largest),
    transition: "border-color ".concat(theme.transitionDuration["default"], "s ease"),
    WebkitAppearance: 'none',
    '&::placeholder': {
      color: theme.color.trunks[100],
      opacity: 1
    },
    '&:hover:not(:focus):not([disabled])': {
      borderColor: theme.color.goku[40]
    },
    // date inputs will be marked as invalid by default
    '&:not(:placeholder-shown):not([type="date"]):invalid': {
      borderColor: theme.color.chiChi[100]
    },
    '&:invalid, :-moz-ui-invalid': {
      boxShadow: 'none' // Firefox Override

    },
    '&:disabled': {
      opacity: 0.5,
      cursor: 'not-allowed'
    }
  }, function (_ref2) {
    var icon = _ref2.icon;
    return icon && {
      paddingRight: (0, _moonUtils.rem)(theme.space.large),
      backgroundImage: (0, _moonUtils.inlineSvg)(icon),
      backgroundPosition: "right ".concat((0, _moonUtils.rem)(theme.space["default"]), " center"),
      backgroundRepeat: 'no-repeat',
      backgroundSize: (0, _moonUtils.rem)(20)
    };
  }, function (_ref3) {
    var error = _ref3.error;
    return error && {
      borderColor: theme.color.chiChi[100]
    };
  }];
});

exports.Input = Input;
var _default = Input;
exports["default"] = _default;
