// @heathmont/moon-components/lib/es/textInput/Input.js
import styled from 'styled-components';
import { rem, inlineSvg } from '@heathmont/moon-utils';
export var Input = styled.input.withConfig({
  displayName: "Input",
  componentId: "bxe8t3-0"
})(function (_ref) {
  var theme = _ref.theme;
  return [{
    width: '100%',
    maxWidth: '100%',
    padding: "".concat(rem(theme.space.small), " ").concat(rem(theme.space["default"])),
    appearance: 'none',
    font: 'inherit',
    fontSize: rem(16),
    lineHeight: 1.5,
    color: theme.color.bulma[100],
    backgroundColor: theme.color.gohan[100],
    border: "".concat(theme.borderWidth, "px solid ").concat(theme.color.beerus[100]),
    borderRadius: rem(theme.radius.largest),
    transition: "border-color ".concat(theme.transitionDuration["default"], "s ease"),
    WebkitAppearance: 'none',
    '&::placeholder': {
      color: theme.color.trunks[100],
      opacity: 1
    },
    '&:hover:not(:focus):not([disabled])': {
      borderColor: theme.color.goku[40]
    },
    // date inputs will be marked as invalid by default
    '&:not(:placeholder-shown):not([type="date"]):invalid': {
      borderColor: theme.color.chiChi[100]
    },
    '&:invalid, :-moz-ui-invalid': {
      boxShadow: 'none' // Firefox Override

    },
    '&:disabled': {
      opacity: 0.5,
      cursor: 'not-allowed'
    }
  }, function (_ref2) {
    var icon = _ref2.icon;
    return icon && {
      paddingRight: rem(theme.space.large),
      backgroundImage: inlineSvg(icon),
      backgroundPosition: "right ".concat(rem(theme.space["default"]), " center"),
      backgroundRepeat: 'no-repeat',
      backgroundSize: rem(20)
    };
  }, function (_ref3) {
    var error = _ref3.error;
    return error && {
      borderColor: theme.color.chiChi[100]
    };
  }];
});
export default Input;
