// @heathmont/moon-components/lib/es/private/input/input.js
import styled from 'styled-components';
import { rem } from '@heathmont/moon-utils';
import { inputBorder, inputColors, inputFontSize, inputSpacingX, inputSpacingY, inputIconSize, inputIconOffset, inputIconPosition, inputAnimationSpeed, inputLineHeight } from './settings';
export var Input = styled.input.withConfig({
  displayName: "input__Input",
  componentId: "x9ri56-0"
})(function (_ref) {
  var theme = _ref.theme;
  return [{
    width: '100%',
    maxWidth: '100%',
    padding: "".concat(rem(inputSpacingY(theme)), " ").concat(rem(inputSpacingX(theme))),
    appearance: 'none',
    font: 'inherit',
    fontSize: rem(inputFontSize),
    lineHeight: rem(inputLineHeight),
    color: inputColors('text')(theme),
    backgroundColor: inputColors('background')(theme),
    border: inputBorder(theme),
    borderRadius: rem(theme.radius.small),
    transition: "border-color ".concat(inputAnimationSpeed(theme), " ease"),
    WebkitAppearance: 'none',
    '&::placeholder': {
      color: inputColors('placeholder')(theme),
      opacity: 1
    },
    '&:hover:not(:focus):not([disabled])': {
      borderColor: inputColors('borderHover')(theme)
    },
    // date inputs will be marked as invalid by default
    '&:not(:placeholder-shown):not([type="date"]):invalid': {
      borderColor: theme.color.chiChi[100]
    },
    '&:invalid, :-moz-ui-invalid': {
      boxShadow: 'none' // Firefox Override

    },
    '&:disabled': {
      opacity: 0.5,
      cursor: 'not-allowed'
    }
  }, function (_ref2) {
    var withIcon = _ref2.withIcon;
    return withIcon && {
      paddingRight: rem(inputIconOffset(theme)),
      backgroundPosition: "right ".concat(rem(inputIconPosition(theme)), " center"),
      backgroundRepeat: 'no-repeat',
      backgroundSize: rem(inputIconSize)
    };
  }, function (_ref3) {
    var error = _ref3.error;
    return error && {
      borderColor: theme.color.chiChi[100]
    };
  }, function (_ref4) {
    var rounded = _ref4.rounded;
    return rounded && {
      borderRadius: rem(theme.radius.largest)
    };
  }];
});
