// @heathmont/moon-components/lib/reel/Reel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _listItemStyleTypeNone = _interopRequireDefault(require("../listItemStyleTypeNone/listItemStyleTypeNone"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * Reel
 *
 * Based on the Reel component from every-layout.dev
 */
var Reel = _styledComponents["default"].div.withConfig({
  displayName: "Reel",
  componentId: "sc-10l7fx7-0"
})(function (_ref) {
  var height = _ref.height,
      space = _ref.space,
      theme = _ref.theme,
      dir = _ref.dir;
  return [{
    display: 'flex',
    overflowX: 'auto',
    overflowY: 'hidden',
    height: height,

    /**
     * Scroll Behavior
     * 1. Hide Scrollbars on browsers that don't support custom scrollbars.
     * 2. Auto-hide scrollbars on IE/Edge.
     * 3. Create 'padding' around the scrollbar.
     */
    WebkitOverflowScrolling: 'touch',
    scrollbarWidth: 'none'
    /* [1] */
    ,
    '-ms-overflow-style': '-ms-autohiding-scrollbar'
    /* [2] */
    ,
    '::-webkit-scrollbar': {
      width: 12,
      height: 12,
      cursor: 'pointer'
    },
    '::-webkit-scrollbar-thumb': {
      backgroundColor: 'transparent',
      backgroundClip: 'content-box'
      /* [3] */
      ,
      borderRadius: (0, _moonUtils.rem)(theme.radius.largest),
      border: '3px solid transparent'
      /* [3] */

    },
    ':hover::-webkit-scrollbar-thumb': {
      backgroundColor: theme.color.goku[40]
    },

    /**
     * Elements
     * 1. Ensure direct child list-items render without bullets
     * 2. Set content width to auto
     * 3. When content is set to images, ensure they all remain the same height.
     */
    '& > li': _listItemStyleTypeNone["default"]
    /* [1] */
    ,
    '& > *': {
      flex: '0 0 auto'
    }
    /* [2] */
    ,
    '& > img': {
      height: '100%',
      flexBasis: 'auto',
      width: 'auto'
    }
    /* [3] */

  }, dir === 'rtl' ? {
    '& > * + *': {
      marginRight: (0, _moonUtils.themed)('space', space)(theme)
    }
  } : {
    '& > * + *': {
      marginLeft: (0, _moonUtils.themed)('space', space)(theme)
    }
  }];
});

Reel.defaultProps = {
  height: 'auto',
  space: 'default'
};
var _default = Reel;
exports["default"] = _default;
