// @heathmont/moon-components/lib/es/chip/Chip.js
import React from 'react';
import styled from 'styled-components';
import { rem } from '@heathmont/moon-utils';
var StyledChip = styled.div.withConfig({
  displayName: "Chip__StyledChip",
  componentId: "sc-1rq31s5-0"
})(["display:flex;flex-direction:row;justify-content:center;align-items:center;border-radius:", ";padding:", ";line-height:", ";font-size:", ";background:", ";color:", ";"], rem(8), rem(8), function (_ref) {
  var size = _ref.size;
  return size === 'small' ? rem(16) : rem(20);
}, function (_ref2) {
  var size = _ref2.size;
  return size === 'small' ? rem(12) : rem(14);
}, function (_ref3) {
  var theme = _ref3.theme,
      active = _ref3.active;
  return active ? theme.color.frieza[10] : theme.color.goten[100];
}, function (_ref4) {
  var theme = _ref4.theme,
      active = _ref4.active;
  return active ? theme.color.frieza[100] : theme.color.bulma[100];
});
var IconRightWrapper = styled.span.withConfig({
  displayName: "Chip__IconRightWrapper",
  componentId: "sc-1rq31s5-1"
})(["margin-left:", ";"], rem(8));
var IconLeftWrapper = styled.span.withConfig({
  displayName: "Chip__IconLeftWrapper",
  componentId: "sc-1rq31s5-2"
})(["margin-right:", ";"], rem(8));

var Chip = function Chip(_ref5) {
  var children = _ref5.children,
      active = _ref5.active,
      size = _ref5.size,
      iconLeft = _ref5.iconLeft,
      iconRight = _ref5.iconRight;
  return /*#__PURE__*/React.createElement(StyledChip, {
    active: active,
    iconLeft: iconLeft,
    iconRight: iconRight,
    size: size
  }, iconLeft && /*#__PURE__*/React.createElement(IconLeftWrapper, {
    size: size
  }, iconLeft), children, iconRight && /*#__PURE__*/React.createElement(IconRightWrapper, {
    size: size
  }, iconRight));
};

export default Chip;
