// @heathmont/moon-components/lib/inlineMixin/inlineMixin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _listItemStyleTypeNone = _interopRequireDefault(require("../listItemStyleTypeNone/listItemStyleTypeNone"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * 1. If we don't know the exact value (i.e. not a `number`), use `calc` and let
 *    the browser calculate spacing.
 */
var inlineMixin = function inlineMixin(space, justifyContent) {
  return {
    display: 'flex',
    flexWrap: 'wrap',
    overflow: 'hidden',
    justifyContent: justifyContent || 'flex-start',
    alignItems: 'center',
    margin: space && typeof space === 'number' ? space / 2 * -1 : "calc(".concat(space, " / 2 * -1)")
    /* [1] */
    ,
    '& > *': {
      margin: space && typeof space === 'number' ? space / 2 : "calc(".concat(space, " / 2)")
      /* [1] */

    },

    /* Ensure direct child list-items render without bullets */
    '& > li': _listItemStyleTypeNone["default"]
  };
};

var _default = inlineMixin;
exports["default"] = _default;
