// @heathmont/moon-components/lib/private/input/settings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.inputIconPosition = exports.inputIconOffset = exports.inputSpacingY = exports.inputSpacingX = exports.inputIconSize = exports.inputLineHeight = exports.inputFontSize = exports.inputColors = exports.inputBorderWidth = exports.inputBorder = exports.inputAnimationSpeed = void 0;

var inputBorderWidth = function inputBorderWidth(_ref) {
  var borderWidth = _ref.borderWidth;
  return borderWidth;
};

exports.inputBorderWidth = inputBorderWidth;

var inputAnimationSpeed = function inputAnimationSpeed(_ref2) {
  var transitionDuration = _ref2.transitionDuration;
  return "".concat(transitionDuration["default"], "s");
};

exports.inputAnimationSpeed = inputAnimationSpeed;
var inputFontSize = 16;
exports.inputFontSize = inputFontSize;
var inputLineHeight = inputFontSize * 1.5;
exports.inputLineHeight = inputLineHeight;

var inputSpacingX = function inputSpacingX(theme) {
  return theme.space["default"] - inputBorderWidth(theme);
}; // px


exports.inputSpacingX = inputSpacingX;

var inputSpacingY = function inputSpacingY(theme) {
  return 10 - inputBorderWidth(theme);
}; // px


exports.inputSpacingY = inputSpacingY;
var inputIconSize = 20; // px

exports.inputIconSize = inputIconSize;

var inputIconPosition = function inputIconPosition(theme) {
  return inputSpacingX(theme);
};

exports.inputIconPosition = inputIconPosition;

var inputIconOffset = function inputIconOffset(theme) {
  return inputIconSize + inputSpacingX(theme) + inputIconPosition(theme);
};

exports.inputIconOffset = inputIconOffset;

var inputColors = function inputColors(key) {
  return function (_ref3) {
    var color = _ref3.color;
    var themedColor = {
      label: color.trunks[100],
      text: color.bulma[100],
      icon: color.trunks[100],
      placeholder: color.trunks[100],
      borderDefault: color.beerus[100],
      borderHover: color.goku[40],
      background: color.gohan[100],
      disabled: color.goku[80]
    };
    return themedColor[key];
  };
};

exports.inputColors = inputColors;

var inputBorder = function inputBorder(theme) {
  return "".concat(inputBorderWidth(theme), "px solid ").concat(inputColors('borderDefault')(theme));
};

exports.inputBorder = inputBorder;
