// @heathmont/moon-themes/lib/es/shangrilaDark/shangrilaDark.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var color = {
  piccolo: {
    120: '#977135',
    100: '#a57f40',
    80: '#b38b4d'
  },
  hit: {
    120: '#313030',
    100: '#3d3d3d',
    80: '#4a4a4a'
  },
  beerus: {
    100: '#83837c'
  },
  goku: {
    100: '#3d3d3d',
    80: '#4a4a4a',
    40: '#636363',
    10: '#7d7d7d'
  },
  gohan: {
    100: '#575757',
    80: '#636363',
    40: '#7d7d7d',
    10: '#969696'
  },
  goten: {
    100: '#ffffff'
  },
  bulma: {
    100: '#ffffff'
  },
  trunks: {
    100: '#c9c9c0'
  }
};

var shangrilaDark = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'ShangriLa.io',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  colorScheme: 'dark'
});

export default shangrilaDark;
