// @heathmont/moon-components/lib/tabs/Tabs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _hideVisually = _interopRequireDefault(require("polished/lib/mixins/hideVisually"));

var _moonUtils = require("@heathmont/moon-utils");

var _moonThemes = require("@heathmont/moon-themes");

var _Reel = _interopRequireDefault(require("../reel/Reel"));

var _private = require("./private");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var TabNav = _styledComponents["default"].nav.withConfig({
  displayName: "Tabs__TabNav",
  componentId: "sc-7gvh1d-0"
})({
  position: 'relative',
  width: '100%'
});

var SkipLink = _styledComponents["default"].a.withConfig({
  displayName: "Tabs__SkipLink",
  componentId: "sc-7gvh1d-1"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      space = _ref$theme.space;
  return {
    position: 'absolute',
    zIndex: 1,
    padding: "0 ".concat((0, _moonUtils.rem)(space.small)),
    top: '50%',
    left: (0, _moonUtils.rem)(space.small),
    transform: 'translateY(-50%)',
    color: color.bulma[100],
    backgroundColor: color.gohan[80],
    '&:not(:focus)': _objectSpread({}, (0, _hideVisually["default"])())
  };
});
/* 1. Include the indicator as part of the item's box-model. */


var TabList = (0, _styledComponents["default"])(_Reel["default"].withComponent('ul')).withConfig({
  displayName: "Tabs__TabList",
  componentId: "sc-7gvh1d-2"
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    paddingBottom: "".concat((0, _private.underlineOffset)(theme), "px")
    /* [1] */

  };
});

var Tab = _styledComponents["default"].li.withConfig({
  displayName: "Tabs__Tab",
  componentId: "sc-7gvh1d-3"
})({});

var Tabs = function Tabs(_ref3) {
  var id = _ref3.id,
      items = _ref3.items,
      dir = _ref3.dir;
  var autoId = id || "nav-skip-".concat((0, _moonUtils.uniqueId)()); // We render everything except null or undefined items
  // tab != null filters both

  var nonEmptyTabs = Array.isArray(items) ? items.filter(function (tab) {
    return tab != null;
  }) : [];

  var _useTheme = (0, _moonThemes.useTheme)(),
      space = _useTheme.space;
  /* eslint-disable react/no-array-index-key */


  return /*#__PURE__*/_react["default"].createElement(TabNav, null, /*#__PURE__*/_react["default"].createElement(SkipLink, {
    href: "#".concat(autoId)
  }, "Skip to content"), /*#__PURE__*/_react["default"].createElement(TabList, {
    dir: dir,
    space: (0, _moonUtils.rem)(space.medium)
  }, Array.isArray(nonEmptyTabs) && nonEmptyTabs.map(function (tab, index) {
    return /*#__PURE__*/_react["default"].createElement(Tab, {
      key: index
    }, tab);
  })), /*#__PURE__*/_react["default"].createElement("span", {
    id: autoId
  }));
  /* eslint-enable react/no-array-index-key */
};

var _default = Tabs;
exports["default"] = _default;
