// @heathmont/moon-themes/lib/es/livecasinoDark/livecasinoDark.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import avertaStd from '../sharedTokens/avertaStd';
import sharedTokens from '../sharedTokens/sharedTokens';
import supportColors from '../supportColors/supportColors';
var boxShadow = '0px 2.4px 7.2px rgba(0, 0, 0, 0.18), 0px 12.8px 28.8px rgba(0, 0, 0, 0.22)';
var color = {
  piccolo: {
    120: '#cf0232',
    100: '#de0d3e',
    80: '#e61e4d'
  },
  hit: {
    120: '#5b5d71',
    100: '#67697e',
    80: '#74768b'
  },
  beerus: {
    100: '#323548'
  },
  goku: {
    100: '#00070f',
    80: '#021327',
    40: '#0c2c4f',
    10: '#1d4572'
  },
  gohan: {
    100: '#191d33',
    80: '#232843',
    40: '#3a405f',
    10: '#52587a'
  },
  goten: {
    100: '#ffffff'
  },
  bulma: {
    100: '#ffffff'
  },
  trunks: {
    100: '#8697a2'
  }
};

var livecasinoDark = _objectSpread(_objectSpread({}, sharedTokens), {}, {
  brand: 'Livecasino.io',
  fontFace: avertaStd.fontFace,
  fontFamily: avertaStd.fontStack,
  fontSize: {
    body: 14
  },
  fontWeight: avertaStd.fontWeight,
  boxShadow: boxShadow,
  color: _objectSpread(_objectSpread({
    text: color.bulma[100],
    background: color.goku[100]
  }, supportColors), color),
  colorScheme: 'dark'
});

export default livecasinoDark;
