// @heathmont/moon-core/lib/es/badge/Badge.js
import React from 'react';
import styled from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';
var StyledBadge = styled.span.withConfig({
  displayName: "Badge__StyledBadge",
  componentId: "fdg9gi-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      fontWeight = _ref$theme.fontWeight,
      radius = _ref$theme.radius;
  return {
    display: 'flex',
    alignItems: 'center',
    fontWeight: fontWeight.semibold,
    textTransform: 'uppercase',
    borderRadius: rem(radius.largest),
    letterSpacing: rem(1) // TODO

  };
}, function (_ref2) {
  var size = _ref2.size;
  return {
    padding: size === 'small' ? "0 ".concat(rem(4)) : "0 ".concat(rem(8)),
    fontSize: size === 'small' ? rem(8) : rem(10),
    lineHeight: size === 'small' ? rem(12) : rem(16)
  };
},
/* If a color or backgroundColor is set, override the modifier styles. */
function (_ref3) {
  var color = _ref3.color,
      backgroundColor = _ref3.backgroundColor,
      theme = _ref3.theme;
  return {
    color: color ? themed('color', color)(theme) : theme.color.goten[100],
    backgroundColor: themed('color', backgroundColor)(theme)
  };
});
StyledBadge.defaultProps = {
  color: 'goten.100',
  backgroundColor: 'piccolo.100'
};
var IconRightWrapper = styled.span.withConfig({
  displayName: "Badge__IconRightWrapper",
  componentId: "fdg9gi-1"
})(["margin-left:", ";"], rem(8));
var IconLeftWrapper = styled.span.withConfig({
  displayName: "Badge__IconLeftWrapper",
  componentId: "fdg9gi-2"
})(["margin-right:", ";"], rem(8));

var Badge = function Badge(_ref4) {
  var children = _ref4.children,
      color = _ref4.color,
      backgroundColor = _ref4.backgroundColor,
      iconLeft = _ref4.iconLeft,
      iconRight = _ref4.iconRight,
      size = _ref4.size;
  return /*#__PURE__*/React.createElement(StyledBadge, {
    backgroundColor: backgroundColor,
    color: color,
    iconLeft: iconLeft,
    iconRight: iconRight,
    size: size
  }, iconLeft && /*#__PURE__*/React.createElement(IconLeftWrapper, {
    size: size
  }, iconLeft), children, iconRight && /*#__PURE__*/React.createElement(IconRightWrapper, {
    size: size
  }, iconRight));
};
/*
default
color - goten
backgroundColor - piccolo.100
*/


export default Badge;
