// @heathmont/moon-assets/logos/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Drop: true,
  LogoAposta10Full: true,
  LogoAposta10Short: true,
  LogoBetaddaFull: true,
  LogoBetaddaShort: true,
  LogoBitcasinoFull: true,
  LogoBitcasinoShort: true,
  LogoBombayFull: true,
  LogoBombayShort: true,
  LogoBtcxe: true,
  LogoCommsFull: true,
  LogoCommsShort: true,
  LogoDrop: true,
  LogoHub88: true,
  LogoHub88Full: true,
  LogoHub88Short: true,
  LogoLabFull: true,
  LogoLivecasinoFull: true,
  LogoLivecasinoShort: true,
  LogoLuckyslotsFull: true,
  LogoLuckyslotsShort: true,
  LogoMissions: true,
  LogoMoneyball: true,
  LogoMoonDesign: true,
  LogoMoonDesignShort: true,
  LogoPoweredByHub88: true,
  LogoSbIo: true,
  LogoSlotsFull: true,
  LogoSlotsShort: true,
  LogoSportsbet: true,
  LogoSportsbetCanada: true,
  LogoSportsbetFull: true,
  LogoSportsbetShort: true,
  LogoSportsbetSponsor: true,
  LogoUtorg: true,
  MoonPay: true
};
Object.defineProperty(exports, "Drop", {
  enumerable: true,
  get: function get() {
    return _Drop["default"];
  }
});
Object.defineProperty(exports, "LogoAposta10Full", {
  enumerable: true,
  get: function get() {
    return _LogoAposta10Full["default"];
  }
});
Object.defineProperty(exports, "LogoAposta10Short", {
  enumerable: true,
  get: function get() {
    return _LogoAposta10Short["default"];
  }
});
Object.defineProperty(exports, "LogoBetaddaFull", {
  enumerable: true,
  get: function get() {
    return _LogoBetaddaFull["default"];
  }
});
Object.defineProperty(exports, "LogoBetaddaShort", {
  enumerable: true,
  get: function get() {
    return _LogoBetaddaShort["default"];
  }
});
Object.defineProperty(exports, "LogoBitcasinoFull", {
  enumerable: true,
  get: function get() {
    return _LogoBitcasinoFull["default"];
  }
});
Object.defineProperty(exports, "LogoBitcasinoShort", {
  enumerable: true,
  get: function get() {
    return _LogoBitcasinoShort["default"];
  }
});
Object.defineProperty(exports, "LogoBombayFull", {
  enumerable: true,
  get: function get() {
    return _LogoBombayFull["default"];
  }
});
Object.defineProperty(exports, "LogoBombayShort", {
  enumerable: true,
  get: function get() {
    return _LogoBombayShort["default"];
  }
});
Object.defineProperty(exports, "LogoBtcxe", {
  enumerable: true,
  get: function get() {
    return _LogoBtcxe["default"];
  }
});
Object.defineProperty(exports, "LogoCommsFull", {
  enumerable: true,
  get: function get() {
    return _LogoCommsFull["default"];
  }
});
Object.defineProperty(exports, "LogoCommsShort", {
  enumerable: true,
  get: function get() {
    return _LogoCommsShort["default"];
  }
});
Object.defineProperty(exports, "LogoDrop", {
  enumerable: true,
  get: function get() {
    return _LogoDrop["default"];
  }
});
Object.defineProperty(exports, "LogoHub88", {
  enumerable: true,
  get: function get() {
    return _LogoHub["default"];
  }
});
Object.defineProperty(exports, "LogoHub88Full", {
  enumerable: true,
  get: function get() {
    return _LogoHub88Full["default"];
  }
});
Object.defineProperty(exports, "LogoHub88Short", {
  enumerable: true,
  get: function get() {
    return _LogoHub88Short["default"];
  }
});
Object.defineProperty(exports, "LogoLabFull", {
  enumerable: true,
  get: function get() {
    return _LogoLabFull["default"];
  }
});
Object.defineProperty(exports, "LogoLivecasinoFull", {
  enumerable: true,
  get: function get() {
    return _LogoLivecasinoFull["default"];
  }
});
Object.defineProperty(exports, "LogoLivecasinoShort", {
  enumerable: true,
  get: function get() {
    return _LogoLivecasinoShort["default"];
  }
});
Object.defineProperty(exports, "LogoLuckyslotsFull", {
  enumerable: true,
  get: function get() {
    return _LogoLuckyslotsFull["default"];
  }
});
Object.defineProperty(exports, "LogoLuckyslotsShort", {
  enumerable: true,
  get: function get() {
    return _LogoLuckyslotsShort["default"];
  }
});
Object.defineProperty(exports, "LogoMissions", {
  enumerable: true,
  get: function get() {
    return _LogoMissions["default"];
  }
});
Object.defineProperty(exports, "LogoMoneyball", {
  enumerable: true,
  get: function get() {
    return _LogoMoneyball["default"];
  }
});
Object.defineProperty(exports, "LogoMoonDesign", {
  enumerable: true,
  get: function get() {
    return _LogoMoonDesign["default"];
  }
});
Object.defineProperty(exports, "LogoMoonDesignShort", {
  enumerable: true,
  get: function get() {
    return _LogoMoonDesignShort["default"];
  }
});
Object.defineProperty(exports, "LogoPoweredByHub88", {
  enumerable: true,
  get: function get() {
    return _LogoPoweredByHub["default"];
  }
});
Object.defineProperty(exports, "LogoSbIo", {
  enumerable: true,
  get: function get() {
    return _LogoSbIo["default"];
  }
});
Object.defineProperty(exports, "LogoSlotsFull", {
  enumerable: true,
  get: function get() {
    return _LogoSlotsFull["default"];
  }
});
Object.defineProperty(exports, "LogoSlotsShort", {
  enumerable: true,
  get: function get() {
    return _LogoSlotsShort["default"];
  }
});
Object.defineProperty(exports, "LogoSportsbet", {
  enumerable: true,
  get: function get() {
    return _LogoSportsbet["default"];
  }
});
Object.defineProperty(exports, "LogoSportsbetCanada", {
  enumerable: true,
  get: function get() {
    return _LogoSportsbetCanada["default"];
  }
});
Object.defineProperty(exports, "LogoSportsbetFull", {
  enumerable: true,
  get: function get() {
    return _LogoSportsbetFull["default"];
  }
});
Object.defineProperty(exports, "LogoSportsbetShort", {
  enumerable: true,
  get: function get() {
    return _LogoSportsbetShort["default"];
  }
});
Object.defineProperty(exports, "LogoSportsbetSponsor", {
  enumerable: true,
  get: function get() {
    return _LogoSportsbetSponsor["default"];
  }
});
Object.defineProperty(exports, "LogoUtorg", {
  enumerable: true,
  get: function get() {
    return _LogoUtorg["default"];
  }
});
Object.defineProperty(exports, "MoonPay", {
  enumerable: true,
  get: function get() {
    return _MoonPay["default"];
  }
});

var _Drop = _interopRequireWildcard(require("./Drop"));

Object.keys(_Drop).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Drop[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Drop[key];
    }
  });
});

var _LogoAposta10Full = _interopRequireWildcard(require("./LogoAposta10Full"));

Object.keys(_LogoAposta10Full).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoAposta10Full[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoAposta10Full[key];
    }
  });
});

var _LogoAposta10Short = _interopRequireWildcard(require("./LogoAposta10Short"));

Object.keys(_LogoAposta10Short).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoAposta10Short[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoAposta10Short[key];
    }
  });
});

var _LogoBetaddaFull = _interopRequireWildcard(require("./LogoBetaddaFull"));

Object.keys(_LogoBetaddaFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoBetaddaFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoBetaddaFull[key];
    }
  });
});

var _LogoBetaddaShort = _interopRequireWildcard(require("./LogoBetaddaShort"));

Object.keys(_LogoBetaddaShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoBetaddaShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoBetaddaShort[key];
    }
  });
});

var _LogoBitcasinoFull = _interopRequireWildcard(require("./LogoBitcasinoFull"));

Object.keys(_LogoBitcasinoFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoBitcasinoFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoBitcasinoFull[key];
    }
  });
});

var _LogoBitcasinoShort = _interopRequireWildcard(require("./LogoBitcasinoShort"));

Object.keys(_LogoBitcasinoShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoBitcasinoShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoBitcasinoShort[key];
    }
  });
});

var _LogoBombayFull = _interopRequireWildcard(require("./LogoBombayFull"));

Object.keys(_LogoBombayFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoBombayFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoBombayFull[key];
    }
  });
});

var _LogoBombayShort = _interopRequireWildcard(require("./LogoBombayShort"));

Object.keys(_LogoBombayShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoBombayShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoBombayShort[key];
    }
  });
});

var _LogoBtcxe = _interopRequireWildcard(require("./LogoBtcxe"));

Object.keys(_LogoBtcxe).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoBtcxe[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoBtcxe[key];
    }
  });
});

var _LogoCommsFull = _interopRequireWildcard(require("./LogoCommsFull"));

Object.keys(_LogoCommsFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoCommsFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoCommsFull[key];
    }
  });
});

var _LogoCommsShort = _interopRequireWildcard(require("./LogoCommsShort"));

Object.keys(_LogoCommsShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoCommsShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoCommsShort[key];
    }
  });
});

var _LogoDrop = _interopRequireWildcard(require("./LogoDrop"));

Object.keys(_LogoDrop).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoDrop[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoDrop[key];
    }
  });
});

var _LogoHub = _interopRequireWildcard(require("./LogoHub88"));

Object.keys(_LogoHub).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoHub[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoHub[key];
    }
  });
});

var _LogoHub88Full = _interopRequireWildcard(require("./LogoHub88Full"));

Object.keys(_LogoHub88Full).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoHub88Full[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoHub88Full[key];
    }
  });
});

var _LogoHub88Short = _interopRequireWildcard(require("./LogoHub88Short"));

Object.keys(_LogoHub88Short).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoHub88Short[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoHub88Short[key];
    }
  });
});

var _LogoLabFull = _interopRequireWildcard(require("./LogoLabFull"));

Object.keys(_LogoLabFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoLabFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoLabFull[key];
    }
  });
});

var _LogoLivecasinoFull = _interopRequireWildcard(require("./LogoLivecasinoFull"));

Object.keys(_LogoLivecasinoFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoLivecasinoFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoLivecasinoFull[key];
    }
  });
});

var _LogoLivecasinoShort = _interopRequireWildcard(require("./LogoLivecasinoShort"));

Object.keys(_LogoLivecasinoShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoLivecasinoShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoLivecasinoShort[key];
    }
  });
});

var _LogoLuckyslotsFull = _interopRequireWildcard(require("./LogoLuckyslotsFull"));

Object.keys(_LogoLuckyslotsFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoLuckyslotsFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoLuckyslotsFull[key];
    }
  });
});

var _LogoLuckyslotsShort = _interopRequireWildcard(require("./LogoLuckyslotsShort"));

Object.keys(_LogoLuckyslotsShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoLuckyslotsShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoLuckyslotsShort[key];
    }
  });
});

var _LogoMissions = _interopRequireWildcard(require("./LogoMissions"));

Object.keys(_LogoMissions).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoMissions[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoMissions[key];
    }
  });
});

var _LogoMoneyball = _interopRequireWildcard(require("./LogoMoneyball"));

Object.keys(_LogoMoneyball).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoMoneyball[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoMoneyball[key];
    }
  });
});

var _LogoMoonDesign = _interopRequireWildcard(require("./LogoMoonDesign"));

Object.keys(_LogoMoonDesign).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoMoonDesign[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoMoonDesign[key];
    }
  });
});

var _LogoMoonDesignShort = _interopRequireWildcard(require("./LogoMoonDesignShort"));

Object.keys(_LogoMoonDesignShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoMoonDesignShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoMoonDesignShort[key];
    }
  });
});

var _LogoPoweredByHub = _interopRequireWildcard(require("./LogoPoweredByHub88"));

Object.keys(_LogoPoweredByHub).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoPoweredByHub[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoPoweredByHub[key];
    }
  });
});

var _LogoSbIo = _interopRequireWildcard(require("./LogoSbIo"));

Object.keys(_LogoSbIo).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSbIo[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSbIo[key];
    }
  });
});

var _LogoSlotsFull = _interopRequireWildcard(require("./LogoSlotsFull"));

Object.keys(_LogoSlotsFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSlotsFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSlotsFull[key];
    }
  });
});

var _LogoSlotsShort = _interopRequireWildcard(require("./LogoSlotsShort"));

Object.keys(_LogoSlotsShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSlotsShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSlotsShort[key];
    }
  });
});

var _LogoSportsbet = _interopRequireWildcard(require("./LogoSportsbet"));

Object.keys(_LogoSportsbet).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSportsbet[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSportsbet[key];
    }
  });
});

var _LogoSportsbetCanada = _interopRequireWildcard(require("./LogoSportsbetCanada"));

Object.keys(_LogoSportsbetCanada).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSportsbetCanada[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSportsbetCanada[key];
    }
  });
});

var _LogoSportsbetFull = _interopRequireWildcard(require("./LogoSportsbetFull"));

Object.keys(_LogoSportsbetFull).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSportsbetFull[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSportsbetFull[key];
    }
  });
});

var _LogoSportsbetShort = _interopRequireWildcard(require("./LogoSportsbetShort"));

Object.keys(_LogoSportsbetShort).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSportsbetShort[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSportsbetShort[key];
    }
  });
});

var _LogoSportsbetSponsor = _interopRequireWildcard(require("./LogoSportsbetSponsor"));

Object.keys(_LogoSportsbetSponsor).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoSportsbetSponsor[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoSportsbetSponsor[key];
    }
  });
});

var _LogoUtorg = _interopRequireWildcard(require("./LogoUtorg"));

Object.keys(_LogoUtorg).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _LogoUtorg[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _LogoUtorg[key];
    }
  });
});

var _MoonPay = _interopRequireWildcard(require("./MoonPay"));

Object.keys(_MoonPay).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _MoonPay[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _MoonPay[key];
    }
  });
});

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
