// @heathmont/moon-assets/logos/LogoCommsShort.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Svg = function Svg(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", _extends({
    width: "1em",
    height: "1em",
    viewBox: "0 0 36 36",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/_react["default"].createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M10.6453 21C4.76607 21 0 16.299 0 10.5C0 4.70101 4.76607 0 10.6453 0C16.5246 0 21.2906 4.70101 21.2906 10.5C21.2906 16.299 16.5246 21 10.6453 21Z",
    fill: "#FF4E64"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M10.6453 36C4.76607 36 0 31.299 0 25.5C0 19.701 4.76607 15 10.6453 15C16.5246 15 21.2906 19.701 21.2906 25.5C21.2906 31.299 16.5246 36 10.6453 36Z",
    fill: "#2896FE"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M25.2816 21C19.4726 21 14.7635 16.299 14.7635 10.5C14.7635 4.70101 19.4726 0 25.2816 0C31.0905 0 35.7996 4.70101 35.7996 10.5C35.7996 16.299 31.0905 21 25.2816 21Z",
    fill: "#885ED5"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M25.2816 36C19.4726 36 14.7635 31.299 14.7635 25.5C14.7635 19.701 19.4726 15 25.2816 15C31.0905 15 35.7996 19.701 35.7996 25.5C35.7996 31.299 31.0905 36 25.2816 36Z",
    fill: "#FBD3E0"
  }));
};

var LogoCommsShort = (0, _styledComponents["default"])(Svg).withConfig({
  displayName: "LogoCommsShort",
  componentId: "sc-19aoej3-0"
})(function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      width = _ref.width,
      fontSize = _ref.fontSize,
      theme = _ref.theme;
  return _objectSpread(_objectSpread({}, color && {
    color: (0, _moonUtils.themed)('color', color)(theme)
  }), {}, {
    height: height,
    width: width,
    fontSize: fontSize
  });
});
LogoCommsShort.defaultProps = {};
var _default = LogoCommsShort;
exports["default"] = _default;
