// @heathmont/moon-assets/logos/LogoMoonDesign.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Svg = function Svg(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", _extends({
    width: "1em",
    height: "100%",
    viewBox: "0 0 89 20",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/_react["default"].createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M5.89483 19.7416V7.84111C6.64564 6.4521 8.29744 5.36341 9.83658 5.36341C11.9013 5.36341 12.9525 6.6398 12.9525 9.23011V19.7416H18.2458V7.91618C18.9591 6.4521 20.5358 5.36341 22.15 5.36341C24.2148 5.36341 25.2659 6.6398 25.2659 9.23011V19.7416H30.5592V8.02881C30.5592 3.11095 27.8938 0.52063 23.877 0.52063C21.174 0.52063 18.6588 1.87211 17.4199 3.71161C16.6315 1.94718 14.4166 0.52063 11.6011 0.52063C9.19842 0.52063 7.28384 1.45915 5.89483 3.29866V1.00866H0.601562V19.7416H5.89483ZM74.5653 7.80356C75.4663 6.41455 77.1556 5.36341 78.8074 5.36341C80.7596 5.36341 82.3738 6.4521 82.3738 9.34274V19.7416H87.6671V8.25406C87.6671 3.44882 85.1894 0.52063 80.8347 0.52063C78.3194 0.52063 76.2547 1.42161 74.5653 3.26111V1.00866H69.2721V19.7416H74.5653V7.80356ZM54.1091 10.1659C54.1091 15.4544 49.822 19.7416 44.5336 19.7416C39.2451 19.7416 34.958 15.4544 34.958 10.1659C34.958 4.87752 39.2451 0.590381 44.5336 0.590381C49.822 0.590381 54.1091 4.87752 54.1091 10.1659ZM60.5265 7.08C62.3185 7.08 63.7712 5.62725 63.7712 3.83519C63.7712 2.04313 62.3185 0.590381 60.5265 0.590381C58.7344 0.590381 57.2816 2.04313 57.2816 3.83519C57.2816 5.62725 58.7344 7.08 60.5265 7.08Z",
    fill: "currentColor"
  }));
};

var LogoMoonDesign = (0, _styledComponents["default"])(Svg).withConfig({
  displayName: "LogoMoonDesign",
  componentId: "sc-18ikbq3-0"
})(function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      width = _ref.width,
      fontSize = _ref.fontSize,
      theme = _ref.theme;
  return _objectSpread(_objectSpread({}, color && {
    color: (0, _moonUtils.themed)('color', color)(theme)
  }), {}, {
    height: height,
    width: width,
    fontSize: fontSize
  });
});
LogoMoonDesign.defaultProps = {};
var _default = LogoMoonDesign;
exports["default"] = _default;
