// @heathmont/moon-components/lib/button/size.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.buttonSize = void 0;

var _moonUtils = require("@heathmont/moon-utils");

/**
 * buttonSize
 *
 * Styles that extend `buttonBase` to change the button's cosmetic appearance.
 * Accessed via `buttonSize(key)(theme)`.
 */
var buttonSize = function buttonSize(key) {
  return function (variant) {
    var themedStyles = {
      xsmall: {
        fontSize: (0, _moonUtils.rem)(12),
        lineHeight: (0, _moonUtils.rem)(16),
        height: variant === 'default' ? '' : (0, _moonUtils.rem)(32),
        padding: variant === 'default' ? 0 : "0 ".concat((0, _moonUtils.rem)(16))
      },
      small: {
        fontSize: (0, _moonUtils.rem)(14),
        lineHeight: (0, _moonUtils.rem)(20),
        height: variant === 'default' ? '' : (0, _moonUtils.rem)(40),
        padding: variant === 'default' ? 0 : "0 ".concat((0, _moonUtils.rem)(20))
      },
      medium: {
        fontSize: (0, _moonUtils.rem)(16),
        lineHeight: (0, _moonUtils.rem)(24),
        height: variant === 'default' ? '' : (0, _moonUtils.rem)(48),
        padding: variant === 'default' ? 0 : "0 ".concat((0, _moonUtils.rem)(24))
      },
      large: {
        fontSize: (0, _moonUtils.rem)(18),
        lineHeight: (0, _moonUtils.rem)(24),
        height: variant === 'default' ? '' : (0, _moonUtils.rem)(56),
        padding: variant === 'default' ? 0 : "0 ".concat((0, _moonUtils.rem)(32))
      }
    };
    return themedStyles[key];
  };
};

exports.buttonSize = buttonSize;
var _default = {
  buttonSize: buttonSize
};
exports["default"] = _default;
