// @heathmont/moon-assets/es/logos/LogoMoonDesignShort.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { themed } from '@heathmont/moon-utils';

var Svg = function Svg(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/React.createElement("path", {
    d: "M8.64668 19.2933C12.3175 19.2933 15.2933 16.3175 15.2933 12.6466C15.2933 8.97581 12.3175 6 8.64668 6C4.97582 6 2 8.97581 2 12.6466C2 16.3175 4.97582 19.2933 8.64668 19.2933Z",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M22 8.2523C22 9.49622 20.9916 10.5046 19.7477 10.5046C18.5038 10.5046 17.4954 9.49622 17.4954 8.2523C17.4954 7.00839 18.5038 6 19.7477 6C20.9916 6 22 7.00839 22 8.2523Z",
    fill: "currentColor"
  }));
};

var LogoMoonDesignShort = styled(Svg).withConfig({
  displayName: "LogoMoonDesignShort",
  componentId: "sc-9vcfqk-0"
})(function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      width = _ref.width,
      fontSize = _ref.fontSize,
      theme = _ref.theme;
  return _objectSpread(_objectSpread({}, color && {
    color: themed('color', color)(theme)
  }), {}, {
    height: height,
    width: width,
    fontSize: fontSize
  });
});
LogoMoonDesignShort.defaultProps = {};
export default LogoMoonDesignShort;
