// @heathmont/moon-assets/es/logos/LogoHub88Short.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { themed } from '@heathmont/moon-utils';

var Svg = function Svg(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: "1em",
    height: "100%",
    viewBox: "0 0 42 41",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/React.createElement("g", {
    fill: "none",
    fillRule: "evenodd",
    opacity: 0.9
  }, /*#__PURE__*/React.createElement("path", {
    d: "M15.547 0C22.444.026 28.017 5.634 28 12.531 27.983 19.43 22.382 25.01 15.484 25 8.587 24.991 3 19.398 3 12.5 3.013 5.584 8.63-.013 15.547 0z",
    fill: "#06BEE1"
  }), /*#__PURE__*/React.createElement("circle", {
    fill: "#1768AC",
    cx: 22.5,
    cy: 27.5,
    r: 12.5
  }), /*#__PURE__*/React.createElement("ellipse", {
    fill: "#2541B2",
    cx: 10,
    cy: 31.5,
    rx: 10,
    ry: 9.5
  }), /*#__PURE__*/React.createElement("path", {
    d: "M35.024 19A7 7 0 1128 26a7.012 7.012 0 017.024-7z",
    fill: "#03256C"
  })));
};

var LogoHub88Short = styled(Svg).withConfig({
  displayName: "LogoHub88Short",
  componentId: "sc-1v78g7x-0"
})(function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      width = _ref.width,
      fontSize = _ref.fontSize,
      theme = _ref.theme;
  return _objectSpread(_objectSpread({}, color && {
    color: themed('color', color)(theme)
  }), {}, {
    height: height,
    width: width,
    fontSize: fontSize
  });
});
LogoHub88Short.defaultProps = {};
export default LogoHub88Short;
