// @heathmont/moon-assets/es/logos/LogoMissions.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { themed } from '@heathmont/moon-utils';

var Svg = function Svg(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: "1em",
    height: "1em",
    viewBox: "0 0 40 40",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/React.createElement("ellipse", {
    cx: 27.1917,
    cy: 21.3057,
    rx: 10.3709,
    ry: 10.3709,
    transform: "rotate(-15 27.1917 21.3057)",
    fill: "#FF385C"
  }), /*#__PURE__*/React.createElement("ellipse", {
    cx: 13.7017,
    cy: 26.8477,
    rx: 10.3709,
    ry: 10.3709,
    transform: "rotate(-15 13.7017 26.8477)",
    fill: "#6A36CA"
  }), /*#__PURE__*/React.createElement("circle", {
    cx: 15.8012,
    cy: 13.7017,
    r: 10.3709,
    transform: "rotate(-15 15.8012 13.7017)",
    fill: "#FFB319"
  }));
};

var LogoMissions = styled(Svg).withConfig({
  displayName: "LogoMissions",
  componentId: "uvtso4-0"
})(function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      width = _ref.width,
      fontSize = _ref.fontSize,
      theme = _ref.theme;
  return _objectSpread(_objectSpread({}, color && {
    color: themed('color', color)(theme)
  }), {}, {
    height: height,
    width: width,
    fontSize: fontSize
  });
});
LogoMissions.defaultProps = {};
export default LogoMissions;
