// @heathmont/moon-assets/es/patterns/PatternSpeed.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { themed } from '@heathmont/moon-utils';

var Svg = function Svg(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: "1em",
    height: "1em",
    viewBox: "0 0 16 16",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/React.createElement("rect", {
    width: 0.650036,
    height: 13.031,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 14.491 6.62866)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.25,
    width: 0.650036,
    height: 13.6139,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 15.2281 9.32983)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.75,
    width: 0.650036,
    height: 12.3069,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 15.7269 10.6319)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.5,
    width: 0.650036,
    height: 14.4873,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 14.6338 12.3535)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.75,
    width: 0.650036,
    height: 7.91358,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 16 13.5236)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.5,
    width: 0.650036,
    height: 10.844,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 10.7341 6.15756)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.75,
    width: 0.650036,
    height: 14.5662,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 14.8522 3.62259)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.5,
    width: 0.650036,
    height: 9.96036,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 15.3185 2.03079)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.5,
    width: 0.650036,
    height: 12.5827,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 12.3506 8.61633)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.25,
    width: 0.650036,
    height: 13.2746,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 13.0864 1.2554)",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    opacity: 0.5,
    width: 0.650036,
    height: 9.18365,
    rx: 0.325018,
    transform: "matrix(-0.266106 -0.963944 -0.967804 0.251703 9.96375 0.626587)",
    fill: "currentColor"
  }));
};

var PatternSpeed = styled(Svg).withConfig({
  displayName: "PatternSpeed",
  componentId: "sc-9gbb7o-0"
})(function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      width = _ref.width,
      fontSize = _ref.fontSize,
      theme = _ref.theme;
  return _objectSpread(_objectSpread({}, color && {
    color: themed('color', color)(theme)
  }), {}, {
    height: height,
    width: width,
    fontSize: fontSize
  });
});
PatternSpeed.defaultProps = {
  color: "piccolo.100"
};
export default PatternSpeed;
