// @heathmont/moon-components/lib/es/progressLinear/ProgressLinear.js
import React from 'react';
import styled from 'styled-components';
import { rem, themed } from '@heathmont/moon-utils';
var ProgressBarWrapper = styled.div.withConfig({
  displayName: "ProgressLinear__ProgressBarWrapper",
  componentId: "sc-1wwz7u1-0"
})({
  position: 'relative'
});
var CurrentProgress = styled.div.withConfig({
  displayName: "ProgressLinear__CurrentProgress",
  componentId: "sc-1wwz7u1-1"
})(function (_ref) {
  var _ref$lineHeight = _ref.lineHeight,
      lineHeight = _ref$lineHeight === void 0 ? 12 : _ref$lineHeight,
      _ref$fontSize = _ref.fontSize,
      fontSize = _ref$fontSize === void 0 ? 10 : _ref$fontSize,
      _ref$theme = _ref.theme,
      fontWeight = _ref$theme.fontWeight,
      color = _ref$theme.color;
  return {
    position: 'absolute',
    top: 0,
    left: '50%',
    transform: 'translateX(-50%)',
    color: color.bulma[100],
    fontSize: rem(fontSize),
    lineHeight: rem(lineHeight),
    letterSpacing: 1,
    fontWeight: fontWeight.semibold,
    textTransform: 'uppercase',
    whiteSpace: 'nowrap',
    textShadow: "-".concat(rem(1), " 0 ").concat(color.gohan['100'], ", 0 ").concat(rem(1), " ").concat(color.gohan['100'], ", ").concat(rem(1), " 0 ").concat(color.gohan['100'], ", 0 -").concat(rem(1), " ").concat(color.gohan['100'])
  };
});
var Progress = styled.div.withConfig({
  displayName: "ProgressLinear__Progress",
  componentId: "sc-1wwz7u1-2"
})(function (_ref2) {
  var theme = _ref2.theme,
      value = _ref2.value,
      backgroundColor = _ref2.backgroundColor,
      _ref2$height = _ref2.height,
      height = _ref2$height === void 0 ? 12 : _ref2$height;
  return {
    minWidth: rem(100),
    backgroundColor: themed('color', backgroundColor)(theme) || theme.color.goku['100'],
    height: rem(height),
    position: 'relative',
    borderRadius: rem(60),
    '&::before': {
      content: '""',
      position: 'absolute',
      backgroundColor: theme.color.piccolo['100'],
      height: '100%',
      width: "".concat(value, "%"),
      borderRadius: rem(60)
    }
  };
});

var ProgressLinear = function ProgressLinear(_ref3) {
  var value = _ref3.value,
      text = _ref3.text,
      backgroundColor = _ref3.backgroundColor,
      height = _ref3.height,
      fontSize = _ref3.fontSize;
  return /*#__PURE__*/React.createElement(ProgressBarWrapper, null, /*#__PURE__*/React.createElement(Progress, {
    height: height,
    value: value,
    backgroundColor: backgroundColor
  }), /*#__PURE__*/React.createElement(CurrentProgress, {
    lineHeight: height,
    fontSize: fontSize
  }, text));
};

export default ProgressLinear;
