// @heathmont/moon-components/lib/es/dialog/Dialog.js
import React from 'react';
import IconClose from '../private/icons/IconClose';
import DialogOverlay from '../dialogOverlay/DialogOverlay';
import DialogContent from '../dialogContent/DialogContent';
import { DialogToggle, DialogToggleText } from './private/toggle';
import { DialogContainer, DialogMain } from './private/layout';

var Dialog = function Dialog(_ref) {
  var children = _ref.children,
      maxWidth = _ref.maxWidth,
      _ref$isOpen = _ref.isOpen,
      isOpen = _ref$isOpen === void 0 ? false : _ref$isOpen,
      onDismiss = _ref.onDismiss,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? 'CENTER' : _ref$position,
      _ref$disableScrollLoc = _ref.disableScrollLock,
      disableScrollLock = _ref$disableScrollLoc === void 0 ? false : _ref$disableScrollLoc;
  return /*#__PURE__*/React.createElement(DialogOverlay, {
    isOpen: isOpen && isOpen,
    onDismiss: onDismiss && onDismiss,
    dangerouslyBypassScrollLock: disableScrollLock
  }, /*#__PURE__*/React.createElement(DialogContent, {
    maxWidth: maxWidth,
    position: position
  }, /*#__PURE__*/React.createElement(DialogContainer, null, /*#__PURE__*/React.createElement(DialogToggle, {
    type: "button",
    onClick: onDismiss && onDismiss
  }, /*#__PURE__*/React.createElement(IconClose, null), /*#__PURE__*/React.createElement(DialogToggleText, null, "Close dialog")), /*#__PURE__*/React.createElement(DialogMain, null, children))));
};

export default Dialog;
