// @heathmont/moon-components/lib/emptyState/EmptyState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "EmptyState__Wrapper",
  componentId: "jz2esn-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      breakpoint = _ref$theme.breakpoint,
      color = _ref$theme.color;
  return _defineProperty({
    borderRadius: (0, _moonUtils.rem)(12),
    backgroundColor: color.gohan[100],
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundRepeat: 'no-repeat',
    backgroundPositionY: '-200px',
    backgroundPositionX: '-300px',
    padding: (0, _moonUtils.rem)(64)
  }, (0, _moonUtils.mq)(breakpoint.large), {
    flexDirection: 'row',
    padding: (0, _moonUtils.rem)(40),
    backgroundPositionX: '160px',
    backgroundPositionY: 'center'
  });
});

var IconWrapper = _styledComponents["default"].div.withConfig({
  displayName: "EmptyState__IconWrapper",
  componentId: "jz2esn-1"
})(function (_ref3) {
  var breakpoint = _ref3.theme.breakpoint;
  return _defineProperty({
    marginBottom: (0, _moonUtils.rem)(40)
  }, (0, _moonUtils.mq)(breakpoint.large), {
    marginBottom: 0
  });
});

var TextWrapper = _styledComponents["default"].div.withConfig({
  displayName: "EmptyState__TextWrapper",
  componentId: "jz2esn-2"
})(function (_ref5) {
  var _ref6;

  var breakpoint = _ref5.theme.breakpoint;
  return _ref6 = {
    textAlign: 'center'
  }, _defineProperty(_ref6, (0, _moonUtils.mq)(breakpoint.large), {
    marginLeft: (0, _moonUtils.rem)(24),
    textAlign: 'left',
    marginTop: 0
  }), _defineProperty(_ref6, "maxWidth", (0, _moonUtils.rem)(400)), _defineProperty(_ref6, (0, _moonUtils.mq)(breakpoint.xlarge), {
    maxWidth: (0, _moonUtils.rem)(900)
  }), _ref6;
});

var TitleWrapper = _styledComponents["default"].p.withConfig({
  displayName: "EmptyState__TitleWrapper",
  componentId: "jz2esn-3"
})(function (_ref7) {
  var fontWeight = _ref7.theme.fontWeight;
  return {
    margin: 0,
    fontWeight: fontWeight.semibold,
    fontSize: (0, _moonUtils.rem)(20),
    lineHeight: (0, _moonUtils.rem)(28)
  };
});

var MessageWrapper = _styledComponents["default"].p.withConfig({
  displayName: "EmptyState__MessageWrapper",
  componentId: "jz2esn-4"
})(function (_ref8) {
  var _ref8$theme = _ref8.theme,
      color = _ref8$theme.color,
      breakpoint = _ref8$theme.breakpoint;
  return _defineProperty({
    margin: 0,
    fontSize: (0, _moonUtils.rem)(16),
    lineHeight: (0, _moonUtils.rem)(24),
    color: color.trunks[100],
    marginTop: (0, _moonUtils.rem)(20)
  }, (0, _moonUtils.mq)(breakpoint.large), {
    marginTop: (0, _moonUtils.rem)(4)
  });
});

var ActionWrapper = _styledComponents["default"].div.withConfig({
  displayName: "EmptyState__ActionWrapper",
  componentId: "jz2esn-5"
})(function (_ref10) {
  var breakpoint = _ref10.theme.breakpoint;
  return _defineProperty({
    marginTop: (0, _moonUtils.rem)(32)
  }, (0, _moonUtils.mq)(breakpoint.large), {
    marginTop: 0,
    marginLeft: 'auto'
  });
});

var EmptyState = function EmptyState(_ref12) {
  var title = _ref12.title,
      message = _ref12.message,
      icon = _ref12.icon,
      action = _ref12.action;
  return /*#__PURE__*/_react["default"].createElement(Wrapper, null, icon && /*#__PURE__*/_react["default"].createElement(IconWrapper, null, icon), (title || message) && /*#__PURE__*/_react["default"].createElement(TextWrapper, null, title && /*#__PURE__*/_react["default"].createElement(TitleWrapper, null, title), message && /*#__PURE__*/_react["default"].createElement(MessageWrapper, null, message)), action && /*#__PURE__*/_react["default"].createElement(ActionWrapper, null, action));
};

var _default = EmptyState;
exports["default"] = _default;
