// @heathmont/moon-components/lib/loader/Loader.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * Animation
 */
var ring = (0, _styledComponents.keyframes)(["0%{transform:rotate(0deg);}100%{transform:rotate(360deg);}"]);
var animation = (0, _styledComponents.css)(["animation:", " 1.2s cubic-bezier(0.5,0,0.5,1) infinite;"], ring);
/**
 * Sub-components
 */

var LoaderContainer = _styledComponents["default"].strong.withConfig({
  displayName: "Loader__LoaderContainer",
  componentId: "vzi49y-0"
})(function (_ref) {
  var space = _ref.theme.space;
  return {
    display: 'inline-block',
    position: 'relative',
    width: (0, _moonUtils.rem)(space["default"]),
    height: (0, _moonUtils.rem)(space["default"])
  };
});

var LoaderRing = _styledComponents["default"].span.withConfig({
  displayName: "Loader__LoaderRing",
  componentId: "vzi49y-1"
})(function (_ref2) {
  var _ref2$theme = _ref2.theme,
      color = _ref2$theme.color,
      borderStyle = _ref2$theme.borderStyle,
      borderWidth = _ref2$theme.borderWidth;
  return {
    '--loader-color': color.bulma[100],
    display: 'block',
    position: 'absolute',
    width: '100%',
    height: '100%',
    borderRadius: '50%',
    boxSizing: 'border-box',
    borderWidth: borderWidth * 2,
    borderStyle: borderStyle,
    borderColor: 'transparent',
    borderTopColor: 'var(--loader-color)'
  };
}, animation, function (_ref3) {
  var animationDelay = _ref3.animationDelay;
  return {
    animationDelay: animationDelay
  };
}, function (_ref4) {
  var color = _ref4.color,
      theme = _ref4.theme;
  return {
    '--loader-color': (0, _moonUtils.themed)('color', color)(theme)
  };
});
/**
 * Component
 */


var Loader = function Loader(_ref5) {
  var color = _ref5.color;
  return /*#__PURE__*/_react["default"].createElement(LoaderContainer, {
    role: "progressbar"
  }, ['-0.45s', '-0.3s', '-0.15s', '0s'].map(function (delay) {
    return /*#__PURE__*/_react["default"].createElement(LoaderRing, {
      key: "loader-".concat(delay),
      color: color,
      animationDelay: delay
    });
  }));
};

var _default = Loader;
exports["default"] = _default;
