// @heathmont/moon-components/lib/badge/Badge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Badge = _styledComponents["default"].span.withConfig({
  displayName: "Badge",
  componentId: "gd8hvq-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      fontWeight = _ref$theme.fontWeight,
      radius = _ref$theme.radius;
  return {
    display: 'flex',
    fontWeight: fontWeight.semibold,
    textTransform: 'uppercase',
    borderRadius: (0, _moonUtils.rem)(radius.largest),
    letterSpacing: (0, _moonUtils.rem)(1)
  };
}, function (_ref2) {
  var size = _ref2.size;
  return {
    padding: size === 'small' ? "0 ".concat((0, _moonUtils.rem)(8)) : "0 ".concat((0, _moonUtils.rem)(4)),
    fontSize: size === 'small' ? (0, _moonUtils.rem)(10) : (0, _moonUtils.rem)(8),
    lineHeight: size === 'small' ? (0, _moonUtils.rem)(16) : (0, _moonUtils.rem)(12)
  };
},
/* If a color or backgroundColor is set, override styles. */
function (_ref3) {
  var color = _ref3.color,
      backgroundColor = _ref3.backgroundColor,
      theme = _ref3.theme;
  return {
    color: (0, _moonUtils.themed)('color', color)(theme),
    backgroundColor: (0, _moonUtils.themed)('color', backgroundColor)(theme)
  };
});

Badge.defaultProps = {
  size: 'xsmall'
};
var _default = Badge;
exports["default"] = _default;
