// @heathmont/moon-components/lib/es/reel/Reel.js
import styled from 'styled-components';
import { themed, rem } from '@heathmont/moon-utils';
import listItemStyleTypeNone from '../listItemStyleTypeNone/listItemStyleTypeNone';

/**
 * Reel
 *
 * Based on the Reel component from every-layout.dev
 */
var Reel = styled.div.withConfig({
  displayName: "Reel",
  componentId: "sc-10l7fx7-0"
})(function (_ref) {
  var height = _ref.height,
      space = _ref.space,
      theme = _ref.theme,
      dir = _ref.dir;
  return [{
    display: 'flex',
    overflowX: 'auto',
    overflowY: 'hidden',
    height: height,

    /**
     * Scroll Behavior
     * 1. Hide Scrollbars on browsers that don't support custom scrollbars.
     * 2. Auto-hide scrollbars on IE/Edge.
     * 3. Create 'padding' around the scrollbar.
     */
    WebkitOverflowScrolling: 'touch',
    scrollbarWidth: 'none'
    /* [1] */
    ,
    '-ms-overflow-style': '-ms-autohiding-scrollbar'
    /* [2] */
    ,
    '::-webkit-scrollbar': {
      width: 12,
      height: 12,
      cursor: 'pointer'
    },
    '::-webkit-scrollbar-thumb': {
      backgroundColor: 'transparent',
      backgroundClip: 'content-box'
      /* [3] */
      ,
      borderRadius: rem(theme.radius.largest),
      border: '3px solid transparent'
      /* [3] */

    },
    ':hover::-webkit-scrollbar-thumb': {
      backgroundColor: theme.color.goku[40]
    },

    /**
     * Elements
     * 1. Ensure direct child list-items render without bullets
     * 2. Set content width to auto
     * 3. When content is set to images, ensure they all remain the same height.
     */
    '& > li': listItemStyleTypeNone
    /* [1] */
    ,
    '& > *': {
      flex: '0 0 auto'
    }
    /* [2] */
    ,
    '& > img': {
      height: '100%',
      flexBasis: 'auto',
      width: 'auto'
    }
    /* [3] */

  }, dir === 'rtl' ? {
    '& > * + *': {
      marginRight: themed('space', space)(theme)
    }
  } : {
    '& > * + *': {
      marginLeft: themed('space', space)(theme)
    }
  }];
});
Reel.defaultProps = {
  height: 'auto',
  space: 'default'
};
export default Reel;
