// @heathmont/moon-components/lib/es/carousel/control.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import styled from 'styled-components';
import hideVisually from 'polished/lib/mixins/hideVisually';
import { rem, mq } from '@heathmont/moon-utils';
/**
 * 1. Arbitrary number to meet with Figma designs, between small and x-small.
 * 2. Flip the indicator on the right, rather than use another SVG import.
 */

export var CarouselControl = styled.button.withConfig({
  displayName: "control__CarouselControl",
  componentId: "sc-10r03pt-0"
})(function (_ref) {
  var _ref2;

  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      border = _ref$theme.border,
      boxShadow = _ref$theme.boxShadow,
      opacity = _ref$theme.opacity,
      radius = _ref$theme.radius,
      zIndex = _ref$theme.zIndex,
      space = _ref$theme.space,
      breakpoint = _ref$theme.breakpoint;
  return _ref2 = {}, _defineProperty(_ref2, mq(breakpoint.small, 'max-width'), {
    display: 'none'
  }), _defineProperty(_ref2, "padding", rem(space.small)), _defineProperty(_ref2, "verticalAlign", 'middle'), _defineProperty(_ref2, "border", border), _defineProperty(_ref2, "boxShadow", boxShadow), _defineProperty(_ref2, "borderColor", color.beerus[100]), _defineProperty(_ref2, "borderRadius", rem(radius.largest)), _defineProperty(_ref2, "backgroundColor", color.gohan[100]), _defineProperty(_ref2, "color", color.trunks[100]), _defineProperty(_ref2, "fontSize", rem(8)), _defineProperty(_ref2, "lineHeight", 0), _defineProperty(_ref2, "textDecoration", 'none'), _defineProperty(_ref2, "cursor", 'pointer'), _defineProperty(_ref2, "position", 'absolute'), _defineProperty(_ref2, "left", '-14px'), _defineProperty(_ref2, "top", '50%'), _defineProperty(_ref2, "transform", 'translateY(-50%)'), _defineProperty(_ref2, "transformOrigin", 'top center'), _defineProperty(_ref2, "zIndex", zIndex.carouselControl), _defineProperty(_ref2, '&:last-child', {
    left: 'auto',
    right: '-14px',
    svg: {
      transform: 'rotate(180deg) '
      /* [2] */

    }
  }), _defineProperty(_ref2, '&:hover', {
    color: color.piccolo[80]
  }), _defineProperty(_ref2, '&:disabled, &[disabled]', {
    cursor: 'not-allowed',
    opacity: opacity.disabled
  }), _ref2;
});
/**
 * `<any>` justification: https://coingaming.atlassian.net/browse/SPO-4963.
 */

export var CarouselControlCaption = styled.span.withConfig({
  displayName: "control__CarouselControlCaption",
  componentId: "sc-10r03pt-1"
})(hideVisually);
export default CarouselControlCaption;
