// @heathmont/moon-components/lib/es/button/size.js
import { rem } from '@heathmont/moon-utils';

/**
 * buttonSize
 *
 * Styles that extend `buttonBase` to change the button's cosmetic appearance.
 * Accessed via `buttonSize(key)(theme)`.
 */
var buttonSize = function buttonSize(key) {
  return function (variant) {
    var themedStyles = {
      xsmall: {
        fontSize: rem(12),
        lineHeight: rem(16),
        height: variant === 'default' ? '' : rem(32),
        padding: variant === 'default' ? 0 : "0 ".concat(rem(16))
      },
      small: {
        fontSize: rem(14),
        lineHeight: rem(20),
        height: variant === 'default' ? '' : rem(40),
        padding: variant === 'default' ? 0 : "0 ".concat(rem(20))
      },
      medium: {
        fontSize: rem(16),
        lineHeight: rem(24),
        height: variant === 'default' ? '' : rem(48),
        padding: variant === 'default' ? 0 : "0 ".concat(rem(24))
      },
      large: {
        fontSize: rem(18),
        lineHeight: rem(24),
        height: variant === 'default' ? '' : rem(56),
        padding: variant === 'default' ? 0 : "0 ".concat(rem(32))
      }
    };
    return themedStyles[key];
  };
};

export { buttonSize };
export default {
  buttonSize: buttonSize
};
