// @heathmont/moon-themes/lib/sharedTokens/avertaStd.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
// TODO set font URL in the App instead
var cdnUrl = "https://sportsbet.io/sportsbet-io/files";
var fontDirPath = "".concat(cdnUrl, "/fonts/averta-std");
var fontFilePath = {
  regular: "".concat(fontDirPath, "/regular"),
  semibold: "".concat(fontDirPath, "/semibold")
};
var fontFamily = 'Averta Std';
var fontStack = "".concat(fontFamily, ", -apple-system, BlinkMacSystemFont, \"Segoe UI\", Roboto, Oxygen-Sans, Ubuntu, Cantarell, \"Helvetica Neue\", sans-serif");
var fontStyle = 'normal';
var fontDisplay = 'swap';
var fontWeight = {
  normal: 400,
  semibold: 500
};
var unicodeRange = 'U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD';
var fontFace = [{
  '@font-face': {
    fontFamily: fontFamily,
    fontStyle: fontStyle,
    fontDisplay: fontDisplay,
    unicodeRange: unicodeRange,
    fontWeight: fontWeight.normal,
    src: "local(\"AvertaStd-Regular\"), local(\"Averta Std Regular\"), url(".concat(fontFilePath.regular, ".woff2) format(\"woff2\")")
  }
}, {
  '@font-face': {
    fontFamily: fontFamily,
    fontStyle: fontStyle,
    fontDisplay: fontDisplay,
    unicodeRange: unicodeRange,
    fontWeight: fontWeight.semibold,
    src: "local(\"AvertaStd-Semibold\"), local(\"Averta Std Semibold\"), url(".concat(fontFilePath.semibold, ".woff2) format(\"woff2\")")
  }
}];
var avertaStd = {
  fontDirPath: fontDirPath,
  fontFilePath: fontFilePath,
  fontFamily: fontFamily,
  fontStack: fontStack,
  fontStyle: fontStyle,
  fontWeight: fontWeight,
  fontFace: fontFace,
  unicodeRange: unicodeRange
};
var _default = avertaStd;
exports["default"] = _default;
