// @heathmont/moon-components/lib/chip/Chip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var StyledChip = _styledComponents["default"].div.withConfig({
  displayName: "Chip__StyledChip",
  componentId: "sc-1rq31s5-0"
})(["display:flex;flex-direction:row;justify-content:center;align-items:center;border-radius:", ";padding:", ";line-height:", ";font-size:", ";background:", ";color:", ";"], (0, _moonUtils.rem)(8), (0, _moonUtils.rem)(8), function (_ref) {
  var size = _ref.size;
  return size === 'small' ? (0, _moonUtils.rem)(16) : (0, _moonUtils.rem)(20);
}, function (_ref2) {
  var size = _ref2.size;
  return size === 'small' ? (0, _moonUtils.rem)(12) : (0, _moonUtils.rem)(14);
}, function (_ref3) {
  var theme = _ref3.theme,
      active = _ref3.active;
  return active ? theme.color.frieza[10] : theme.color.goten[100];
}, function (_ref4) {
  var theme = _ref4.theme,
      active = _ref4.active;
  return active ? theme.color.frieza[100] : theme.color.bulma[100];
});

var IconRightWrapper = _styledComponents["default"].span.withConfig({
  displayName: "Chip__IconRightWrapper",
  componentId: "sc-1rq31s5-1"
})(["margin-left:", ";"], (0, _moonUtils.rem)(8));

var IconLeftWrapper = _styledComponents["default"].span.withConfig({
  displayName: "Chip__IconLeftWrapper",
  componentId: "sc-1rq31s5-2"
})(["margin-right:", ";"], (0, _moonUtils.rem)(8));

var Chip = function Chip(_ref5) {
  var children = _ref5.children,
      active = _ref5.active,
      size = _ref5.size,
      iconLeft = _ref5.iconLeft,
      iconRight = _ref5.iconRight;
  return /*#__PURE__*/_react["default"].createElement(StyledChip, {
    active: active,
    iconLeft: iconLeft,
    iconRight: iconRight,
    size: size
  }, iconLeft && /*#__PURE__*/_react["default"].createElement(IconLeftWrapper, {
    size: size
  }, iconLeft), children, iconRight && /*#__PURE__*/_react["default"].createElement(IconRightWrapper, {
    size: size
  }, iconRight));
};

var _default = Chip;
exports["default"] = _default;
