// @heathmont/moon-components/lib/header/Header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.HeaderTitle = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/**
 * 1. A tiny boost to the vertical rhythm of succeeding elements.
 */
var Header = _styledComponents["default"].header.withConfig({
  displayName: "Header",
  componentId: "sbaz2c-0"
})(function (_ref) {
  var space = _ref.theme.space;
  return {
    marginTop: (0, _moonUtils.rem)(space["default"]),
    paddingBottom: (0, _moonUtils.rem)(space.small)
    /* [1] */
    ,
    position: 'relative',
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'baseline'
  };
});

var HeaderTitle = _styledComponents["default"].h1.withConfig({
  displayName: "Header__HeaderTitle",
  componentId: "sbaz2c-1"
})(function (_ref2) {
  var _ref2$theme = _ref2.theme,
      breakpoint = _ref2$theme.breakpoint,
      fontWeight = _ref2$theme.fontWeight,
      space = _ref2$theme.space;
  return _defineProperty({
    display: 'inline-block',
    margin: "0 ".concat((0, _moonUtils.rem)(space.large), " 0 0"),
    fontSize: (0, _moonUtils.rem)(20),
    lineHeight: (0, _moonUtils.rem)(24),
    fontWeight: fontWeight.semibold
  }, (0, _moonUtils.mq)(breakpoint.small), {
    fontSize: (0, _moonUtils.rem)(32),
    lineHeight: (0, _moonUtils.rem)(40)
  });
});

exports.HeaderTitle = HeaderTitle;
var _default = Header;
exports["default"] = _default;
