// @heathmont/moon-assets/patterns/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  PatternCircles: true,
  PatternLines: true,
  PatternRipple: true,
  PatternSpeed: true,
  PatternStars: true,
  PatternWave: true
};
Object.defineProperty(exports, "PatternCircles", {
  enumerable: true,
  get: function get() {
    return _PatternCircles["default"];
  }
});
Object.defineProperty(exports, "PatternLines", {
  enumerable: true,
  get: function get() {
    return _PatternLines["default"];
  }
});
Object.defineProperty(exports, "PatternRipple", {
  enumerable: true,
  get: function get() {
    return _PatternRipple["default"];
  }
});
Object.defineProperty(exports, "PatternSpeed", {
  enumerable: true,
  get: function get() {
    return _PatternSpeed["default"];
  }
});
Object.defineProperty(exports, "PatternStars", {
  enumerable: true,
  get: function get() {
    return _PatternStars["default"];
  }
});
Object.defineProperty(exports, "PatternWave", {
  enumerable: true,
  get: function get() {
    return _PatternWave["default"];
  }
});

var _PatternCircles = _interopRequireWildcard(require("./PatternCircles"));

Object.keys(_PatternCircles).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PatternCircles[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _PatternCircles[key];
    }
  });
});

var _PatternLines = _interopRequireWildcard(require("./PatternLines"));

Object.keys(_PatternLines).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PatternLines[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _PatternLines[key];
    }
  });
});

var _PatternRipple = _interopRequireWildcard(require("./PatternRipple"));

Object.keys(_PatternRipple).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PatternRipple[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _PatternRipple[key];
    }
  });
});

var _PatternSpeed = _interopRequireWildcard(require("./PatternSpeed"));

Object.keys(_PatternSpeed).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PatternSpeed[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _PatternSpeed[key];
    }
  });
});

var _PatternStars = _interopRequireWildcard(require("./PatternStars"));

Object.keys(_PatternStars).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PatternStars[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _PatternStars[key];
    }
  });
});

var _PatternWave = _interopRequireWildcard(require("./PatternWave"));

Object.keys(_PatternWave).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PatternWave[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _PatternWave[key];
    }
  });
});

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
