// @heathmont/moon-assets/patterns/PatternWave.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Svg = function Svg(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", _extends({
    width: "1em",
    height: "100%",
    viewBox: "0 0 215 75",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/_react["default"].createElement("rect", {
    x: 105.003,
    y: 70.0003,
    width: 5.00003,
    height: 50.0002,
    rx: 2.50001,
    transform: "rotate(-180 105.003 70.0003)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.6,
    x: 95.0012,
    y: 65.0003,
    width: 5.00003,
    height: 55.0002,
    rx: 2.50001,
    transform: "rotate(-180 95.0012 65.0003)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.4,
    width: 5.00001,
    height: 40.0001,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 85.0032 14.9999)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.2,
    width: 5.00001,
    height: 60.0003,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 75.0017 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.4,
    width: 5.00001,
    height: 45.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 64.9992 24.9998)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.2,
    width: 5.00001,
    height: 50.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 55.001 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.6,
    width: 5.00001,
    height: 55.0002,
    rx: 2.5,
    transform: "matrix(-1 -4.89277e-08 -1.78445e-07 1 44.9993 14.9999)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.3,
    width: 5.00001,
    height: 65.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 35.0012 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.2,
    width: 5.00001,
    height: 55.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 24.9996 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.4,
    width: 5.00001,
    height: 55.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 15.0015 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.6,
    width: 5.00001,
    height: 45.0002,
    rx: 2.5,
    transform: "matrix(-1 -4.89277e-08 -1.78445e-07 1 5 19.9998)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    x: 215,
    y: 70.0003,
    width: 5.00003,
    height: 50.0002,
    rx: 2.50001,
    transform: "rotate(-180 215 70.0003)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.6,
    x: 205.002,
    y: 62.3786,
    width: 5.00002,
    height: 57.3789,
    rx: 2.50001,
    transform: "rotate(-180 205.002 62.3786)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.4,
    width: 5.00001,
    height: 45.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 195 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.2,
    width: 5.00001,
    height: 60.0003,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 185.002 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.4,
    width: 5.00001,
    height: 60.0003,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 175.001 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.2,
    width: 5.00001,
    height: 60.0003,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 165.002 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.6,
    width: 5.00001,
    height: 70.0003,
    rx: 2.5,
    transform: "matrix(-1 -2.58096e-08 -1.5885e-07 1 155.001 0)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.3,
    width: 5.00001,
    height: 55.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 145.002 14.9999)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.2,
    width: 5.00001,
    height: 55.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 135.001 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.4,
    width: 5.00001,
    height: 55.0002,
    rx: 2.50001,
    transform: "matrix(-1 -8.14215e-08 -1.07231e-07 1 125.003 9.99994)",
    fill: "currentColor"
  }), /*#__PURE__*/_react["default"].createElement("rect", {
    opacity: 0.6,
    width: 5.00001,
    height: 50.0002,
    rx: 2.5,
    transform: "matrix(-1 -4.89277e-08 -1.78445e-07 1 115.001 14.9999)",
    fill: "currentColor"
  }));
};

var PatternWave = (0, _styledComponents["default"])(Svg).withConfig({
  displayName: "PatternWave",
  componentId: "t3wzw-0"
})(function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      width = _ref.width,
      fontSize = _ref.fontSize,
      theme = _ref.theme;
  return _objectSpread(_objectSpread({}, color && {
    color: (0, _moonUtils.themed)('color', color)(theme)
  }), {}, {
    height: height,
    width: width,
    fontSize: fontSize
  });
});
PatternWave.defaultProps = {
  color: "piccolo.100"
};
var _default = PatternWave;
exports["default"] = _default;
