// @heathmont/moon-components/lib/tagInput/TagInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.DropdownWrapper = exports.RemoveAllButton = exports.ArrowContainer = exports.TagRemove = exports.TagInputMain = exports.TagInputWrapper = exports.TagInputField = exports.Tag = exports.RemoveBubble = void 0;

var _moonUtils = require("@heathmont/moon-utils");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Bubble = _interopRequireDefault(require("../bubble/Bubble"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var RemoveBubble = (0, _styledComponents["default"])(_Bubble["default"]).withConfig({
  displayName: "TagInput__RemoveBubble",
  componentId: "sc-1qkcwn0-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    position: 'absolute',
    bottom: '100%',
    right: 0,
    pointerEvents: 'none',
    transition: "opacity ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    opacity: 0,
    width: 'auto'
  };
});
exports.RemoveBubble = RemoveBubble;

var Tag = _styledComponents["default"].div.withConfig({
  displayName: "TagInput__Tag",
  componentId: "sc-1qkcwn0-1"
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    display: 'inline-flex',
    alignItems: 'center',
    backgroundColor: theme.color.goku[100],
    borderRadius: (0, _moonUtils.rem)(8),
    padding: (0, _moonUtils.rem)(8),
    color: theme.color.bulma[100],
    fontSize: (0, _moonUtils.rem)(12),
    minHeight: (0, _moonUtils.rem)(30),
    fontWeight: theme.fontWeight.normal,
    position: 'relative',
    gap: (0, _moonUtils.rem)(6)
  };
});

exports.Tag = Tag;

var TagInputField = _styledComponents["default"].input.withConfig({
  displayName: "TagInput__TagInputField",
  componentId: "sc-1qkcwn0-2"
})(function (_ref3) {
  var theme = _ref3.theme;
  return {
    fontSize: (0, _moonUtils.rem)(16),
    border: 'none',
    background: 'none',
    outline: 'none',
    caretColor: theme.color.piccolo[100],
    color: theme.color.bulma[100],
    minWidth: (0, _moonUtils.rem)(52),
    margin: "0 0 ".concat((0, _moonUtils.rem)(4), " 0"),
    flex: 1,
    '&::placeholder': {
      color: theme.color.trunks[100],
      opacity: 1
    }
  };
});

exports.TagInputField = TagInputField;

var TagInputWrapper = _styledComponents["default"].div.withConfig({
  displayName: "TagInput__TagInputWrapper",
  componentId: "sc-1qkcwn0-3"
})({
  position: 'relative'
});

exports.TagInputWrapper = TagInputWrapper;

var TagInputMain = _styledComponents["default"].div.withConfig({
  displayName: "TagInput__TagInputMain",
  componentId: "sc-1qkcwn0-4"
})(function (_ref4) {
  var theme = _ref4.theme;
  return [_defineProperty({
    display: 'flex',
    width: '100%',
    minHeight: (0, _moonUtils.rem)(48),
    alignItems: 'center',
    justifyContent: 'flex-start',
    flexWrap: 'wrap',
    position: 'relative',
    padding: "".concat((0, _moonUtils.rem)(8), " ").concat((0, _moonUtils.rem)(40), " ").concat((0, _moonUtils.rem)(4), " ").concat((0, _moonUtils.rem)(12)),
    backgroundColor: theme.color.gohan[100],
    border: "".concat(theme.borderWidth, "px solid ").concat(theme.color.beerus[100]),
    borderRadius: (0, _moonUtils.rem)(theme.radius["default"]),
    transition: "border-color ".concat(theme.transitionDuration["default"], "s ease"),
    '&:hover': {
      borderColor: theme.color.goku[40]
    },
    '&:focus-within': {
      borderColor: theme.color.piccolo[100]
    }
  }, Tag, {
    margin: "0 ".concat((0, _moonUtils.rem)(4), " ").concat((0, _moonUtils.rem)(4), " 0")
  }), function (_ref6) {
    var focused = _ref6.focused;
    return focused && {
      borderColor: theme.color.piccolo[100],
      '&:hover': {
        borderColor: theme.color.piccolo[100]
      }
    };
  }, function (_ref7) {
    var error = _ref7.error;
    return error && {
      borderColor: theme.color.chiChi[100],
      '&:focus-within, &:hover': {
        borderColor: theme.color.chiChi[100]
      }
    };
  }, function (_ref8) {
    var disabled = _ref8.disabled;
    return disabled && _defineProperty({
      opacity: 0.5,
      cursor: 'not-allowed'
    }, TagInputField, {
      cursor: 'not-allowed'
    });
  }];
});

exports.TagInputMain = TagInputMain;

var TagRemove = _styledComponents["default"].button.withConfig({
  displayName: "TagInput__TagRemove",
  componentId: "sc-1qkcwn0-5"
})(function (_ref10) {
  var _ref11;

  var theme = _ref10.theme;
  return _ref11 = {
    appearance: 'none',
    background: 'none',
    border: 'none',
    padding: 0,
    fontSize: (0, _moonUtils.rem)(16),
    display: 'inline-flex',
    alignItems: 'center',
    color: theme.color.trunks[100],
    cursor: 'pointer',
    transition: "color ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    position: 'relative',
    outline: 'none'
  }, _defineProperty(_ref11, RemoveBubble, {
    marginRight: (0, _moonUtils.rem)(-12),
    marginBottom: (0, _moonUtils.rem)(18)
  }), _defineProperty(_ref11, '&:hover, &:focus', _defineProperty({
    color: theme.color.chiChi[100]
  }, RemoveBubble, {
    opacity: 1
  })), _ref11;
});

exports.TagRemove = TagRemove;

var ArrowContainer = _styledComponents["default"].div.withConfig({
  displayName: "TagInput__ArrowContainer",
  componentId: "sc-1qkcwn0-6"
})(function (_ref12) {
  var theme = _ref12.theme;
  return {
    position: 'absolute',
    top: '50%',
    right: (0, _moonUtils.rem)(18),
    fontSize: (0, _moonUtils.rem)(16),
    transform: 'translateY(-50%)',
    color: theme.color.trunks[100]
  };
});

exports.ArrowContainer = ArrowContainer;

var RemoveAllButton = _styledComponents["default"].button.withConfig({
  displayName: "TagInput__RemoveAllButton",
  componentId: "sc-1qkcwn0-7"
})(function (_ref13) {
  var _ref14;

  var theme = _ref13.theme;
  return _ref14 = {
    background: 'none',
    border: 'none',
    padding: 0,
    fontSize: (0, _moonUtils.rem)(16),
    color: theme.color.trunks[100],
    cursor: 'pointer',
    display: 'flex',
    alignItems: 'center',
    position: 'absolute',
    top: (0, _moonUtils.rem)(16),
    right: (0, _moonUtils.rem)(16),
    transition: "color ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    outline: 'none',
    whiteSpace: 'nowrap'
  }, _defineProperty(_ref14, RemoveBubble, {
    marginRight: (0, _moonUtils.rem)(-12),
    marginBottom: (0, _moonUtils.rem)(24)
  }), _defineProperty(_ref14, '&:hover, &:focus', _defineProperty({
    color: theme.color.piccolo[100]
  }, RemoveBubble, {
    opacity: 1
  })), _ref14;
});

exports.RemoveAllButton = RemoveAllButton;

var DropdownWrapper = _styledComponents["default"].div.withConfig({
  displayName: "TagInput__DropdownWrapper",
  componentId: "sc-1qkcwn0-8"
})({
  top: '100%',
  left: 0,
  position: 'absolute',
  marginTop: (0, _moonUtils.rem)(10),
  maxWidth: '100%'
});

exports.DropdownWrapper = DropdownWrapper;
var _default = {
  RemoveBubble: RemoveBubble,
  Tag: Tag,
  TagInputWrapper: TagInputWrapper,
  TagInputMain: TagInputMain,
  TagInputField: TagInputField,
  TagRemove: TagRemove,
  ArrowContainer: ArrowContainer,
  RemoveAllButton: RemoveAllButton,
  DropdownWrapper: DropdownWrapper
};
exports["default"] = _default;
