// @heathmont/moon-components/lib/dialogContent/DialogContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _rem = _interopRequireDefault(require("polished/lib/helpers/rem"));

var _dialog = require("@reach/dialog");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * 1. Arbitrary figure from design.
 */
var DialogContent = (0, _styledComponents["default"])(_dialog.DialogContent).withConfig({
  displayName: "DialogContent",
  componentId: "sc-1lj8flh-0"
})(function (_ref) {
  var space = _ref.theme.space;
  return [{
    margin: '0 auto',
    position: 'relative',
    maxWidth: (0, _rem["default"])(608)
    /* [1] */
    ,
    padding: (0, _rem["default"])(space["default"]),
    outline: 'none'
  }, function (_ref2) {
    var maxWidth = _ref2.maxWidth;
    return {
      maxWidth: maxWidth
    };
  }, function (_ref3) {
    var position = _ref3.position;
    return position === 'TOP' ? {
      margin: '8vh auto'
    } : {
      top: '50%',
      transform: 'translateY(-50%)'
    };
  }];
});
var _default = DialogContent;
exports["default"] = _default;
