// @heathmont/moon-components/lib/mediaObject/MediaObject.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var MediaObject = _styledComponents["default"].div.withConfig({
  displayName: "MediaObject",
  componentId: "sc-1gcatrn-0"
})(function (_ref) {
  var _ref$alignItems = _ref.alignItems,
      alignItems = _ref$alignItems === void 0 ? 'center' : _ref$alignItems,
      _ref$gridGap = _ref.gridGap,
      gridGap = _ref$gridGap === void 0 ? 'default' : _ref$gridGap,
      _ref$mediaWidth = _ref.mediaWidth,
      mediaWidth = _ref$mediaWidth === void 0 ? 'large' : _ref$mediaWidth,
      reverse = _ref.reverse,
      _ref$textAlign = _ref.textAlign,
      textAlign = _ref$textAlign === void 0 ? 'left' : _ref$textAlign,
      theme = _ref.theme;
  return [{
    alignItems: alignItems,
    gridGap: (0, _moonUtils.themed)('space', gridGap)(theme),
    textAlign: textAlign,
    display: 'grid',
    gridTemplate: "max-content / ".concat((0, _moonUtils.themed)('space', mediaWidth)(theme), " auto"),
    'img,svg': {
      display: 'block',
      width: '100%',
      height: 'auto'
    }
  }, reverse && {
    direction: 'rtl',
    '& > *': {
      direction: 'ltr'
    }
  }];
});

var _default = MediaObject;
exports["default"] = _default;
