// @heathmont/moon-components/lib/es/header/Header.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import styled from 'styled-components';
import { mq, rem } from '@heathmont/moon-utils';
/**
 * 1. A tiny boost to the vertical rhythm of succeeding elements.
 */

var Header = styled.header.withConfig({
  displayName: "Header",
  componentId: "sbaz2c-0"
})(function (_ref) {
  var space = _ref.theme.space;
  return {
    marginTop: rem(space["default"]),
    paddingBottom: rem(space.small)
    /* [1] */
    ,
    position: 'relative',
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'baseline'
  };
});
var HeaderTitle = styled.h1.withConfig({
  displayName: "Header__HeaderTitle",
  componentId: "sbaz2c-1"
})(function (_ref2) {
  var _ref2$theme = _ref2.theme,
      breakpoint = _ref2$theme.breakpoint,
      fontWeight = _ref2$theme.fontWeight,
      space = _ref2$theme.space;
  return _defineProperty({
    display: 'inline-block',
    margin: "0 ".concat(rem(space.large), " 0 0"),
    fontSize: rem(20),
    lineHeight: rem(24),
    fontWeight: fontWeight.semibold
  }, mq(breakpoint.small), {
    fontSize: rem(32),
    lineHeight: rem(40)
  });
});
export { HeaderTitle };
export default Header;
