// @heathmont/moon-components/lib/es/header/HeaderBack.js
var _excluded = ["children", "element"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import hideVisually from 'polished/lib/mixins/hideVisually';
import { rem } from '@heathmont/moon-utils';
import styled from 'styled-components';
import IconArrowLeft from '../private/icons/IconArrowLeft';
/**
 * Types
 *
 * 1. String or React Component.
 * 2. Allows us to pass in any props associated to the `element`.
 */

/* eslint-disable @typescript-eslint/no-explicit-any */

/* eslint-enable @typescript-eslint/no-explicit-any */
var HeaderBackLink = styled.a.withConfig({
  displayName: "HeaderBack__HeaderBackLink",
  componentId: "sc-15sp96a-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      space = _ref$theme.space;
  return _defineProperty({
    margin: "0 ".concat(rem(space.small), " 0 0"),
    padding: 0,
    alignSelf: 'center',
    verticalAlign: 'middle',
    backgroundColor: 'transparent',
    border: 0,
    appearance: 'none'
  }, "> ".concat(IconArrowLeft), {
    color: color.piccolo[100],
    fontSize: rem(space.medium)
  });
});
/**
 * Component
 */

var HeaderBack = function HeaderBack(_ref3) {
  var children = _ref3.children,
      _ref3$element = _ref3.element,
      element = _ref3$element === void 0 ? 'a' : _ref3$element,
      props = _objectWithoutProperties(_ref3, _excluded);

  return /*#__PURE__*/React.createElement(HeaderBackLink, _extends({
    as: element
  }, props), /*#__PURE__*/React.createElement(IconArrowLeft, null), /*#__PURE__*/React.createElement("span", {
    style: hideVisually()
  }, children));
};

export default HeaderBack;
