// @heathmont/moon-components/lib/progressLinear/ProgressLinear.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ProgressBarWrapper = _styledComponents["default"].div.withConfig({
  displayName: "ProgressLinear__ProgressBarWrapper",
  componentId: "sc-1wwz7u1-0"
})({
  position: 'relative'
});

var CurrentProgress = _styledComponents["default"].div.withConfig({
  displayName: "ProgressLinear__CurrentProgress",
  componentId: "sc-1wwz7u1-1"
})(function (_ref) {
  var _ref$lineHeight = _ref.lineHeight,
      lineHeight = _ref$lineHeight === void 0 ? 12 : _ref$lineHeight,
      _ref$fontSize = _ref.fontSize,
      fontSize = _ref$fontSize === void 0 ? 10 : _ref$fontSize,
      _ref$theme = _ref.theme,
      fontWeight = _ref$theme.fontWeight,
      color = _ref$theme.color;
  return {
    position: 'absolute',
    top: 0,
    left: '50%',
    transform: 'translateX(-50%)',
    color: color.bulma[100],
    fontSize: (0, _moonUtils.rem)(fontSize),
    lineHeight: (0, _moonUtils.rem)(lineHeight),
    letterSpacing: 1,
    fontWeight: fontWeight.semibold,
    textTransform: 'uppercase',
    whiteSpace: 'nowrap',
    textShadow: "-".concat((0, _moonUtils.rem)(1), " 0 ").concat(color.gohan['100'], ", 0 ").concat((0, _moonUtils.rem)(1), " ").concat(color.gohan['100'], ", ").concat((0, _moonUtils.rem)(1), " 0 ").concat(color.gohan['100'], ", 0 -").concat((0, _moonUtils.rem)(1), " ").concat(color.gohan['100'])
  };
});

var Progress = _styledComponents["default"].div.withConfig({
  displayName: "ProgressLinear__Progress",
  componentId: "sc-1wwz7u1-2"
})(function (_ref2) {
  var theme = _ref2.theme,
      value = _ref2.value,
      backgroundColor = _ref2.backgroundColor,
      _ref2$height = _ref2.height,
      height = _ref2$height === void 0 ? 12 : _ref2$height;
  return {
    minWidth: (0, _moonUtils.rem)(100),
    backgroundColor: (0, _moonUtils.themed)('color', backgroundColor)(theme) || theme.color.goku['100'],
    height: (0, _moonUtils.rem)(height),
    position: 'relative',
    borderRadius: (0, _moonUtils.rem)(60),
    '&::before': {
      content: '""',
      position: 'absolute',
      backgroundColor: theme.color.piccolo['100'],
      height: '100%',
      width: "".concat(value, "%"),
      borderRadius: (0, _moonUtils.rem)(60)
    }
  };
});

var ProgressLinear = function ProgressLinear(_ref3) {
  var value = _ref3.value,
      text = _ref3.text,
      backgroundColor = _ref3.backgroundColor,
      height = _ref3.height,
      fontSize = _ref3.fontSize;
  return /*#__PURE__*/_react["default"].createElement(ProgressBarWrapper, null, /*#__PURE__*/_react["default"].createElement(Progress, {
    height: height,
    value: value,
    backgroundColor: backgroundColor
  }), /*#__PURE__*/_react["default"].createElement(CurrentProgress, {
    lineHeight: height,
    fontSize: fontSize
  }, text));
};

var _default = ProgressLinear;
exports["default"] = _default;
