// @heathmont/moon-components/lib/es/form/Form.js
var _excluded = ["children", "legend", "maxWidth", "fullWidth"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import styled from 'styled-components';
import hideVisually from 'polished/lib/mixins/hideVisually';
import Stack from '../stack/Stack';
/**
 * Types
 */

/* Prevents TypeScript errors when using HTMLFormElement directly */

var Fieldset = styled.fieldset.withConfig({
  displayName: "Form__Fieldset",
  componentId: "sc-11bajar-0"
})({
  border: 0
});
/**
 * Components
 */

var Form = function Form(_ref) {
  var children = _ref.children,
      legend = _ref.legend,
      maxWidth = _ref.maxWidth,
      fullWidth = _ref.fullWidth,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement("form", _extends({
    style: {
      maxWidth: maxWidth,
      width: fullWidth ? '100%' : 'auto'
    }
  }, props), /*#__PURE__*/React.createElement(Fieldset, null, /*#__PURE__*/React.createElement("legend", {
    style: hideVisually()
  }, legend), /*#__PURE__*/React.createElement(Stack, {
    as: "ul"
  }, children)));
};
/**
 * `<any>` justification: https://coingaming.atlassian.net/browse/SPO-4963.
 */


export default Form;
