// @heathmont/moon-components/lib/dialogOverlay/DialogOverlay.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _rgba = _interopRequireDefault(require("polished/lib/color/rgba"));

var _dialog = require("@reach/dialog");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * For further info, see the Reach UI documentation:
 * https://ui.reach.tech/dialog/
 */
var DialogOverlay = (0, _styledComponents["default"])(_dialog.DialogOverlay).withConfig({
  displayName: "DialogOverlay",
  componentId: "sc-1cg17mq-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      color = _ref$theme.color,
      zIndex = _ref$theme.zIndex;
  return {
    position: 'fixed',
    top: '0',
    right: '0',
    bottom: '0',
    left: '0',
    overflow: 'auto',
    backgroundColor: color.gohan[100] && (0, _rgba["default"])(color.gohan[100], 0.75),
    zIndex: zIndex.dialog
  };
});
var _default = DialogOverlay;
exports["default"] = _default;
