// @heathmont/moon-components/lib/es/tagList/TagList.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { rem } from '@heathmont/moon-utils';
import { IconCopy } from '@heathmont/moon-assets';
import styled from 'styled-components';
import Bubble from '../bubble/Bubble';
import { Tag } from '../tagInput/TagInput';
export var CopyBubble = styled(Bubble).withConfig({
  displayName: "TagList__CopyBubble",
  componentId: "sc-1ebbyur-0"
})(function (_ref) {
  var theme = _ref.theme;
  return {
    position: 'absolute',
    top: '100%',
    right: 0,
    pointerEvents: 'none',
    transition: "opacity ".concat(theme.transitionDuration["default"], "s ease-in-out"),
    opacity: 0,
    marginTop: rem(16),
    marginRight: rem(-8)
  };
});
export var TagListWrapper = styled.div.withConfig({
  displayName: "TagList__TagListWrapper",
  componentId: "sc-1ebbyur-1"
})(function (_ref2) {
  var theme = _ref2.theme;
  return _defineProperty({
    display: 'inline-block',
    position: 'relative',
    marginBottom: rem(-4)
  }, Tag, {
    color: theme.color.bulma[100]
  });
});
export var TagListInternal = styled.div.withConfig({
  displayName: "TagList__TagListInternal",
  componentId: "sc-1ebbyur-2"
})(_defineProperty({
  display: 'flex',
  flexWrap: 'wrap'
}, Tag, {
  margin: "0 ".concat(rem(4), " ").concat(rem(4), " 0"),
  '&:last-child': {
    marginRight: 0
  }
}));
export var ShowMore = styled.button.withConfig({
  displayName: "TagList__ShowMore",
  componentId: "sc-1ebbyur-3"
})(function (_ref4) {
  var theme = _ref4.theme;
  return {
    display: 'inline-flex',
    alignItems: 'center',
    backgroundColor: theme.color.goku[100],
    borderRadius: rem(8),
    padding: rem(8),
    color: theme.color.bulma[100],
    fontSize: rem(12),
    height: rem(30),
    fontWeight: theme.fontWeight.normal,
    position: 'relative',
    cursor: 'pointer',
    border: 'none',
    transition: "color ".concat(theme.transitionDuration["default"], "s ease, background-color ").concat(theme.transitionDuration["default"], "s ease"),
    marginBottom: rem(4),
    '&:hover, &:focus': {
      color: theme.color.goten[100],
      backgroundColor: theme.color.piccolo[100]
    }
  };
});
export var Dropdown = styled.div.withConfig({
  displayName: "TagList__Dropdown",
  componentId: "sc-1ebbyur-4"
})(function (_ref5) {
  var theme = _ref5.theme;
  return _defineProperty({
    display: 'flex',
    flexWrap: 'wrap',
    background: theme.color.goten[100],
    borderRadius: rem(theme.radius["default"]),
    boxShadow: theme.boxShadow,
    position: 'absolute',
    padding: "".concat(rem(8), " ").concat(rem(44), " ").concat(rem(4), " ").concat(rem(12)),
    width: '100%',
    maxHeight: rem(160),
    overflow: 'auto',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    zIndex: theme.zIndex.toggle
  }, Tag, {
    margin: "0 ".concat(rem(4), " ").concat(rem(4), " 0")
  });
});
export var CopyWrapper = styled.div.withConfig({
  displayName: "TagList__CopyWrapper",
  componentId: "sc-1ebbyur-5"
})({
  position: 'absolute',
  top: rem(8),
  right: rem(12)
});
export var CopyButton = styled.button.withConfig({
  displayName: "TagList__CopyButton",
  componentId: "sc-1ebbyur-6"
})(function (_ref7) {
  var _hoverFocus, _ref8;

  var theme = _ref7.theme;
  return _ref8 = {
    height: rem(24),
    width: rem(24),
    border: 'none',
    background: 'none',
    outline: 'none',
    padding: 0,
    cursor: 'pointer',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative'
  }, _defineProperty(_ref8, IconCopy, {
    height: rem(16)
  }), _defineProperty(_ref8, '&::before', {
    content: '""',
    position: 'absolute',
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    transitionDuration: "".concat(theme.transitionDuration["default"], "s"),
    transitionTimingFunction: 'ease',
    backgroundColor: theme.color.piccolo[100],
    borderRadius: '50%',
    opacity: 0,
    transform: 'none',
    transitionProperty: 'transform, opacity'
  }), _defineProperty(_ref8, '&:hover, &:focus', (_hoverFocus = {}, _defineProperty(_hoverFocus, CopyBubble, {
    opacity: 1
  }), _defineProperty(_hoverFocus, IconCopy, {
    color: theme.color.piccolo[100]
  }), _defineProperty(_hoverFocus, '&::before', {
    transform: 'scale(1.4)',
    opacity: 0.1
  }), _hoverFocus)), _ref8;
});
export default {
  CopyBubble: CopyBubble,
  TagListWrapper: TagListWrapper,
  TagListInternal: TagListInternal,
  ShowMore: ShowMore,
  Dropdown: Dropdown,
  CopyWrapper: CopyWrapper,
  CopyButton: CopyButton
};
