// @heathmont/moon-components/lib/keylineText/KeylineText.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _Caption = _interopRequireDefault(require("../caption/Caption"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var KeylineText = (0, _styledComponents["default"])(_Caption["default"]).withConfig({
  displayName: "KeylineText",
  componentId: "wzk90x-0"
})(function (_ref) {
  var _ref$theme = _ref.theme,
      borderWidth = _ref$theme.borderWidth,
      color = _ref$theme.color,
      space = _ref$theme.space;
  return {
    display: 'flex',
    width: '100%',
    alignItems: 'center',
    whiteSpace: 'nowrap',
    color: color.trunks[100],
    '&::before, &::after': {
      content: '""',
      display: 'inline-block',
      width: '100%',
      flex: '0 1 auto',
      height: borderWidth,
      backgroundColor: color.beerus[100]
    },
    '&::before': {
      marginRight: (0, _moonUtils.rem)(space.small)
    },
    '&::after': {
      marginLeft: (0, _moonUtils.rem)(space.small)
    }
  };
});
var _default = KeylineText;
exports["default"] = _default;
