// @heathmont/moon-components/lib/dialog/Dialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconClose = _interopRequireDefault(require("../private/icons/IconClose"));

var _DialogOverlay = _interopRequireDefault(require("../dialogOverlay/DialogOverlay"));

var _DialogContent = _interopRequireDefault(require("../dialogContent/DialogContent"));

var _toggle = require("./private/toggle");

var _layout = require("./private/layout");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Dialog = function Dialog(_ref) {
  var children = _ref.children,
      maxWidth = _ref.maxWidth,
      _ref$isOpen = _ref.isOpen,
      isOpen = _ref$isOpen === void 0 ? false : _ref$isOpen,
      onDismiss = _ref.onDismiss,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? 'CENTER' : _ref$position,
      _ref$disableScrollLoc = _ref.disableScrollLock,
      disableScrollLock = _ref$disableScrollLoc === void 0 ? false : _ref$disableScrollLoc;
  return /*#__PURE__*/_react["default"].createElement(_DialogOverlay["default"], {
    isOpen: isOpen && isOpen,
    onDismiss: onDismiss && onDismiss,
    dangerouslyBypassScrollLock: disableScrollLock
  }, /*#__PURE__*/_react["default"].createElement(_DialogContent["default"], {
    maxWidth: maxWidth,
    position: position
  }, /*#__PURE__*/_react["default"].createElement(_layout.DialogContainer, null, /*#__PURE__*/_react["default"].createElement(_toggle.DialogToggle, {
    type: "button",
    onClick: onDismiss && onDismiss
  }, /*#__PURE__*/_react["default"].createElement(_IconClose["default"], null), /*#__PURE__*/_react["default"].createElement(_toggle.DialogToggleText, null, "Close dialog")), /*#__PURE__*/_react["default"].createElement(_layout.DialogMain, null, children))));
};

var _default = Dialog;
exports["default"] = _default;
