// @heathmont/moon-components/lib/search/private/components/Form.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchForm = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _Input = require("./Input");

var _Results = require("./Results");

var _settings = require("./settings");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/**
 * Search Form
 * The functional container for the search component.
 * We aren't using Design System form because we use focus-within logic here
 */
var SearchForm = _styledComponents["default"].form.withConfig({
  displayName: "Form__SearchForm",
  componentId: "sc-7uq563-0"
})(function (_ref) {
  var theme = _ref.theme;
  return _defineProperty({
    fontSize: (0, _moonUtils.rem)(theme.space["default"]),
    position: 'relative',
    overflow: 'visible',
    width: '100%'
  }, "&:focus-within, ".concat(_Input.SearchInputActive), _defineProperty({
    zIndex: _settings.zIndex.searchInputActive
  }, "~ ".concat(_Results.Results, ":not(:empty)"), (0, _Results.resultsActive)(theme)));
});

exports.SearchForm = SearchForm;
