// @heathmont/moon-core/lib/search/private/components/Input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchBox = exports.SearchInput = exports.SearchMenuIcon = exports.SearchInputIcon = exports.SearchClearIcon = exports.SearchClearButton = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonComponents = require("@heathmont/moon-components");

var _moonUtils = require("@heathmont/moon-utils");

var _IconErrorCircle = _interopRequireDefault(require("../../../private/icons/IconErrorCircle"));

var _IconMenu = _interopRequireDefault(require("../../../private/icons/IconMenu"));

var _IconSearch = _interopRequireDefault(require("../../../private/icons/IconSearch"));

var _settings = require("./settings");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var borderWidth = _settings.inputConfig.borderWidth,
    padding = _settings.inputConfig.padding;
var SearchClearButton = (0, _styledComponents["default"])(_moonComponents.Button).withConfig({
  displayName: "Input__SearchClearButton",
  componentId: "vvx0zy-0"
})(function (_ref) {
  var space = _ref.theme.space;
  return {
    position: 'absolute',
    top: 0,
    right: borderWidth,
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    padding: "0 ".concat((0, _moonUtils.rem)(space.small)),
    border: 'none',
    background: 'transparent',
    appearance: 'none',
    zIndex: _settings.zIndex.searchIcon
  };
});
exports.SearchClearButton = SearchClearButton;
var SearchClearIcon = (0, _styledComponents["default"])(_IconErrorCircle["default"]).withConfig({
  displayName: "Input__SearchClearIcon",
  componentId: "vvx0zy-1"
})(function (_ref2) {
  var _ref2$theme = _ref2.theme,
      color = _ref2$theme.color,
      fontSize = _ref2$theme.fontSize,
      space = _ref2$theme.space;
  return {
    color: color.trunks[100],
    fontSize: (0, _moonUtils.rem)(fontSize.body),
    width: space["default"],
    height: space["default"]
  };
});
exports.SearchClearIcon = SearchClearIcon;
var SearchInputIcon = (0, _styledComponents["default"])(_IconSearch["default"]).withConfig({
  displayName: "Input__SearchInputIcon",
  componentId: "vvx0zy-2"
})(function (_ref3) {
  var inputSize = _ref3.inputSize,
      _ref3$theme = _ref3.theme,
      color = _ref3$theme.color,
      fontSize = _ref3$theme.fontSize,
      space = _ref3$theme.space;
  return {
    position: 'absolute',
    top: '50%',
    left: (0, _moonUtils.rem)(padding[inputSize] + borderWidth),
    transform: 'translateY(-50%)',
    zIndex: _settings.zIndex.searchIcon,
    fontSize: (0, _moonUtils.rem)(fontSize.body),
    color: color.trunks[100],
    pointerEvents: 'none',
    width: space["default"],
    height: space["default"]
  };
});
exports.SearchInputIcon = SearchInputIcon;
var SearchMenuIcon = (0, _styledComponents["default"])(_IconMenu["default"]).withConfig({
  displayName: "Input__SearchMenuIcon",
  componentId: "vvx0zy-3"
})(function (_ref4) {
  var _ref4$theme = _ref4.theme,
      color = _ref4$theme.color,
      fontSize = _ref4$theme.fontSize,
      space = _ref4$theme.space;
  return {
    position: 'absolute',
    top: '50%',
    right: (0, _moonUtils.rem)(space["default"]),
    color: color.trunks[100],
    fontSize: (0, _moonUtils.rem)(fontSize.body),
    transform: 'translateY(-50%)',
    zIndex: _settings.zIndex.searchIcon,
    width: space["default"],
    height: space["default"]
  };
});
exports.SearchMenuIcon = SearchMenuIcon;

var SearchInput = _styledComponents["default"].input.withConfig({
  displayName: "Input__SearchInput",
  componentId: "vvx0zy-4"
})(function (_ref5) {
  var hasBorder = _ref5.hasBorder,
      inputSize = _ref5.inputSize,
      _ref5$theme = _ref5.theme,
      breakpoint = _ref5$theme.breakpoint,
      color = _ref5$theme.color,
      fontSize = _ref5$theme.fontSize,
      fontWeight = _ref5$theme.fontWeight,
      radius = _ref5$theme.radius,
      space = _ref5$theme.space;
  return _defineProperty({
    fontWeight: fontWeight.normal,
    fontSize: (0, _moonUtils.rem)(fontSize.body),
    width: '100%',
    lineHeight: 1,
    position: 'relative',
    paddingTop: 0,
    paddingRight: (0, _moonUtils.rem)(padding[inputSize] + space["default"]),
    paddingBottom: 0,
    paddingLeft: (0, _moonUtils.rem)(2 * padding[inputSize] + space["default"]),
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    backgroundColor: hasBorder ? color.goku[100] : color.gohan[100],
    color: color.trunks[100],
    borderRadius: (0, _moonUtils.rem)(radius["default"]),
    border: hasBorder ? "".concat((0, _moonUtils.rem)(borderWidth), " solid transparent") : 'none',
    outline: 0,
    zIndex: _settings.zIndex.searchInput,
    '&::placeholder': {
      color: 'inherit'
    },
    '&:hover': {
      backgroundColor: color.gohan[100],
      borderColor: color.beerus[100]
    },
    '&:active, &:focus': {
      backgroundColor: color.gohan[100],
      borderColor: color.piccolo[100],
      color: color.bulma[100]
    }
  }, (0, _moonUtils.mq)(breakpoint.medium, 'max-width'), {
    '.popup-open &': {
      borderColor: color.piccolo[100]
    }
  });
});

exports.SearchInput = SearchInput;

var SearchBox = _styledComponents["default"].span.withConfig({
  displayName: "Input__SearchBox",
  componentId: "vvx0zy-5"
})(function (_ref7) {
  var fontSize = _ref7.theme.fontSize;
  return {
    display: 'flex',
    position: 'relative',
    fontSize: (0, _moonUtils.rem)(fontSize.body),
    overflow: 'visible',
    width: '100%'
  };
}, function (_ref8) {
  var size = _ref8.size;
  return !!size && {
    height: (0, _moonUtils.rem)(_settings.inputConfig.height[size])
  };
});

exports.SearchBox = SearchBox;
