// @heathmont/moon-core/lib/search/private/components/Styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Results = exports.ModalClose = exports.FlexWrapper = exports.SearchForm = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _moonComponents = require("@heathmont/moon-components");

var _settings = require("./settings");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SearchForm = _styledComponents["default"].form.withConfig({
  displayName: "Styles__SearchForm",
  componentId: "sc-1urf7j8-0"
})({
  display: 'flex',
  position: 'relative',
  overflow: 'visible',
  width: '100%'
});

exports.SearchForm = SearchForm;

var FlexWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Styles__FlexWrapper",
  componentId: "sc-1urf7j8-1"
})(function (_ref) {
  var space = _ref.theme.space;
  return {
    position: 'relative',
    display: 'flex',
    justifyContent: 'space-between',
    gap: (0, _moonUtils.rem)(space.small)
  };
});

exports.FlexWrapper = FlexWrapper;
var ModalClose = (0, _styledComponents["default"])(_moonComponents.Button).withConfig({
  displayName: "Styles__ModalClose",
  componentId: "sc-1urf7j8-2"
})(function (_ref2) {
  var _ref3;

  var _ref2$theme = _ref2.theme,
      base = _ref2$theme.base,
      breakpoint = _ref2$theme.breakpoint,
      color = _ref2$theme.color,
      fontWeight = _ref2$theme.fontWeight,
      space = _ref2$theme.space;
  return _ref3 = {
    position: 'relative',
    display: 'flex',
    flex: 0,
    border: 0,
    lineHeight: 1,
    padding: "0 ".concat((0, _moonUtils.rem)(space["default"])),
    fontWeight: fontWeight.normal,
    fontSize: (0, _moonUtils.rem)(base.fontSize),
    overflow: 'visible',
    color: color.piccolo[100]
  }, _defineProperty(_ref3, '&:hover:not([disabled])', {
    color: color.piccolo[100]
  }), _defineProperty(_ref3, (0, _moonUtils.mq)(breakpoint.medium, 'min-width'), {
    display: 'none'
  }), _ref3;
});
exports.ModalClose = ModalClose;

var Results = _styledComponents["default"].div.withConfig({
  displayName: "Styles__Results",
  componentId: "sc-1urf7j8-3"
})(function (_ref4) {
  var _ref4$theme = _ref4.theme,
      borderWidth = _ref4$theme.borderWidth,
      boxShadow = _ref4$theme.boxShadow,
      breakpoint = _ref4$theme.breakpoint,
      color = _ref4$theme.color,
      radius = _ref4$theme.radius,
      space = _ref4$theme.space;
  return _defineProperty({
    position: 'absolute',
    left: 0,
    top: (0, _moonUtils.rem)(space.small),
    color: color.trunks[100],
    width: '100%',
    zIndex: _settings.zIndex.searchResults
  }, (0, _moonUtils.mq)(breakpoint.medium, 'min-width'), {
    width: 'auto',
    minWidth: '300px',
    maxWidth: '100%',
    boxSizing: 'border-box',
    padding: "".concat((0, _moonUtils.rem)(space.small), " ").concat((0, _moonUtils.rem)(space.xsmall)),
    borderRadius: (0, _moonUtils.rem)(radius["default"]),
    borderWidth: borderWidth,
    borderStyle: 'solid',
    borderColor: color.beerus[100],
    backgroundColor: color.gohan[100],
    boxShadow: boxShadow
  });
});

exports.Results = Results;
