// @heathmont/moon-components/lib/switch/Switch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _moonUtils = require("@heathmont/moon-utils");

var _hideVisually = _interopRequireDefault(require("polished/lib/mixins/hideVisually"));

var _IconMoon = _interopRequireDefault(require("../private/icons/IconMoon"));

var _IconSun = _interopRequireDefault(require("../private/icons/IconSun"));

var _Inline = _interopRequireDefault(require("../inline/Inline"));

var _Caption = _interopRequireDefault(require("../caption/Caption"));

var _excluded = ["className", "captionChecked", "captionUnchecked", "checked", "colorScheme", "id"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var switchWidthProperty = '--switch-width';
var switchWidth = "var(".concat(switchWidthProperty, ")");
var switchHeightProperty = '--switch-height';
var switchHeight = "var(".concat(switchHeightProperty, ")");

var Slider = _styledComponents["default"].span.withConfig({
  displayName: "Switch__Slider",
  componentId: "sc-11hx91j-0"
})(function (_ref) {
  var colorScheme = _ref.colorScheme,
      _ref$theme = _ref.theme,
      color = _ref$theme.color,
      themeColorScheme = _ref$theme.colorScheme,
      space = _ref$theme.space,
      transitionDuration = _ref$theme.transitionDuration;
  return {
    position: 'absolute',
    width: switchWidth,
    height: switchHeight,
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    cursor: 'pointer',
    backgroundColor: color.gohan[100],
    backgroundImage: colorScheme ? "".concat((0, _moonUtils.inlineSvg)( /*#__PURE__*/_react["default"].createElement(_IconMoon["default"], null)), ", ").concat((0, _moonUtils.inlineSvg)( /*#__PURE__*/_react["default"].createElement(_IconSun["default"], null))) : undefined,
    backgroundPosition: "left ".concat((0, _moonUtils.rem)(space.small), " center, right ").concat((0, _moonUtils.rem)(space.small), " center"),
    backgroundRepeat: 'no-repeat',
    borderRadius: "calc(".concat(switchWidth, " + ").concat((0, _moonUtils.rem)(2), ")"),
    transitionProperty: 'background-color',
    transitionDuration: "".concat(transitionDuration.slow, "s"),
    transitionTimingFunction: 'ease',
    '&::before': {
      '--switch-indicator-size': "calc(".concat(switchHeight, " - ").concat((0, _moonUtils.rem)(space.small), ")"),
      content: '""',
      position: 'absolute',
      width: 'var(--switch-indicator-size)',
      height: 'var(--switch-indicator-size)',
      left: (0, _moonUtils.rem)(space.xsmall),
      bottom: (0, _moonUtils.rem)(space.xsmall),
      backgroundColor: themeColorScheme ? color.goku[100] : color.trunks[100],
      borderRadius: '50%',
      transition: 'inherit',
      transitionProperty: 'background-color, transform',
      willChange: 'transform'
    }
  };
});

var Label = _styledComponents["default"].label.withConfig({
  displayName: "Switch__Label",
  componentId: "sc-11hx91j-1"
})(function (_ref2) {
  var _ref3;

  var space = _ref2.theme.space;
  return _ref3 = {}, _defineProperty(_ref3, switchWidthProperty, (0, _moonUtils.rem)(space.large * 2)), _defineProperty(_ref3, switchHeightProperty, (0, _moonUtils.rem)(space.large)), _defineProperty(_ref3, "position", 'relative'), _defineProperty(_ref3, "display", 'inline-block'), _defineProperty(_ref3, "width", switchWidth), _defineProperty(_ref3, "height", switchHeight), _defineProperty(_ref3, "flexShrink", 0), _ref3;
});

var Input = _styledComponents["default"].input.withConfig({
  displayName: "Switch__Input",
  componentId: "sc-11hx91j-2"
})(function (_ref4) {
  var colorScheme = _ref4.colorScheme,
      color = _ref4.theme.color;
  return _objectSpread(_objectSpread({}, (0, _hideVisually["default"])()), {}, _defineProperty({}, "&:checked + ".concat(Slider), {
    backgroundColor: colorScheme ? undefined : color.piccolo[100],
    '&::before': {
      backgroundColor: colorScheme ? undefined : color.goten[100],
      transform: "translateX(calc(".concat(switchWidth, " / 2))")
    }
  }));
});

Input.defaultProps = {
  type: 'checkbox'
};
var SwitchCaption = (0, _styledComponents["default"])(_Caption["default"]).withConfig({
  displayName: "Switch__SwitchCaption",
  componentId: "sc-11hx91j-3"
})(function (_ref5) {
  var checked = _ref5.checked,
      _ref5$theme = _ref5.theme,
      color = _ref5$theme.color,
      transitionDuration = _ref5$theme.transitionDuration;
  return [{
    transition: "color ".concat(transitionDuration["default"], "s")
  }, checked && {
    color: checked ? color.trunks[100] : color.bulma[100]
  }];
});

var Switch = function Switch(_ref6) {
  var className = _ref6.className,
      captionChecked = _ref6.captionChecked,
      captionUnchecked = _ref6.captionUnchecked,
      _ref6$checked = _ref6.checked,
      checked = _ref6$checked === void 0 ? false : _ref6$checked,
      colorScheme = _ref6.colorScheme,
      id = _ref6.id,
      props = _objectWithoutProperties(_ref6, _excluded);

  var autoId = id || "Switch-".concat((0, _moonUtils.uniqueId)());
  var labelProps = {
    className: className
  };

  var inputProps = _objectSpread({
    id: autoId,
    colorScheme: colorScheme,
    checked: checked
  }, props);

  var sliderProps = {
    colorScheme: colorScheme
  };

  if (!captionUnchecked && !captionChecked) {
    return /*#__PURE__*/_react["default"].createElement(Label, labelProps, /*#__PURE__*/_react["default"].createElement(Input, inputProps), /*#__PURE__*/_react["default"].createElement(Slider, sliderProps));
  }

  return /*#__PURE__*/_react["default"].createElement(_Inline["default"], {
    space: "small",
    style: {
      overflow: 'visible'
    }
  }, captionUnchecked && /*#__PURE__*/_react["default"].createElement(SwitchCaption, {
    checked: checked
  }, captionUnchecked), /*#__PURE__*/_react["default"].createElement(Label, labelProps, /*#__PURE__*/_react["default"].createElement(Input, inputProps), /*#__PURE__*/_react["default"].createElement(Slider, sliderProps)), captionChecked && /*#__PURE__*/_react["default"].createElement(SwitchCaption, {
    checked: !checked
  }, captionChecked));
};

var _default = Switch;
exports["default"] = _default;
